<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.450">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Haky Im">
<meta name="dcterms.date" content="2024-08-29">
<meta name="description" content="compare ratxcan mixed effects with and without loco">

<title>HakyImLab Notes - compare ratxcan with and without loco</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": true,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>

  <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<link rel="stylesheet" href="../../styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../../index.html">
    <span class="navbar-title">HakyImLab Notes</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../about.html" rel="" target="">
 <span class="menu-text">Miscel</span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/hakyimlab/web-lab-notes-quarto" rel="" target=""><i class="bi bi-github" role="img">
</i> 
 <span class="menu-text"></span></a>
  </li>  
</ul>
            <div class="quarto-navbar-tools">
</div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#load-libraries-and-functions" id="toc-load-libraries-and-functions" class="nav-link active" data-scroll-target="#load-libraries-and-functions">load libraries and functions</a></li>
  <li><a href="#define-data-and-software-paths-for-r" id="toc-define-data-and-software-paths-for-r" class="nav-link" data-scroll-target="#define-data-and-software-paths-for-r">define data and software paths for R</a></li>
  <li><a href="#define-data-and-software-for-the-terminal" id="toc-define-data-and-software-for-the-terminal" class="nav-link" data-scroll-target="#define-data-and-software-for-the-terminal">define data and software for the terminal</a></li>
  <li><a href="#read-grm-matrix" id="toc-read-grm-matrix" class="nav-link" data-scroll-target="#read-grm-matrix">read grm matrix</a></li>
  <li><a href="#define-myplot" id="toc-define-myplot" class="nav-link" data-scroll-target="#define-myplot">define myplot</a>
  <ul class="collapse">
  <li><a href="#need-debug-simulate-y-sigma12epsilon-and-run-assoc-with-expr_mat" id="toc-need-debug-simulate-y-sigma12epsilon-and-run-assoc-with-expr_mat" class="nav-link" data-scroll-target="#need-debug-simulate-y-sigma12epsilon-and-run-assoc-with-expr_mat">NEED DEBUG simulate <span class="math inline">\(Y = Sigma^{1/2}\epsilon\)</span> and run assoc with expr_mat</a></li>
  </ul></li>
  <li><a href="#read-gene-annotation" id="toc-read-gene-annotation" class="nav-link" data-scroll-target="#read-gene-annotation">read gene annotation</a></li>
  <li><a href="#read-predicted-expression" id="toc-read-predicted-expression" class="nav-link" data-scroll-target="#read-predicted-expression">read predicted expression</a></li>
  <li><a href="#define-function-lmm-with-grm" id="toc-define-function-lmm-with-grm" class="nav-link" data-scroll-target="#define-function-lmm-with-grm">define function lmm with GRM</a></li>
  <li><a href="#run-ratxcan" id="toc-run-ratxcan" class="nav-link" data-scroll-target="#run-ratxcan">run ratXcan</a></li>
  <li><a href="#calc-grm-without-chr-1" id="toc-calc-grm-without-chr-1" class="nav-link" data-scroll-target="#calc-grm-without-chr-1">calc GRM without chr 1</a></li>
  <li><a href="#compute-grm-loco-chr01-and-lmmgrm-with-updated-h2" id="toc-compute-grm-loco-chr01-and-lmmgrm-with-updated-h2" class="nav-link" data-scroll-target="#compute-grm-loco-chr01-and-lmmgrm-with-updated-h2">compute GRM loco chr01 and lmmGRM with updated h2</a></li>
  <li><a href="#compare-pvalues-in-chr-1-vs-other-chromosomes" id="toc-compare-pvalues-in-chr-1-vs-other-chromosomes" class="nav-link" data-scroll-target="#compare-pvalues-in-chr-1-vs-other-chromosomes">compare pvalues in chr 1 vs other chromosomes</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">compare ratxcan with and without loco</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>
</div>

<div>
  <div class="description">
    compare ratxcan mixed effects with and without loco
  </div>
</div>


<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Haky Im </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">August 29, 2024</p>
    </div>
  </div>
  
    
  </div>
  

</header>

<section id="load-libraries-and-functions" class="level2">
<h2 class="anchored" data-anchor-id="load-libraries-and-functions">load libraries and functions</h2>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co">#options(error=recover)</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="co">#options(error=browser)</span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">error=</span><span class="cn">NULL</span>)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="do">## compare observed correlation with null correlation</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressMessages</span>(devtools<span class="sc">::</span><span class="fu">source_gist</span>(<span class="st">"a925fea01b365a8c605e"</span>)) <span class="do">## load qqR fn https://gist.github.com/hakyim/a925fea01b365a8c605e</span></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressMessages</span>(devtools<span class="sc">::</span><span class="fu">source_gist</span>(<span class="st">"38431b74c6c0bf90c12f"</span>)) <span class="do">## qqunif https://gist.github.com/hakyim/38431b74c6c0bf90c12f</span></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressMessages</span>(devtools<span class="sc">::</span><span class="fu">source_gist</span>(<span class="st">"115403f16bec0a0e871f3616d552ce9b"</span>)) <span class="do">## source ratxcan functions https://gist.github.com/hakyim/115403f16bec0a0e871f3616d552ce9b </span></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressMessages</span>(<span class="fu">library</span>(tidyverse))</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressMessages</span>(<span class="fu">library</span>(glue))</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressMessages</span>(<span class="fu">library</span>(RSQLite))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'RSQLite' was built under R version 4.3.3</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co">#suppressMessages(library(expm))</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="co">#suppressMessages(library(readxl))</span></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="co"># install.packages("devtools")</span></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a><span class="co"># library("devtools")</span></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a><span class="co"># install_github("jdstorey/qvalue")</span></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressMessages</span>(<span class="fu">library</span>(qvalue))</span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a><span class="co"># if (!require("BiocManager", quietly = TRUE))</span></span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a><span class="co">#     install.packages("BiocManager")</span></span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a><span class="co"># BiocManager::install("biomaRt")</span></span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a><span class="do">##suppressMessages(library(biomaRt))</span></span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a><span class="do">##install.packages("ggrepel")</span></span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressMessages</span>(<span class="fu">library</span>(ggrepel))</span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a>recalc<span class="ot">=</span><span class="cn">FALSE</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="define-data-and-software-paths-for-r" class="level2">
<h2 class="anchored" data-anchor-id="define-data-and-software-paths-for-r">define data and software paths for R</h2>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>WEBDATA<span class="ot">=</span><span class="st">"/Users/haekyungim/Library/CloudStorage/Box-Box/LargeFiles/imlab-data/data-Github/web-data"</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>PLINK<span class="ot">=</span><span class="st">"/Users/haekyungim/bin/plink_mac_20231211/plink"</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>GCTA<span class="ot">=</span><span class="st">"/Users/haekyungim/bin/gcta-1.94.2-MacOS-ARM-x86_64/gcta64"</span></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>INPUT <span class="ot">&lt;-</span> <span class="fu">glue</span>(<span class="st">"{WEBDATA}/ratxcan-tutorial"</span>) <span class="do">## this has the input data </span></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>OUTPUT <span class="ot">&lt;-</span> <span class="fu">glue</span>(<span class="st">"{WEBDATA}/2023-11-27-ratxcan-tutorial/scratch"</span>) <span class="do">## this has the output data, intermediate results</span></span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>GENO <span class="ot">=</span> <span class="fu">glue</span>(<span class="st">"{WEBDATA}/2023-11-27-ratxcan-tutorial/data/genotype/"</span>)</span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a>OUT_TEMPO <span class="ot">=</span> <span class="fu">glue</span>(<span class="st">"{OUTPUT}/testing-2024-08-29"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="define-data-and-software-for-the-terminal" class="level2">
<h2 class="anchored" data-anchor-id="define-data-and-software-for-the-terminal">define data and software for the terminal</h2>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="va">WEBDATA</span><span class="op">=</span><span class="st">"/Users/haekyungim/Library/CloudStorage/Box-Box/LargeFiles/imlab-data/data-Github/web-data"</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="va">PLINK</span><span class="op">=</span><span class="st">"/Users/haekyungim/bin/plink_mac_20231211/plink"</span></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a><span class="va">GCTA</span><span class="op">=</span><span class="st">"/Users/haekyungim/bin/gcta-1.94.2-MacOS-ARM-x86_64/gcta64"</span></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a><span class="va">INPUT</span><span class="op">=</span><span class="va">$WEBDATA</span>/ratxcan-tutorial</span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a><span class="va">OUTPUT</span><span class="op">=</span><span class="va">$WEBDATA</span>/2023-11-27-ratxcan-tutorial/scratch</span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a><span class="va">GENO</span><span class="op">=</span><span class="va">$WEBDATA</span>/2023-11-27-ratxcan-tutorial/data/genotype/</span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a><span class="va">OUT_TEMPO</span><span class="op">=</span><span class="va">$OUTPUT</span>/testing-2024-08-29</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="read-grm-matrix" class="level2">
<h2 class="anchored" data-anchor-id="read-grm-matrix">read grm matrix</h2>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>grm_mat <span class="ot">&lt;-</span> <span class="fu">read_GRMBin</span>(<span class="fu">glue</span>(<span class="st">"{OUTPUT}/rat6k_autosome.grm"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="define-myplot" class="level2">
<h2 class="anchored" data-anchor-id="define-myplot">define myplot</h2>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>myplot <span class="ot">&lt;-</span> <span class="cf">function</span>(tempres, <span class="at">post_titulo=</span><span class="st">""</span>,<span class="at">semilla=</span><span class="st">""</span>) {</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create a data frame with specific columns</span></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">p0.01_yes =</span> <span class="fu">apply</span>(tempres<span class="sc">$</span>pmat_correct, <span class="dv">2</span>, <span class="cf">function</span>(x) <span class="fu">mean</span>(x <span class="sc">&lt;</span> <span class="fl">0.01</span>)),</span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">p0.01_no =</span> <span class="fu">apply</span>(tempres<span class="sc">$</span>pmat_raw, <span class="dv">2</span>, <span class="cf">function</span>(x) <span class="fu">mean</span>(x <span class="sc">&lt;</span> <span class="fl">0.01</span>)),</span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">p0.05_yes =</span> <span class="fu">apply</span>(tempres<span class="sc">$</span>pmat_correct, <span class="dv">2</span>, <span class="cf">function</span>(x) <span class="fu">mean</span>(x <span class="sc">&lt;</span> <span class="fl">0.05</span>)),</span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">p0.05_no =</span> <span class="fu">apply</span>(tempres<span class="sc">$</span>pmat_raw, <span class="dv">2</span>, <span class="cf">function</span>(x) <span class="fu">mean</span>(x <span class="sc">&lt;</span> <span class="fl">0.05</span>)),</span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">p0.10_yes =</span> <span class="fu">apply</span>(tempres<span class="sc">$</span>pmat_correct, <span class="dv">2</span>, <span class="cf">function</span>(x) <span class="fu">mean</span>(x <span class="sc">&lt;</span> <span class="fl">0.10</span>)), </span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">p0.10_no =</span> <span class="fu">apply</span>(tempres<span class="sc">$</span>pmat_raw, <span class="dv">2</span>, <span class="cf">function</span>(x) <span class="fu">mean</span>(x <span class="sc">&lt;</span> <span class="fl">0.10</span>))</span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a>    <span class="co"># ... [rest of your code for creating df] ...</span></span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Pivot the data frame to long format, specifying the columns to keep</span></span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a>df_long <span class="ot">&lt;-</span> <span class="fu">pivot_longer</span>(df, <span class="at">cols =</span> <span class="fu">starts_with</span>(<span class="st">"p"</span>))</span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-16"><a href="#cb7-16" aria-hidden="true" tabindex="-1"></a>df_long <span class="ot">&lt;-</span> df_long <span class="sc">%&gt;%</span> <span class="fu">separate</span>(name,<span class="at">into =</span> <span class="fu">c</span>(<span class="st">"threshold"</span>,<span class="st">"corrected"</span>),<span class="at">sep=</span><span class="st">"_"</span>) <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">proportion=</span>value)</span>
<span id="cb7-17"><a href="#cb7-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-18"><a href="#cb7-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Rename the name column to replace p0.xx with p&lt;0.xx</span></span>
<span id="cb7-19"><a href="#cb7-19" aria-hidden="true" tabindex="-1"></a>df_long <span class="ot">&lt;-</span> df_long <span class="sc">%&gt;%</span></span>
<span id="cb7-20"><a href="#cb7-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">threshold =</span> <span class="fu">gsub</span>(<span class="st">"p0</span><span class="sc">\\</span><span class="st">."</span>, <span class="st">"p&lt;0."</span>, threshold))</span>
<span id="cb7-21"><a href="#cb7-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-22"><a href="#cb7-22" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create boxplots with mean</span></span>
<span id="cb7-23"><a href="#cb7-23" aria-hidden="true" tabindex="-1"></a>  pp <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(df_long, <span class="fu">aes</span>(<span class="at">x =</span> threshold, <span class="at">y =</span> proportion, <span class="at">fill =</span> corrected)) <span class="sc">+</span></span>
<span id="cb7-24"><a href="#cb7-24" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_boxplot</span>(<span class="at">alpha =</span> <span class="fl">0.6</span>) <span class="sc">+</span></span>
<span id="cb7-25"><a href="#cb7-25" aria-hidden="true" tabindex="-1"></a>    <span class="fu">stat_summary</span>(<span class="at">fun =</span> mean, <span class="at">geom =</span> <span class="st">"point"</span>, <span class="at">shape =</span> <span class="dv">3</span>, <span class="at">size =</span> <span class="dv">2</span>, <span class="at">stroke =</span> <span class="dv">2</span>, <span class="at">color =</span> <span class="st">"blue"</span>,                  <span class="co">#position = position_dodge(width = 0.8)) +</span></span>
<span id="cb7-26"><a href="#cb7-26" aria-hidden="true" tabindex="-1"></a>                 <span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> <span class="sc">-</span><span class="fl">0.1</span>)) <span class="sc">+</span></span>
<span id="cb7-27"><a href="#cb7-27" aria-hidden="true" tabindex="-1"></a>    <span class="co">#stat_summary(fun = mean, geom = "crossbar",  size = .5, color = "blue") +</span></span>
<span id="cb7-28"><a href="#cb7-28" aria-hidden="true" tabindex="-1"></a>    <span class="co">#stat_summary(fun = mean, geom = "crossbar",  size = .5, color = "darkgray") +</span></span>
<span id="cb7-29"><a href="#cb7-29" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="fu">c</span>(<span class="fl">0.01</span>, <span class="fl">0.05</span>, <span class="fl">0.10</span>), <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">color =</span> <span class="st">"gray"</span>) <span class="sc">+</span></span>
<span id="cb7-30"><a href="#cb7-30" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_minimal</span>(<span class="at">base_size =</span> <span class="dv">17</span>) <span class="sc">+</span></span>
<span id="cb7-31"><a href="#cb7-31" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="fu">glue</span>(<span class="st">"Type I Error Calibration {semilla} {post_titulo}"</span>)) <span class="sc">+</span></span>
<span id="cb7-32"><a href="#cb7-32" aria-hidden="true" tabindex="-1"></a>    <span class="fu">xlab</span>(<span class="st">"significance"</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"false positive rate"</span>)</span>
<span id="cb7-33"><a href="#cb7-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-34"><a href="#cb7-34" aria-hidden="true" tabindex="-1"></a>  pp</span>
<span id="cb7-35"><a href="#cb7-35" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="need-debug-simulate-y-sigma12epsilon-and-run-assoc-with-expr_mat" class="level3">
<h3 class="anchored" data-anchor-id="need-debug-simulate-y-sigma12epsilon-and-run-assoc-with-expr_mat">NEED DEBUG simulate <span class="math inline">\(Y = Sigma^{1/2}\epsilon\)</span> and run assoc with expr_mat</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>nsam<span class="ot">=</span><span class="fu">nrow</span>(grm_mat)</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a><span class="co">#ind=1:nsam</span></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>ind<span class="ot">=</span><span class="dv">1</span><span class="sc">:</span><span class="dv">1000</span></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>test_mat <span class="ot">=</span> grm_mat[ind,ind]</span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>nsam<span class="ot">=</span><span class="fu">nrow</span>(test_mat)</span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>Sigma <span class="ot">=</span> test_mat <span class="sc">*</span> h2 <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">-</span> h2) <span class="sc">*</span> <span class="fu">diag</span>(<span class="fu">rep</span>(<span class="dv">1</span>,nsam))</span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>Sig_eigen <span class="ot">=</span> <span class="fu">eigen</span>(Sigma)</span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(Sig_eigen<span class="sc">$</span>vectors) <span class="ot">=</span> <span class="fu">rownames</span>(Sigma)</span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a><span class="do">##sighalf = Sig_eigen$vectors %*% diag( sqrt(  Sig_eigen$values  ) ) %*% t(Sig_eigen$vectors)</span></span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a><span class="do">## make this multiplication more efficient using sweep</span></span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a>sighalf <span class="ot">=</span> Sig_eigen<span class="sc">$</span>vectors <span class="sc">%*%</span> <span class="fu">sweep</span>(<span class="fu">t</span>(Sig_eigen<span class="sc">$</span>vectors),<span class="dv">1</span>,<span class="fu">sqrt</span>(  Sig_eigen<span class="sc">$</span>values ),<span class="st">"*"</span>)</span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true" tabindex="-1"></a>sim_sigma_pheno <span class="ot">=</span> sighalf <span class="sc">%*%</span> <span class="fu">matrix</span>(<span class="fu">rnorm</span>(nsam <span class="sc">*</span> nsim), nsam, nsim) </span>
<span id="cb8-14"><a href="#cb8-14" aria-hidden="true" tabindex="-1"></a>sim_sigma_pheno<span class="ot">=</span><span class="fu">cbind</span>(<span class="at">FID=</span><span class="fu">rownames</span>(sim_sigma_pheno),<span class="at">IID=</span><span class="fu">rownames</span>(sim_sigma_pheno),<span class="fu">as.data.frame</span>(sim_sigma_pheno))</span>
<span id="cb8-15"><a href="#cb8-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-16"><a href="#cb8-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-17"><a href="#cb8-17" aria-hidden="true" tabindex="-1"></a>tic<span class="ot">=</span><span class="fu">Sys.time</span>()</span>
<span id="cb8-18"><a href="#cb8-18" aria-hidden="true" tabindex="-1"></a>tempres_sigma_pheno <span class="ot">&lt;-</span> <span class="fu">lmmGRM</span>(sim_sigma_pheno,grm_mat, h2,pred_expr,<span class="at">pheno_id_col=</span><span class="dv">1</span>, <span class="at">pheno_value_cols=</span><span class="dv">2</span><span class="sc">+</span>(<span class="dv">1</span><span class="sc">:</span>nsim))</span>
<span id="cb8-19"><a href="#cb8-19" aria-hidden="true" tabindex="-1"></a>toc<span class="ot">=</span><span class="fu">Sys.time</span>()</span>
<span id="cb8-20"><a href="#cb8-20" aria-hidden="true" tabindex="-1"></a>toc <span class="sc">-</span> tic</span>
<span id="cb8-21"><a href="#cb8-21" aria-hidden="true" tabindex="-1"></a>pp<span class="ot">&lt;-</span><span class="fu">myplot</span>(tempres_sigma_pheno,<span class="at">post_titulo =</span> <span class="fu">glue</span>(<span class="st">"sigma_pheno n= {nsam} - ii={ii}"</span>))</span>
<span id="cb8-22"><a href="#cb8-22" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(ii,<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb8-23"><a href="#cb8-23" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(pp)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</section>
<section id="read-gene-annotation" class="level2">
<h2 class="anchored" data-anchor-id="read-gene-annotation">read gene annotation</h2>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="co">#gene_annotation &lt;- readRDS(glue("{INPUT}/data/expression/gene_annotation.RDS"))</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>gene_annotation <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="fu">glue</span>(<span class="st">"{WEBDATA}/2023-11-27-ratxcan-tutorial/data/expression/gene_annotation.RDS"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="read-predicted-expression" class="level2">
<h2 class="anchored" data-anchor-id="read-predicted-expression">read predicted expression</h2>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>read_pred_expr <span class="ot">=</span> <span class="cf">function</span>(filename)</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>  <span class="do">##usage: Br_pred_expr = read_pred_expr(glue("{OUTPUT}/Br-hki-rat6k__predict.txt"))</span></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>  pred_expr <span class="ot">&lt;-</span> vroom<span class="sc">::</span><span class="fu">vroom</span>(filename) <span class="sc">%&gt;%</span> </span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>FID) <span class="sc">%&gt;%</span>  <span class="co"># Remove the FID column</span></span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">IID =</span> <span class="fu">str_split</span>(IID, <span class="st">"_"</span>, <span class="at">simplify =</span> <span class="cn">TRUE</span>)[, <span class="dv">1</span>])  <span class="co"># Keep the first part of IID</span></span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a>  pred_expr</span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a><span class="co"># pred_expr = read_pred_expr(glue("{OUTPUT}/Br-hki-rat6k__predict.txt"))</span></span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a><span class="co"># ## WHYYYY br-hki-rat6k__predict.txt has no chr 1????</span></span>
<span id="cb10-12"><a href="#cb10-12" aria-hidden="true" tabindex="-1"></a><span class="co"># df1= tibble(gene=names(pred_expr)) %&gt;% mutate(isinac=TRUE) %&gt;% left_join(gene_annotation,by=c("gene"="gene_id")) </span></span>
<span id="cb10-13"><a href="#cb10-13" aria-hidden="true" tabindex="-1"></a><span class="co"># df1 %&gt;% count(chr)</span></span>
<span id="cb10-14"><a href="#cb10-14" aria-hidden="true" tabindex="-1"></a><span class="co"># cat("{OUTPUT}/Br_rat6k__predict.txt has no chr 1 genes!!! Where did I lose them?\n")</span></span>
<span id="cb10-15"><a href="#cb10-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-16"><a href="#cb10-16" aria-hidden="true" tabindex="-1"></a>pred_expr <span class="ot">=</span> <span class="fu">read_pred_expr</span>(<span class="fu">glue</span>(<span class="st">"{OUTPUT}/AC-filtered__predict.txt"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 5628 Columns: 5881
── Column specification ────────────────────────────────────────────────────────
Delimiter: "\t"
chr    (2): FID, IID
dbl (5879): ENSRNOG00000015552, ENSRNOG00000016054, ENSRNOG00000049505, ENSR...

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="do">## checked that I have chr1 genes</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>df1<span class="ot">=</span> <span class="fu">tibble</span>(<span class="at">gene=</span><span class="fu">names</span>(pred_expr)[<span class="sc">-</span><span class="dv">1</span>]) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">isinac=</span><span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(gene_annotation,<span class="at">by=</span><span class="fu">c</span>(<span class="st">"gene"</span><span class="ot">=</span><span class="st">"gene_id"</span>)) </span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>df1 <span class="sc">%&gt;%</span> <span class="fu">count</span>(chr)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 20 × 2
   chr       n
   &lt;chr&gt; &lt;int&gt;
 1 1       748
 2 10      542
 3 11      166
 4 12      186
 5 13      190
 6 14      228
 7 15      177
 8 16      177
 9 17      127
10 18      171
11 19      187
12 2       396
13 20      203
14 3       452
15 4       344
16 5       387
17 6       267
18 7       394
19 8       312
20 9       225</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"{OUTPUT}/AC-filtered__predict.txt has genes in "</span>,<span class="fu">length</span>(<span class="fu">unique</span>(df1<span class="sc">$</span>chr)), <span class="st">"chr"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>{OUTPUT}/AC-filtered__predict.txt has genes in  20 chr</code></pre>
</div>
</div>
</section>
<section id="define-function-lmm-with-grm" class="level2">
<h2 class="anchored" data-anchor-id="define-function-lmm-with-grm">define function lmm with GRM</h2>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="do">## HERE WE USE THE FULL GRM MATRIX AND CALCULATE THE INVERSE OF THE SIGMA MATRIX</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a><span class="do">## define lmm association function </span></span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>lmmGRM <span class="ot">=</span> <span class="cf">function</span>(pheno, grm_mat, h2, pred_expr,<span class="at">pheno_id_col=</span><span class="dv">1</span>,<span class="at">pheno_value_cols=</span><span class="dv">6</span><span class="sc">:</span><span class="dv">6</span>,<span class="at">out=</span><span class="cn">NULL</span>)</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>  <span class="do">## input pheno is a data frame with id column pheno_id_col=1 by default</span></span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>  <span class="do">## phenotype values are in pheno_value_cols, 6:6 by default (SCORE column location in plink output), it can have more than one phenotype</span></span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a>  <span class="do">## but h2 has to be the same, this is useful when running simulations with different h2</span></span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a>  <span class="do">## call lmmXcan(pheno %&gt;% select(IID,SCORE))</span></span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a>  <span class="do">## format pheno to matrix form</span></span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true" tabindex="-1"></a>  phenomat <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(pheno[,pheno_value_cols])</span>
<span id="cb16-12"><a href="#cb16-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(phenomat) <span class="ot">&lt;-</span> pheno[[pheno_id_col]]</span>
<span id="cb16-13"><a href="#cb16-13" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb16-14"><a href="#cb16-14" aria-hidden="true" tabindex="-1"></a>  <span class="do">## turn pred_expr into matrix with rownames =IID, keep only IIDs in ymat</span></span>
<span id="cb16-15"><a href="#cb16-15" aria-hidden="true" tabindex="-1"></a>  exp_mat <span class="ot">=</span> <span class="fu">as.matrix</span>(pred_expr <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span>IID))</span>
<span id="cb16-16"><a href="#cb16-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(exp_mat) <span class="ot">=</span> pred_expr<span class="sc">$</span>IID</span>
<span id="cb16-17"><a href="#cb16-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-18"><a href="#cb16-18" aria-hidden="true" tabindex="-1"></a>  <span class="do">## align pheno and expr matrices</span></span>
<span id="cb16-19"><a href="#cb16-19" aria-hidden="true" tabindex="-1"></a>  idlist <span class="ot">=</span> <span class="fu">intersect</span>(<span class="fu">rownames</span>(phenomat), <span class="fu">rownames</span>(exp_mat))</span>
<span id="cb16-20"><a href="#cb16-20" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb16-21"><a href="#cb16-21" aria-hidden="true" tabindex="-1"></a>  nsam <span class="ot">=</span> <span class="fu">length</span>(idlist)</span>
<span id="cb16-22"><a href="#cb16-22" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb16-23"><a href="#cb16-23" aria-hidden="true" tabindex="-1"></a>  <span class="do">## CALCULATE SIGMA</span></span>
<span id="cb16-24"><a href="#cb16-24" aria-hidden="true" tabindex="-1"></a>  ID_mat <span class="ot">=</span> <span class="fu">diag</span>(<span class="fu">rep</span>(<span class="dv">1</span>,nsam))</span>
<span id="cb16-25"><a href="#cb16-25" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb16-26"><a href="#cb16-26" aria-hidden="true" tabindex="-1"></a>  <span class="co">#testing_scale_grm = TRUE</span></span>
<span id="cb16-27"><a href="#cb16-27" aria-hidden="true" tabindex="-1"></a>  <span class="co">#if(testing_scale_grm) grm_mat = sweep( sweep(grm_mat,2, 1/sqrt(diag(grm_mat)),"*"), 1, 1/sqrt(diag(grm_mat)),"*")    </span></span>
<span id="cb16-28"><a href="#cb16-28" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb16-29"><a href="#cb16-29" aria-hidden="true" tabindex="-1"></a>  Sigma <span class="ot">=</span> grm_mat[idlist,idlist] <span class="sc">*</span> h2 <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">-</span> h2) <span class="sc">*</span> ID_mat</span>
<span id="cb16-30"><a href="#cb16-30" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb16-31"><a href="#cb16-31" aria-hidden="true" tabindex="-1"></a>  Sig_eigen <span class="ot">=</span> <span class="fu">eigen</span>(Sigma)</span>
<span id="cb16-32"><a href="#cb16-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(Sig_eigen<span class="sc">$</span>vectors) <span class="ot">=</span> <span class="fu">rownames</span>(Sigma)</span>
<span id="cb16-33"><a href="#cb16-33" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb16-34"><a href="#cb16-34" aria-hidden="true" tabindex="-1"></a>  isighalf <span class="ot">=</span> Sig_eigen<span class="sc">$</span>vectors <span class="sc">%*%</span> <span class="fu">diag</span>( <span class="dv">1</span> <span class="sc">/</span> <span class="fu">sqrt</span>(  Sig_eigen<span class="sc">$</span>values  ) ) <span class="sc">%*%</span> <span class="fu">t</span>(Sig_eigen<span class="sc">$</span>vectors)</span>
<span id="cb16-35"><a href="#cb16-35" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb16-36"><a href="#cb16-36" aria-hidden="true" tabindex="-1"></a>  <span class="do">## perform raw association</span></span>
<span id="cb16-37"><a href="#cb16-37" aria-hidden="true" tabindex="-1"></a>  cormat_raw <span class="ot">=</span> <span class="fu">matrix_lm</span>(phenomat[idlist,, <span class="at">drop =</span> <span class="cn">FALSE</span>], exp_mat[idlist,])</span>
<span id="cb16-38"><a href="#cb16-38" aria-hidden="true" tabindex="-1"></a>  pmat_raw <span class="ot">=</span> <span class="fu">cor2pval</span>(cormat_raw,nsam)</span>
<span id="cb16-39"><a href="#cb16-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(pmat_raw) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"cor_"</span>, <span class="st">"pval_"</span>, <span class="fu">colnames</span>(pmat_raw))</span>
<span id="cb16-40"><a href="#cb16-40" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb16-41"><a href="#cb16-41" aria-hidden="true" tabindex="-1"></a>  <span class="do">## perform corrected association</span></span>
<span id="cb16-42"><a href="#cb16-42" aria-hidden="true" tabindex="-1"></a>  cormat_correct <span class="ot">=</span> <span class="fu">matrix_lm</span>(isighalf<span class="sc">%*%</span> phenomat[idlist,, <span class="at">drop =</span> <span class="cn">FALSE</span>], isighalf <span class="sc">%*%</span> exp_mat[idlist,])</span>
<span id="cb16-43"><a href="#cb16-43" aria-hidden="true" tabindex="-1"></a>  pmat_correct <span class="ot">=</span> <span class="fu">cor2pval</span>(cormat_correct,nsam)</span>
<span id="cb16-44"><a href="#cb16-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(pmat_correct) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"cor_"</span>, <span class="st">"pval_"</span>, <span class="fu">colnames</span>(pmat_correct))</span>
<span id="cb16-45"><a href="#cb16-45" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb16-46"><a href="#cb16-46" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.null</span>(out))</span>
<span id="cb16-47"><a href="#cb16-47" aria-hidden="true" tabindex="-1"></a>  {</span>
<span id="cb16-48"><a href="#cb16-48" aria-hidden="true" tabindex="-1"></a>    <span class="fu">saveRDS</span>(cormat_correct,<span class="at">file =</span> <span class="fu">glue</span>(<span class="st">"{out}_cormat_correct.RDS"</span>))</span>
<span id="cb16-49"><a href="#cb16-49" aria-hidden="true" tabindex="-1"></a>    <span class="fu">saveRDS</span>(pmat_correct,  <span class="at">file =</span> <span class="fu">glue</span>(<span class="st">"{out}_pmat_correct.RDS"</span>))</span>
<span id="cb16-50"><a href="#cb16-50" aria-hidden="true" tabindex="-1"></a>    <span class="fu">saveRDS</span>(cormat_raw,    <span class="at">file =</span> <span class="fu">glue</span>(<span class="st">"{out}_cormat_raw.RDS"</span>))</span>
<span id="cb16-51"><a href="#cb16-51" aria-hidden="true" tabindex="-1"></a>    <span class="fu">saveRDS</span>(pmat_raw,      <span class="at">file =</span> <span class="fu">glue</span>(<span class="st">"{out}_pmat_raw.RDS"</span>))</span>
<span id="cb16-52"><a href="#cb16-52" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb16-53"><a href="#cb16-53" aria-hidden="true" tabindex="-1"></a>  res <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb16-54"><a href="#cb16-54" aria-hidden="true" tabindex="-1"></a>    <span class="at">cormat_correct=</span>cormat_correct, </span>
<span id="cb16-55"><a href="#cb16-55" aria-hidden="true" tabindex="-1"></a>    <span class="at">pmat_correct=</span>pmat_correct, </span>
<span id="cb16-56"><a href="#cb16-56" aria-hidden="true" tabindex="-1"></a>    <span class="at">cormat_raw=</span>cormat_raw, </span>
<span id="cb16-57"><a href="#cb16-57" aria-hidden="true" tabindex="-1"></a>    <span class="at">pmat_raw=</span>pmat_raw)</span>
<span id="cb16-58"><a href="#cb16-58" aria-hidden="true" tabindex="-1"></a>  res</span>
<span id="cb16-59"><a href="#cb16-59" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb16-60"><a href="#cb16-60" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="run-ratxcan" class="level2">
<h2 class="anchored" data-anchor-id="run-ratxcan">run ratXcan</h2>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="do">## read simulated Y</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>pheno <span class="ot">=</span> <span class="fu">read_table</span>(<span class="fu">glue</span>(<span class="st">"{WEBDATA}/2023-11-27-ratxcan-tutorial/scratch/sim/tempo/PRS_output_100-32240.profile"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>
── Column specification ────────────────────────────────────────────────────────
cols(
  FID = col_character(),
  IID = col_character(),
  PHENO = col_double(),
  CNT = col_double(),
  CNT2 = col_double(),
  SCORE = col_double()
)</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="do">## calculate h2 of pheno</span></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a><span class="do">## mpheno 4 means fourth column after FID and IID</span></span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a><span class="do">## $GCTA --grm $OUTPUT/rat6k_autosome --reml --pheno $OUTPUT/sim/tempo/PRS_output_100-32240.profile --mpheno 4 --out $OUTPUT/test-2024-08-29</span></span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a><span class="do">## create phenotype with h2</span></span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a>h2<span class="ot">=</span><span class="fl">0.5</span></span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a>pheno<span class="sc">$</span>trait <span class="ot">=</span> (pheno<span class="sc">$</span>SCORE <span class="sc">-</span> <span class="fu">mean</span>(pheno<span class="sc">$</span>SCORE))<span class="sc">/</span><span class="fu">sd</span>(pheno<span class="sc">$</span>SCORE) <span class="sc">*</span> <span class="fu">sqrt</span>(h2) <span class="sc">+</span> <span class="fu">sqrt</span>(<span class="dv">1</span> <span class="sc">-</span> h2)<span class="sc">*</span><span class="fu">rnorm</span>(<span class="fu">nrow</span>(pheno))</span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(recalc)</span>
<span id="cb19-10"><a href="#cb19-10" aria-hidden="true" tabindex="-1"></a>{res <span class="ot">=</span> <span class="fu">lmmGRM</span>(pheno, grm_mat, h2, pred_expr,<span class="at">pheno_id_col=</span><span class="dv">1</span>,<span class="at">pheno_value_cols=</span><span class="dv">7</span><span class="sc">:</span><span class="dv">7</span>,<span class="at">out=</span><span class="cn">NULL</span>)</span>
<span id="cb19-11"><a href="#cb19-11" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(res,<span class="fu">glue</span>(<span class="st">"{OUT_TEMPO}/ratxcan-ac-res.RDS"</span>))</span>
<span id="cb19-12"><a href="#cb19-12" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span></span>
<span id="cb19-13"><a href="#cb19-13" aria-hidden="true" tabindex="-1"></a>  res <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="fu">glue</span>(<span class="st">"{OUT_TEMPO}/ratxcan-ac-res.RDS"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="calc-grm-without-chr-1" class="level2">
<h2 class="anchored" data-anchor-id="calc-grm-without-chr-1">calc GRM without chr 1</h2>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="va">$PLINK</span> <span class="at">--bfile</span> <span class="va">$GENO</span>/rat6k_autosome <span class="at">--chr</span> 1 <span class="at">--write-snplist</span> <span class="at">--out</span> <span class="va">$OUT_TEMPO</span>/chr1_snps</span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a><span class="va">$GCTA</span> <span class="at">--bfile</span> <span class="va">$GENO</span>/rat6k_autosome <span class="at">--exclude</span> <span class="va">$OUT_TEMPO</span>/chr1_snps.snplist <span class="at">--make-grm-bin</span> <span class="at">--out</span> <span class="va">$OUT_TEMPO</span>/rat6k_autosome_loco_chr01 <span class="at">--thread-num</span> 8</span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a><span class="co">## that took </span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<pre class="text cell-code"><code>*******************************************************************
* Genome-wide Complex Trait Analysis (GCTA)
* version v1.94.1 Mac
* (C) 2010-present, Yang Lab, Westlake University
* Please report bugs to Jian Yang &lt;jian.yang@westlake.edu.cn&gt;
*******************************************************************
Analysis started at 16:52:27 CDT on Thu Aug 29 2024.
Hostname: MED-ML-464.local

Accepted options:
--bfile /Users/haekyungim/Library/CloudStorage/Box-Box/LargeFiles/imlab-data/data-Github/web-data/2023-11-27-ratxcan-tutorial/data/genotype//rat6k_autosome
--exclude /Users/haekyungim/Library/CloudStorage/Box-Box/LargeFiles/imlab-data/data-Github/web-data/2023-11-27-ratxcan-tutorial/scratch/testing-2024-08-29/chr1_snps.snplist
--make-grm-bin
--out /Users/haekyungim/Library/CloudStorage/Box-Box/LargeFiles/imlab-data/data-Github/web-data/2023-11-27-ratxcan-tutorial/scratch/rat6k_autosome_loco_chr1
--thread-num 8

Note: the program will be running on 8 threads.

Reading PLINK FAM file from [/Users/haekyungim/Library/CloudStorage/Box-Box/LargeFiles/imlab-data/data-Github/web-data/2023-11-27-ratxcan-tutorial/data/genotype//rat6k_autosome.fam].
5628 individuals to be included from [/Users/haekyungim/Library/CloudStorage/Box-Box/LargeFiles/imlab-data/data-Github/web-data/2023-11-27-ratxcan-tutorial/data/genotype//rat6k_autosome.fam].
Reading PLINK BIM file from [/Users/haekyungim/Library/CloudStorage/Box-Box/LargeFiles/imlab-data/data-Github/web-data/2023-11-27-ratxcan-tutorial/data/genotype//rat6k_autosome.bim].
179895 SNPs to be included from [/Users/haekyungim/Library/CloudStorage/Box-Box/LargeFiles/imlab-data/data-Github/web-data/2023-11-27-ratxcan-tutorial/data/genotype//rat6k_autosome.bim].
Reading a list of SNPs from [/Users/haekyungim/Library/CloudStorage/Box-Box/LargeFiles/imlab-data/data-Github/web-data/2023-11-27-ratxcan-tutorial/scratch/testing-2024-08-29/chr1_snps.snplist].
17602 SNPs are excluded from [/Users/haekyungim/Library/CloudStorage/Box-Box/LargeFiles/imlab-data/data-Github/web-data/2023-11-27-ratxcan-tutorial/scratch/testing-2024-08-29/chr1_snps.snplist] and there are 162293 SNPs remaining.
Reading PLINK BED file from [/Users/haekyungim/Library/CloudStorage/Box-Box/LargeFiles/imlab-data/data-Github/web-data/2023-11-27-ratxcan-tutorial/data/genotype//rat6k_autosome.bed] in SNP-major format ...
Genotype data for 5628 individuals and 162293 SNPs to be included from [/Users/haekyungim/Library/CloudStorage/Box-Box/LargeFiles/imlab-data/data-Github/web-data/2023-11-27-ratxcan-tutorial/data/genotype//rat6k_autosome.bed].
Calculating allele frequencies ...
Recoding genotypes (individual major mode) ...

Calculating the genetic relationship matrix (GRM) ... (Note: default speed-optimized mode, may use huge RAM)

Summary of the GRM:
Mean of diagonals = 0.990807
Variance of diagonals = 0.00404564
Mean of off-diagonals = -0.000176097
Variance of off-diagonals = 0.00216189
GRM of 5628 individuals has been saved in the file [/Users/haekyungim/Library/CloudStorage/Box-Box/LargeFiles/imlab-data/data-Github/web-data/2023-11-27-ratxcan-tutorial/scratch/rat6k_autosome_loco_chr1.grm.bin] (in binary format).
Number of SNPs to calculate the genetic relationship between each pair of individuals has been saved in the file [/Users/haekyungim/Library/CloudStorage/Box-Box/LargeFiles/imlab-data/data-Github/web-data/2023-11-27-ratxcan-tutorial/scratch/rat6k_autosome_loco_chr1.grm.N.bin] (in binary format).
IDs for the GRM file [/Users/haekyungim/Library/CloudStorage/Box-Box/LargeFiles/imlab-data/data-Github/web-data/2023-11-27-ratxcan-tutorial/scratch/rat6k_autosome_loco_chr1.grm.bin] have been saved in the file [/Users/haekyungim/Library/CloudStorage/Box-Box/LargeFiles/imlab-data/data-Github/web-data/2023-11-27-ratxcan-tutorial/scratch/rat6k_autosome_loco_chr1.grm.id].
</code></pre>
</details>
</div>
</section>
<section id="compute-grm-loco-chr01-and-lmmgrm-with-updated-h2" class="level2">
<h2 class="anchored" data-anchor-id="compute-grm-loco-chr01-and-lmmgrm-with-updated-h2">compute GRM loco chr01 and lmmGRM with updated h2</h2>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>grm_mat_no_chr01 <span class="ot">=</span> <span class="fu">read_GRMBin</span>(<span class="fu">glue</span>(<span class="st">"{OUT_TEMPO}/rat6k_autosome_loco_chr01.grm"</span>))</span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a><span class="do">## need to recalculate h2</span></span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a><span class="co"># $GCTA --reml --grm $OUT_TEMPO/rat6k_autosome  --out $OUT_TEMPO/rat6k_autosome_loco_chr01 --thread-num 8</span></span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a><span class="do">## save the phenotype with error added</span></span>
<span id="cb22-8"><a href="#cb22-8" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(pheno <span class="sc">%&gt;%</span> <span class="fu">select</span>(IID, FID, trait),<span class="at">file=</span><span class="fu">glue</span>(<span class="st">"{OUT_TEMPO}/pheno_PRS_output_100-32240.txt"</span>))</span>
<span id="cb22-9"><a href="#cb22-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-10"><a href="#cb22-10" aria-hidden="true" tabindex="-1"></a><span class="co"># $GCTA --grm $OUT_TEMPO/rat6k_autosome_loco_chr01 --reml --pheno $OUT_TEMPO/pheno_PRS_output_100-32240.txt --mpheno 1 --out $OUT_TEMPO/loco_chr01 --thread-num 8</span></span>
<span id="cb22-11"><a href="#cb22-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-12"><a href="#cb22-12" aria-hidden="true" tabindex="-1"></a>h2_loco_chr01 <span class="ot">=</span> <span class="fu">read_tsv</span>(<span class="fu">glue</span>(<span class="st">"{OUT_TEMPO}/loco_chr01.hsq"</span>)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Source<span class="sc">==</span><span class="st">"V(G)/Vp"</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(Variance)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: One or more parsing issues, call `problems()` on your data frame for details,
e.g.:
  dat &lt;- vroom(...)
  problems(dat)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 10 Columns: 3
── Column specification ────────────────────────────────────────────────────────
Delimiter: "\t"
chr (1): Source
dbl (2): Variance, SE

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(recalc)</span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>  res_no_chr01 <span class="ot">=</span> <span class="fu">lmmGRM</span>(pheno, grm_mat_no_chr01, <span class="at">h2=</span> h2_loco_chr01, pred_expr,<span class="at">pheno_id_col=</span><span class="dv">1</span>,<span class="at">pheno_value_cols=</span><span class="dv">7</span><span class="sc">:</span><span class="dv">7</span>,<span class="at">out=</span><span class="cn">NULL</span>)</span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">saveRDS</span>(res_no_chr01,<span class="at">file=</span><span class="fu">glue</span>(<span class="st">"{OUT_TEMPO}/ratxcan-ac-res-loco-chr01.RDS"</span>))</span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> res_no_chr01<span class="ot">=</span><span class="fu">readRDS</span>(<span class="fu">glue</span>(<span class="st">"{OUT_TEMPO}/ratxcan-ac-res-loco-chr01.RDS"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="compare-pvalues-in-chr-1-vs-other-chromosomes" class="level2">
<h2 class="anchored" data-anchor-id="compare-pvalues-in-chr-1-vs-other-chromosomes">compare pvalues in chr 1 vs other chromosomes</h2>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a>pmat2df <span class="ot">=</span> <span class="cf">function</span>(pmat)</span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">=</span> <span class="fu">data.frame</span>(pmat)</span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a>  df<span class="sc">$</span>gene <span class="ot">=</span> <span class="fu">rownames</span>(pmat)</span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(df) <span class="ot">=</span> <span class="cn">NULL</span></span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a>  df <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(gene_annotation <span class="sc">%&gt;%</span> <span class="fu">select</span>(gene_id,chr,gene_name),<span class="at">by=</span><span class="fu">c</span>(<span class="st">"gene"</span><span class="ot">=</span><span class="st">"gene_id"</span>) )</span>
<span id="cb26-7"><a href="#cb26-7" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb26-8"><a href="#cb26-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-9"><a href="#cb26-9" aria-hidden="true" tabindex="-1"></a>df_loco_chr01 <span class="ot">=</span> <span class="fu">pmat2df</span>(res_no_chr01<span class="sc">$</span>pmat_correct) </span>
<span id="cb26-10"><a href="#cb26-10" aria-hidden="true" tabindex="-1"></a>df <span class="ot">=</span> <span class="fu">pmat2df</span>(res<span class="sc">$</span>pmat_correct) </span>
<span id="cb26-11"><a href="#cb26-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-12"><a href="#cb26-12" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow=</span><span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">2</span>))</span>
<span id="cb26-13"><a href="#cb26-13" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(df_loco_chr01 <span class="sc">%&gt;%</span> <span class="fu">filter</span>(chr<span class="sc">==</span><span class="st">"1"</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(trait),<span class="at">main=</span><span class="st">"ratxcan GRM no chr1 - genes in chr1"</span>)</span>
<span id="cb26-14"><a href="#cb26-14" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(df_loco_chr01 <span class="sc">%&gt;%</span> <span class="fu">filter</span>(chr<span class="sc">!=</span><span class="st">"1"</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(trait),<span class="at">main=</span><span class="st">"ratxcan GRM no chr1 - genes chr!=1"</span>)</span>
<span id="cb26-15"><a href="#cb26-15" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(df <span class="sc">%&gt;%</span> <span class="fu">filter</span>(chr<span class="sc">==</span><span class="st">"1"</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(trait),<span class="at">main=</span><span class="st">"ratxcan GRM all - genes in chr1"</span>)</span>
<span id="cb26-16"><a href="#cb26-16" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(df <span class="sc">%&gt;%</span> <span class="fu">filter</span>(chr<span class="sc">!=</span><span class="st">"1"</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(trait),<span class="at">main=</span><span class="st">"ratxcan GRM all - genes chr!=1"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="index_files/figure-html/compare pvalues in chr 1 vs other chromosomes-1.png" class="img-fluid" width="672"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">1</span>))</span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow=</span><span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">2</span>))</span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a>rango <span class="ot">=</span> <span class="fu">range</span>(<span class="sc">-</span><span class="fu">log10</span>(<span class="fu">c</span>(df_loco_chr01<span class="sc">$</span>trait, df<span class="sc">$</span>trait)))</span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a><span class="fu">qqunif</span>(df_loco_chr01 <span class="sc">%&gt;%</span> <span class="fu">filter</span>(chr<span class="sc">==</span><span class="st">"1"</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(trait),<span class="at">main=</span><span class="st">"ratxcan GRM no chr1 - genes in chr1"</span>,<span class="at">BH=</span><span class="cn">FALSE</span>,<span class="at">CI=</span><span class="cn">FALSE</span>,<span class="at">xlim=</span>rango,<span class="at">ylim=</span>rango)</span>
<span id="cb27-6"><a href="#cb27-6" aria-hidden="true" tabindex="-1"></a><span class="fu">qqunif</span>(df_loco_chr01 <span class="sc">%&gt;%</span> <span class="fu">filter</span>(chr<span class="sc">!=</span><span class="st">"1"</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(trait),<span class="at">main=</span><span class="st">"ratxcan GRM no chr1 - genes chr!=1"</span>,<span class="at">BH=</span><span class="cn">FALSE</span>,<span class="at">CI=</span><span class="cn">FALSE</span>,<span class="at">xlim=</span>rango,<span class="at">ylim=</span>rango)</span>
<span id="cb27-7"><a href="#cb27-7" aria-hidden="true" tabindex="-1"></a><span class="fu">qqunif</span>(df <span class="sc">%&gt;%</span> <span class="fu">filter</span>(chr<span class="sc">==</span><span class="st">"1"</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(trait),<span class="at">main=</span><span class="st">"ratxcan genes in chr1"</span>,<span class="at">BH=</span><span class="cn">FALSE</span>,<span class="at">CI=</span><span class="cn">FALSE</span>,<span class="at">xlim=</span>rango,<span class="at">ylim=</span>rango)</span>
<span id="cb27-8"><a href="#cb27-8" aria-hidden="true" tabindex="-1"></a><span class="fu">qqunif</span>(df <span class="sc">%&gt;%</span> <span class="fu">filter</span>(chr<span class="sc">!=</span><span class="st">"1"</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(trait),<span class="at">main=</span><span class="st">"ratxcan genes chr!=1"</span>,<span class="at">BH=</span><span class="cn">FALSE</span>,<span class="at">CI=</span><span class="cn">FALSE</span>,<span class="at">xlim=</span>rango,<span class="at">ylim=</span>rango)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="index_files/figure-html/compare pvalues in chr 1 vs other chromosomes-2.png" class="img-fluid" width="672"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">1</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="callout callout-style-default callout-warning callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Take home
</div>
</div>
<div class="callout-body-container callout-body">
<p>removing chr1 from GRM negates the correction of ratxcan in chr1</p>
</div>
</div>


<!-- -->

</section>

<div id="quarto-appendix" class="default"><section class="quarto-appendix-contents"><h2 class="anchored quarto-appendix-heading">Reuse</h2><div id="quarto-reuse" class="quarto-appendix-contents"><div>© HakyImLab and Listed Authors - CC BY 4.0 for Text and figures - MIT for code</div></div></section></div></main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb29" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> compare ratxcan with and without loco</span></span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a><span class="an">date:</span><span class="co"> 2024-08-29</span></span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a><span class="an">author:</span><span class="co"> Haky Im</span></span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a><span class="an">editor_options:</span><span class="co"> </span></span>
<span id="cb29-6"><a href="#cb29-6" aria-hidden="true" tabindex="-1"></a><span class="co">  chunk_output_type: console</span></span>
<span id="cb29-7"><a href="#cb29-7" aria-hidden="true" tabindex="-1"></a><span class="an">description:</span><span class="co"> compare ratxcan mixed effects with and without loco</span></span>
<span id="cb29-8"><a href="#cb29-8" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb29-9"><a href="#cb29-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-10"><a href="#cb29-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-11"><a href="#cb29-11" aria-hidden="true" tabindex="-1"></a><span class="fu">## load libraries and functions</span></span>
<span id="cb29-12"><a href="#cb29-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-13"><a href="#cb29-13" aria-hidden="true" tabindex="-1"></a><span class="in">```{r load libraries and functions}</span></span>
<span id="cb29-14"><a href="#cb29-14" aria-hidden="true" tabindex="-1"></a><span class="co">#options(error=recover)</span></span>
<span id="cb29-15"><a href="#cb29-15" aria-hidden="true" tabindex="-1"></a><span class="co">#options(error=browser)</span></span>
<span id="cb29-16"><a href="#cb29-16" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">error=</span><span class="cn">NULL</span>)</span>
<span id="cb29-17"><a href="#cb29-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-18"><a href="#cb29-18" aria-hidden="true" tabindex="-1"></a><span class="do">## compare observed correlation with null correlation</span></span>
<span id="cb29-19"><a href="#cb29-19" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressMessages</span>(devtools<span class="sc">::</span><span class="fu">source_gist</span>(<span class="st">"a925fea01b365a8c605e"</span>)) <span class="do">## load qqR fn https://gist.github.com/hakyim/a925fea01b365a8c605e</span></span>
<span id="cb29-20"><a href="#cb29-20" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressMessages</span>(devtools<span class="sc">::</span><span class="fu">source_gist</span>(<span class="st">"38431b74c6c0bf90c12f"</span>)) <span class="do">## qqunif https://gist.github.com/hakyim/38431b74c6c0bf90c12f</span></span>
<span id="cb29-21"><a href="#cb29-21" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressMessages</span>(devtools<span class="sc">::</span><span class="fu">source_gist</span>(<span class="st">"115403f16bec0a0e871f3616d552ce9b"</span>)) <span class="do">## source ratxcan functions https://gist.github.com/hakyim/115403f16bec0a0e871f3616d552ce9b </span></span>
<span id="cb29-22"><a href="#cb29-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-23"><a href="#cb29-23" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressMessages</span>(<span class="fu">library</span>(tidyverse))</span>
<span id="cb29-24"><a href="#cb29-24" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressMessages</span>(<span class="fu">library</span>(glue))</span>
<span id="cb29-25"><a href="#cb29-25" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressMessages</span>(<span class="fu">library</span>(RSQLite))</span>
<span id="cb29-26"><a href="#cb29-26" aria-hidden="true" tabindex="-1"></a><span class="co">#suppressMessages(library(expm))</span></span>
<span id="cb29-27"><a href="#cb29-27" aria-hidden="true" tabindex="-1"></a><span class="co">#suppressMessages(library(readxl))</span></span>
<span id="cb29-28"><a href="#cb29-28" aria-hidden="true" tabindex="-1"></a><span class="co"># install.packages("devtools")</span></span>
<span id="cb29-29"><a href="#cb29-29" aria-hidden="true" tabindex="-1"></a><span class="co"># library("devtools")</span></span>
<span id="cb29-30"><a href="#cb29-30" aria-hidden="true" tabindex="-1"></a><span class="co"># install_github("jdstorey/qvalue")</span></span>
<span id="cb29-31"><a href="#cb29-31" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressMessages</span>(<span class="fu">library</span>(qvalue))</span>
<span id="cb29-32"><a href="#cb29-32" aria-hidden="true" tabindex="-1"></a><span class="co"># if (!require("BiocManager", quietly = TRUE))</span></span>
<span id="cb29-33"><a href="#cb29-33" aria-hidden="true" tabindex="-1"></a><span class="co">#     install.packages("BiocManager")</span></span>
<span id="cb29-34"><a href="#cb29-34" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb29-35"><a href="#cb29-35" aria-hidden="true" tabindex="-1"></a><span class="co"># BiocManager::install("biomaRt")</span></span>
<span id="cb29-36"><a href="#cb29-36" aria-hidden="true" tabindex="-1"></a><span class="do">##suppressMessages(library(biomaRt))</span></span>
<span id="cb29-37"><a href="#cb29-37" aria-hidden="true" tabindex="-1"></a><span class="do">##install.packages("ggrepel")</span></span>
<span id="cb29-38"><a href="#cb29-38" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressMessages</span>(<span class="fu">library</span>(ggrepel))</span>
<span id="cb29-39"><a href="#cb29-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-40"><a href="#cb29-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-41"><a href="#cb29-41" aria-hidden="true" tabindex="-1"></a>recalc<span class="ot">=</span><span class="cn">FALSE</span></span>
<span id="cb29-42"><a href="#cb29-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-43"><a href="#cb29-43" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-44"><a href="#cb29-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-45"><a href="#cb29-45" aria-hidden="true" tabindex="-1"></a><span class="fu">## define data and software paths for R</span></span>
<span id="cb29-46"><a href="#cb29-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-47"><a href="#cb29-47" aria-hidden="true" tabindex="-1"></a><span class="in">```{r define data and software paths}</span></span>
<span id="cb29-48"><a href="#cb29-48" aria-hidden="true" tabindex="-1"></a>WEBDATA<span class="ot">=</span><span class="st">"/Users/haekyungim/Library/CloudStorage/Box-Box/LargeFiles/imlab-data/data-Github/web-data"</span></span>
<span id="cb29-49"><a href="#cb29-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-50"><a href="#cb29-50" aria-hidden="true" tabindex="-1"></a>PLINK<span class="ot">=</span><span class="st">"/Users/haekyungim/bin/plink_mac_20231211/plink"</span></span>
<span id="cb29-51"><a href="#cb29-51" aria-hidden="true" tabindex="-1"></a>GCTA<span class="ot">=</span><span class="st">"/Users/haekyungim/bin/gcta-1.94.2-MacOS-ARM-x86_64/gcta64"</span></span>
<span id="cb29-52"><a href="#cb29-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-53"><a href="#cb29-53" aria-hidden="true" tabindex="-1"></a>INPUT <span class="ot">&lt;-</span> <span class="fu">glue</span>(<span class="st">"{WEBDATA}/ratxcan-tutorial"</span>) <span class="do">## this has the input data </span></span>
<span id="cb29-54"><a href="#cb29-54" aria-hidden="true" tabindex="-1"></a>OUTPUT <span class="ot">&lt;-</span> <span class="fu">glue</span>(<span class="st">"{WEBDATA}/2023-11-27-ratxcan-tutorial/scratch"</span>) <span class="do">## this has the output data, intermediate results</span></span>
<span id="cb29-55"><a href="#cb29-55" aria-hidden="true" tabindex="-1"></a>GENO <span class="ot">=</span> <span class="fu">glue</span>(<span class="st">"{WEBDATA}/2023-11-27-ratxcan-tutorial/data/genotype/"</span>)</span>
<span id="cb29-56"><a href="#cb29-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-57"><a href="#cb29-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-58"><a href="#cb29-58" aria-hidden="true" tabindex="-1"></a>OUT_TEMPO <span class="ot">=</span> <span class="fu">glue</span>(<span class="st">"{OUTPUT}/testing-2024-08-29"</span>)</span>
<span id="cb29-59"><a href="#cb29-59" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-60"><a href="#cb29-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-61"><a href="#cb29-61" aria-hidden="true" tabindex="-1"></a><span class="fu">## define data and software for the terminal</span></span>
<span id="cb29-62"><a href="#cb29-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-63"><a href="#cb29-63" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash define data and software paths bash}</span></span>
<span id="cb29-64"><a href="#cb29-64" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: FALSE</span></span>
<span id="cb29-65"><a href="#cb29-65" aria-hidden="true" tabindex="-1"></a><span class="va">WEBDATA</span><span class="op">=</span><span class="st">"/Users/haekyungim/Library/CloudStorage/Box-Box/LargeFiles/imlab-data/data-Github/web-data"</span></span>
<span id="cb29-66"><a href="#cb29-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-67"><a href="#cb29-67" aria-hidden="true" tabindex="-1"></a><span class="va">PLINK</span><span class="op">=</span><span class="st">"/Users/haekyungim/bin/plink_mac_20231211/plink"</span></span>
<span id="cb29-68"><a href="#cb29-68" aria-hidden="true" tabindex="-1"></a><span class="va">GCTA</span><span class="op">=</span><span class="st">"/Users/haekyungim/bin/gcta-1.94.2-MacOS-ARM-x86_64/gcta64"</span></span>
<span id="cb29-69"><a href="#cb29-69" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb29-70"><a href="#cb29-70" aria-hidden="true" tabindex="-1"></a><span class="va">INPUT</span><span class="op">=</span><span class="va">$WEBDATA</span>/ratxcan-tutorial</span>
<span id="cb29-71"><a href="#cb29-71" aria-hidden="true" tabindex="-1"></a><span class="va">OUTPUT</span><span class="op">=</span><span class="va">$WEBDATA</span>/2023-11-27-ratxcan-tutorial/scratch</span>
<span id="cb29-72"><a href="#cb29-72" aria-hidden="true" tabindex="-1"></a><span class="va">GENO</span><span class="op">=</span><span class="va">$WEBDATA</span>/2023-11-27-ratxcan-tutorial/data/genotype/</span>
<span id="cb29-73"><a href="#cb29-73" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-74"><a href="#cb29-74" aria-hidden="true" tabindex="-1"></a><span class="va">OUT_TEMPO</span><span class="op">=</span><span class="va">$OUTPUT</span>/testing-2024-08-29</span>
<span id="cb29-75"><a href="#cb29-75" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-76"><a href="#cb29-76" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-77"><a href="#cb29-77" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-78"><a href="#cb29-78" aria-hidden="true" tabindex="-1"></a><span class="fu">## read grm matrix</span></span>
<span id="cb29-79"><a href="#cb29-79" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-80"><a href="#cb29-80" aria-hidden="true" tabindex="-1"></a><span class="in">```{r read GRM mat}</span></span>
<span id="cb29-81"><a href="#cb29-81" aria-hidden="true" tabindex="-1"></a>grm_mat <span class="ot">&lt;-</span> <span class="fu">read_GRMBin</span>(<span class="fu">glue</span>(<span class="st">"{OUTPUT}/rat6k_autosome.grm"</span>))</span>
<span id="cb29-82"><a href="#cb29-82" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-83"><a href="#cb29-83" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-84"><a href="#cb29-84" aria-hidden="true" tabindex="-1"></a><span class="fu">## define myplot</span></span>
<span id="cb29-85"><a href="#cb29-85" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-86"><a href="#cb29-86" aria-hidden="true" tabindex="-1"></a><span class="in">```{r define myplot}</span></span>
<span id="cb29-87"><a href="#cb29-87" aria-hidden="true" tabindex="-1"></a>myplot <span class="ot">&lt;-</span> <span class="cf">function</span>(tempres, <span class="at">post_titulo=</span><span class="st">""</span>,<span class="at">semilla=</span><span class="st">""</span>) {</span>
<span id="cb29-88"><a href="#cb29-88" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create a data frame with specific columns</span></span>
<span id="cb29-89"><a href="#cb29-89" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb29-90"><a href="#cb29-90" aria-hidden="true" tabindex="-1"></a>    <span class="at">p0.01_yes =</span> <span class="fu">apply</span>(tempres<span class="sc">$</span>pmat_correct, <span class="dv">2</span>, <span class="cf">function</span>(x) <span class="fu">mean</span>(x <span class="sc">&lt;</span> <span class="fl">0.01</span>)),</span>
<span id="cb29-91"><a href="#cb29-91" aria-hidden="true" tabindex="-1"></a>    <span class="at">p0.01_no =</span> <span class="fu">apply</span>(tempres<span class="sc">$</span>pmat_raw, <span class="dv">2</span>, <span class="cf">function</span>(x) <span class="fu">mean</span>(x <span class="sc">&lt;</span> <span class="fl">0.01</span>)),</span>
<span id="cb29-92"><a href="#cb29-92" aria-hidden="true" tabindex="-1"></a>    <span class="at">p0.05_yes =</span> <span class="fu">apply</span>(tempres<span class="sc">$</span>pmat_correct, <span class="dv">2</span>, <span class="cf">function</span>(x) <span class="fu">mean</span>(x <span class="sc">&lt;</span> <span class="fl">0.05</span>)),</span>
<span id="cb29-93"><a href="#cb29-93" aria-hidden="true" tabindex="-1"></a>    <span class="at">p0.05_no =</span> <span class="fu">apply</span>(tempres<span class="sc">$</span>pmat_raw, <span class="dv">2</span>, <span class="cf">function</span>(x) <span class="fu">mean</span>(x <span class="sc">&lt;</span> <span class="fl">0.05</span>)),</span>
<span id="cb29-94"><a href="#cb29-94" aria-hidden="true" tabindex="-1"></a>    <span class="at">p0.10_yes =</span> <span class="fu">apply</span>(tempres<span class="sc">$</span>pmat_correct, <span class="dv">2</span>, <span class="cf">function</span>(x) <span class="fu">mean</span>(x <span class="sc">&lt;</span> <span class="fl">0.10</span>)), </span>
<span id="cb29-95"><a href="#cb29-95" aria-hidden="true" tabindex="-1"></a>    <span class="at">p0.10_no =</span> <span class="fu">apply</span>(tempres<span class="sc">$</span>pmat_raw, <span class="dv">2</span>, <span class="cf">function</span>(x) <span class="fu">mean</span>(x <span class="sc">&lt;</span> <span class="fl">0.10</span>))</span>
<span id="cb29-96"><a href="#cb29-96" aria-hidden="true" tabindex="-1"></a>    <span class="co"># ... [rest of your code for creating df] ...</span></span>
<span id="cb29-97"><a href="#cb29-97" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb29-98"><a href="#cb29-98" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-99"><a href="#cb29-99" aria-hidden="true" tabindex="-1"></a><span class="co"># Pivot the data frame to long format, specifying the columns to keep</span></span>
<span id="cb29-100"><a href="#cb29-100" aria-hidden="true" tabindex="-1"></a>df_long <span class="ot">&lt;-</span> <span class="fu">pivot_longer</span>(df, <span class="at">cols =</span> <span class="fu">starts_with</span>(<span class="st">"p"</span>))</span>
<span id="cb29-101"><a href="#cb29-101" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-102"><a href="#cb29-102" aria-hidden="true" tabindex="-1"></a>df_long <span class="ot">&lt;-</span> df_long <span class="sc">%&gt;%</span> <span class="fu">separate</span>(name,<span class="at">into =</span> <span class="fu">c</span>(<span class="st">"threshold"</span>,<span class="st">"corrected"</span>),<span class="at">sep=</span><span class="st">"_"</span>) <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">proportion=</span>value)</span>
<span id="cb29-103"><a href="#cb29-103" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-104"><a href="#cb29-104" aria-hidden="true" tabindex="-1"></a><span class="co"># Rename the name column to replace p0.xx with p&lt;0.xx</span></span>
<span id="cb29-105"><a href="#cb29-105" aria-hidden="true" tabindex="-1"></a>df_long <span class="ot">&lt;-</span> df_long <span class="sc">%&gt;%</span></span>
<span id="cb29-106"><a href="#cb29-106" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">threshold =</span> <span class="fu">gsub</span>(<span class="st">"p0</span><span class="sc">\\</span><span class="st">."</span>, <span class="st">"p&lt;0."</span>, threshold))</span>
<span id="cb29-107"><a href="#cb29-107" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-108"><a href="#cb29-108" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create boxplots with mean</span></span>
<span id="cb29-109"><a href="#cb29-109" aria-hidden="true" tabindex="-1"></a>  pp <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(df_long, <span class="fu">aes</span>(<span class="at">x =</span> threshold, <span class="at">y =</span> proportion, <span class="at">fill =</span> corrected)) <span class="sc">+</span></span>
<span id="cb29-110"><a href="#cb29-110" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_boxplot</span>(<span class="at">alpha =</span> <span class="fl">0.6</span>) <span class="sc">+</span></span>
<span id="cb29-111"><a href="#cb29-111" aria-hidden="true" tabindex="-1"></a>    <span class="fu">stat_summary</span>(<span class="at">fun =</span> mean, <span class="at">geom =</span> <span class="st">"point"</span>, <span class="at">shape =</span> <span class="dv">3</span>, <span class="at">size =</span> <span class="dv">2</span>, <span class="at">stroke =</span> <span class="dv">2</span>, <span class="at">color =</span> <span class="st">"blue"</span>,                  <span class="co">#position = position_dodge(width = 0.8)) +</span></span>
<span id="cb29-112"><a href="#cb29-112" aria-hidden="true" tabindex="-1"></a>                 <span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> <span class="sc">-</span><span class="fl">0.1</span>)) <span class="sc">+</span></span>
<span id="cb29-113"><a href="#cb29-113" aria-hidden="true" tabindex="-1"></a>    <span class="co">#stat_summary(fun = mean, geom = "crossbar",  size = .5, color = "blue") +</span></span>
<span id="cb29-114"><a href="#cb29-114" aria-hidden="true" tabindex="-1"></a>    <span class="co">#stat_summary(fun = mean, geom = "crossbar",  size = .5, color = "darkgray") +</span></span>
<span id="cb29-115"><a href="#cb29-115" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="fu">c</span>(<span class="fl">0.01</span>, <span class="fl">0.05</span>, <span class="fl">0.10</span>), <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">color =</span> <span class="st">"gray"</span>) <span class="sc">+</span></span>
<span id="cb29-116"><a href="#cb29-116" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_minimal</span>(<span class="at">base_size =</span> <span class="dv">17</span>) <span class="sc">+</span></span>
<span id="cb29-117"><a href="#cb29-117" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="fu">glue</span>(<span class="st">"Type I Error Calibration {semilla} {post_titulo}"</span>)) <span class="sc">+</span></span>
<span id="cb29-118"><a href="#cb29-118" aria-hidden="true" tabindex="-1"></a>    <span class="fu">xlab</span>(<span class="st">"significance"</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"false positive rate"</span>)</span>
<span id="cb29-119"><a href="#cb29-119" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-120"><a href="#cb29-120" aria-hidden="true" tabindex="-1"></a>  pp</span>
<span id="cb29-121"><a href="#cb29-121" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb29-122"><a href="#cb29-122" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-123"><a href="#cb29-123" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-124"><a href="#cb29-124" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-125"><a href="#cb29-125" aria-hidden="true" tabindex="-1"></a><span class="fu">### NEED DEBUG simulate $Y = Sigma^{1/2}\epsilon$ and run assoc with expr_mat</span></span>
<span id="cb29-126"><a href="#cb29-126" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-127"><a href="#cb29-127" aria-hidden="true" tabindex="-1"></a><span class="in">```{r LONG RUN Y sigma^.5 epsimat}</span></span>
<span id="cb29-128"><a href="#cb29-128" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: FALSE</span></span>
<span id="cb29-129"><a href="#cb29-129" aria-hidden="true" tabindex="-1"></a>nsam<span class="ot">=</span><span class="fu">nrow</span>(grm_mat)</span>
<span id="cb29-130"><a href="#cb29-130" aria-hidden="true" tabindex="-1"></a><span class="co">#ind=1:nsam</span></span>
<span id="cb29-131"><a href="#cb29-131" aria-hidden="true" tabindex="-1"></a>ind<span class="ot">=</span><span class="dv">1</span><span class="sc">:</span><span class="dv">1000</span></span>
<span id="cb29-132"><a href="#cb29-132" aria-hidden="true" tabindex="-1"></a>test_mat <span class="ot">=</span> grm_mat[ind,ind]</span>
<span id="cb29-133"><a href="#cb29-133" aria-hidden="true" tabindex="-1"></a>nsam<span class="ot">=</span><span class="fu">nrow</span>(test_mat)</span>
<span id="cb29-134"><a href="#cb29-134" aria-hidden="true" tabindex="-1"></a>Sigma <span class="ot">=</span> test_mat <span class="sc">*</span> h2 <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">-</span> h2) <span class="sc">*</span> <span class="fu">diag</span>(<span class="fu">rep</span>(<span class="dv">1</span>,nsam))</span>
<span id="cb29-135"><a href="#cb29-135" aria-hidden="true" tabindex="-1"></a>Sig_eigen <span class="ot">=</span> <span class="fu">eigen</span>(Sigma)</span>
<span id="cb29-136"><a href="#cb29-136" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(Sig_eigen<span class="sc">$</span>vectors) <span class="ot">=</span> <span class="fu">rownames</span>(Sigma)</span>
<span id="cb29-137"><a href="#cb29-137" aria-hidden="true" tabindex="-1"></a><span class="do">##sighalf = Sig_eigen$vectors %*% diag( sqrt(  Sig_eigen$values  ) ) %*% t(Sig_eigen$vectors)</span></span>
<span id="cb29-138"><a href="#cb29-138" aria-hidden="true" tabindex="-1"></a><span class="do">## make this multiplication more efficient using sweep</span></span>
<span id="cb29-139"><a href="#cb29-139" aria-hidden="true" tabindex="-1"></a>sighalf <span class="ot">=</span> Sig_eigen<span class="sc">$</span>vectors <span class="sc">%*%</span> <span class="fu">sweep</span>(<span class="fu">t</span>(Sig_eigen<span class="sc">$</span>vectors),<span class="dv">1</span>,<span class="fu">sqrt</span>(  Sig_eigen<span class="sc">$</span>values ),<span class="st">"*"</span>)</span>
<span id="cb29-140"><a href="#cb29-140" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-141"><a href="#cb29-141" aria-hidden="true" tabindex="-1"></a>sim_sigma_pheno <span class="ot">=</span> sighalf <span class="sc">%*%</span> <span class="fu">matrix</span>(<span class="fu">rnorm</span>(nsam <span class="sc">*</span> nsim), nsam, nsim) </span>
<span id="cb29-142"><a href="#cb29-142" aria-hidden="true" tabindex="-1"></a>sim_sigma_pheno<span class="ot">=</span><span class="fu">cbind</span>(<span class="at">FID=</span><span class="fu">rownames</span>(sim_sigma_pheno),<span class="at">IID=</span><span class="fu">rownames</span>(sim_sigma_pheno),<span class="fu">as.data.frame</span>(sim_sigma_pheno))</span>
<span id="cb29-143"><a href="#cb29-143" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-144"><a href="#cb29-144" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-145"><a href="#cb29-145" aria-hidden="true" tabindex="-1"></a>tic<span class="ot">=</span><span class="fu">Sys.time</span>()</span>
<span id="cb29-146"><a href="#cb29-146" aria-hidden="true" tabindex="-1"></a>tempres_sigma_pheno <span class="ot">&lt;-</span> <span class="fu">lmmGRM</span>(sim_sigma_pheno,grm_mat, h2,pred_expr,<span class="at">pheno_id_col=</span><span class="dv">1</span>, <span class="at">pheno_value_cols=</span><span class="dv">2</span><span class="sc">+</span>(<span class="dv">1</span><span class="sc">:</span>nsim))</span>
<span id="cb29-147"><a href="#cb29-147" aria-hidden="true" tabindex="-1"></a>toc<span class="ot">=</span><span class="fu">Sys.time</span>()</span>
<span id="cb29-148"><a href="#cb29-148" aria-hidden="true" tabindex="-1"></a>toc <span class="sc">-</span> tic</span>
<span id="cb29-149"><a href="#cb29-149" aria-hidden="true" tabindex="-1"></a>pp<span class="ot">&lt;-</span><span class="fu">myplot</span>(tempres_sigma_pheno,<span class="at">post_titulo =</span> <span class="fu">glue</span>(<span class="st">"sigma_pheno n= {nsam} - ii={ii}"</span>))</span>
<span id="cb29-150"><a href="#cb29-150" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(ii,<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb29-151"><a href="#cb29-151" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(pp)</span>
<span id="cb29-152"><a href="#cb29-152" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-153"><a href="#cb29-153" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-154"><a href="#cb29-154" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-155"><a href="#cb29-155" aria-hidden="true" tabindex="-1"></a><span class="fu">## read gene annotation</span></span>
<span id="cb29-156"><a href="#cb29-156" aria-hidden="true" tabindex="-1"></a><span class="in">```{r read gene annotation}</span></span>
<span id="cb29-157"><a href="#cb29-157" aria-hidden="true" tabindex="-1"></a><span class="co">#gene_annotation &lt;- readRDS(glue("{INPUT}/data/expression/gene_annotation.RDS"))</span></span>
<span id="cb29-158"><a href="#cb29-158" aria-hidden="true" tabindex="-1"></a>gene_annotation <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="fu">glue</span>(<span class="st">"{WEBDATA}/2023-11-27-ratxcan-tutorial/data/expression/gene_annotation.RDS"</span>))</span>
<span id="cb29-159"><a href="#cb29-159" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-160"><a href="#cb29-160" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-161"><a href="#cb29-161" aria-hidden="true" tabindex="-1"></a><span class="fu">## read predicted expression</span></span>
<span id="cb29-162"><a href="#cb29-162" aria-hidden="true" tabindex="-1"></a><span class="in">```{r read predicted expression}</span></span>
<span id="cb29-163"><a href="#cb29-163" aria-hidden="true" tabindex="-1"></a>read_pred_expr <span class="ot">=</span> <span class="cf">function</span>(filename)</span>
<span id="cb29-164"><a href="#cb29-164" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb29-165"><a href="#cb29-165" aria-hidden="true" tabindex="-1"></a>  <span class="do">##usage: Br_pred_expr = read_pred_expr(glue("{OUTPUT}/Br-hki-rat6k__predict.txt"))</span></span>
<span id="cb29-166"><a href="#cb29-166" aria-hidden="true" tabindex="-1"></a>  pred_expr <span class="ot">&lt;-</span> vroom<span class="sc">::</span><span class="fu">vroom</span>(filename) <span class="sc">%&gt;%</span> </span>
<span id="cb29-167"><a href="#cb29-167" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>FID) <span class="sc">%&gt;%</span>  <span class="co"># Remove the FID column</span></span>
<span id="cb29-168"><a href="#cb29-168" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">IID =</span> <span class="fu">str_split</span>(IID, <span class="st">"_"</span>, <span class="at">simplify =</span> <span class="cn">TRUE</span>)[, <span class="dv">1</span>])  <span class="co"># Keep the first part of IID</span></span>
<span id="cb29-169"><a href="#cb29-169" aria-hidden="true" tabindex="-1"></a>  pred_expr</span>
<span id="cb29-170"><a href="#cb29-170" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb29-171"><a href="#cb29-171" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-172"><a href="#cb29-172" aria-hidden="true" tabindex="-1"></a><span class="co"># pred_expr = read_pred_expr(glue("{OUTPUT}/Br-hki-rat6k__predict.txt"))</span></span>
<span id="cb29-173"><a href="#cb29-173" aria-hidden="true" tabindex="-1"></a><span class="co"># ## WHYYYY br-hki-rat6k__predict.txt has no chr 1????</span></span>
<span id="cb29-174"><a href="#cb29-174" aria-hidden="true" tabindex="-1"></a><span class="co"># df1= tibble(gene=names(pred_expr)) %&gt;% mutate(isinac=TRUE) %&gt;% left_join(gene_annotation,by=c("gene"="gene_id")) </span></span>
<span id="cb29-175"><a href="#cb29-175" aria-hidden="true" tabindex="-1"></a><span class="co"># df1 %&gt;% count(chr)</span></span>
<span id="cb29-176"><a href="#cb29-176" aria-hidden="true" tabindex="-1"></a><span class="co"># cat("{OUTPUT}/Br_rat6k__predict.txt has no chr 1 genes!!! Where did I lose them?\n")</span></span>
<span id="cb29-177"><a href="#cb29-177" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-178"><a href="#cb29-178" aria-hidden="true" tabindex="-1"></a>pred_expr <span class="ot">=</span> <span class="fu">read_pred_expr</span>(<span class="fu">glue</span>(<span class="st">"{OUTPUT}/AC-filtered__predict.txt"</span>))</span>
<span id="cb29-179"><a href="#cb29-179" aria-hidden="true" tabindex="-1"></a><span class="do">## checked that I have chr1 genes</span></span>
<span id="cb29-180"><a href="#cb29-180" aria-hidden="true" tabindex="-1"></a>df1<span class="ot">=</span> <span class="fu">tibble</span>(<span class="at">gene=</span><span class="fu">names</span>(pred_expr)[<span class="sc">-</span><span class="dv">1</span>]) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">isinac=</span><span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(gene_annotation,<span class="at">by=</span><span class="fu">c</span>(<span class="st">"gene"</span><span class="ot">=</span><span class="st">"gene_id"</span>)) </span>
<span id="cb29-181"><a href="#cb29-181" aria-hidden="true" tabindex="-1"></a>df1 <span class="sc">%&gt;%</span> <span class="fu">count</span>(chr)</span>
<span id="cb29-182"><a href="#cb29-182" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"{OUTPUT}/AC-filtered__predict.txt has genes in "</span>,<span class="fu">length</span>(<span class="fu">unique</span>(df1<span class="sc">$</span>chr)), <span class="st">"chr"</span>)</span>
<span id="cb29-183"><a href="#cb29-183" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-184"><a href="#cb29-184" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-185"><a href="#cb29-185" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-186"><a href="#cb29-186" aria-hidden="true" tabindex="-1"></a><span class="fu">## define function lmm with GRM</span></span>
<span id="cb29-187"><a href="#cb29-187" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-188"><a href="#cb29-188" aria-hidden="true" tabindex="-1"></a><span class="in">```{r define lmmGRM function}</span></span>
<span id="cb29-189"><a href="#cb29-189" aria-hidden="true" tabindex="-1"></a><span class="do">## HERE WE USE THE FULL GRM MATRIX AND CALCULATE THE INVERSE OF THE SIGMA MATRIX</span></span>
<span id="cb29-190"><a href="#cb29-190" aria-hidden="true" tabindex="-1"></a><span class="do">## define lmm association function </span></span>
<span id="cb29-191"><a href="#cb29-191" aria-hidden="true" tabindex="-1"></a>lmmGRM <span class="ot">=</span> <span class="cf">function</span>(pheno, grm_mat, h2, pred_expr,<span class="at">pheno_id_col=</span><span class="dv">1</span>,<span class="at">pheno_value_cols=</span><span class="dv">6</span><span class="sc">:</span><span class="dv">6</span>,<span class="at">out=</span><span class="cn">NULL</span>)</span>
<span id="cb29-192"><a href="#cb29-192" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb29-193"><a href="#cb29-193" aria-hidden="true" tabindex="-1"></a>  <span class="do">## input pheno is a data frame with id column pheno_id_col=1 by default</span></span>
<span id="cb29-194"><a href="#cb29-194" aria-hidden="true" tabindex="-1"></a>  <span class="do">## phenotype values are in pheno_value_cols, 6:6 by default (SCORE column location in plink output), it can have more than one phenotype</span></span>
<span id="cb29-195"><a href="#cb29-195" aria-hidden="true" tabindex="-1"></a>  <span class="do">## but h2 has to be the same, this is useful when running simulations with different h2</span></span>
<span id="cb29-196"><a href="#cb29-196" aria-hidden="true" tabindex="-1"></a>  <span class="do">## call lmmXcan(pheno %&gt;% select(IID,SCORE))</span></span>
<span id="cb29-197"><a href="#cb29-197" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb29-198"><a href="#cb29-198" aria-hidden="true" tabindex="-1"></a>  <span class="do">## format pheno to matrix form</span></span>
<span id="cb29-199"><a href="#cb29-199" aria-hidden="true" tabindex="-1"></a>  phenomat <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(pheno[,pheno_value_cols])</span>
<span id="cb29-200"><a href="#cb29-200" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(phenomat) <span class="ot">&lt;-</span> pheno[[pheno_id_col]]</span>
<span id="cb29-201"><a href="#cb29-201" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb29-202"><a href="#cb29-202" aria-hidden="true" tabindex="-1"></a>  <span class="do">## turn pred_expr into matrix with rownames =IID, keep only IIDs in ymat</span></span>
<span id="cb29-203"><a href="#cb29-203" aria-hidden="true" tabindex="-1"></a>  exp_mat <span class="ot">=</span> <span class="fu">as.matrix</span>(pred_expr <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span>IID))</span>
<span id="cb29-204"><a href="#cb29-204" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(exp_mat) <span class="ot">=</span> pred_expr<span class="sc">$</span>IID</span>
<span id="cb29-205"><a href="#cb29-205" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-206"><a href="#cb29-206" aria-hidden="true" tabindex="-1"></a>  <span class="do">## align pheno and expr matrices</span></span>
<span id="cb29-207"><a href="#cb29-207" aria-hidden="true" tabindex="-1"></a>  idlist <span class="ot">=</span> <span class="fu">intersect</span>(<span class="fu">rownames</span>(phenomat), <span class="fu">rownames</span>(exp_mat))</span>
<span id="cb29-208"><a href="#cb29-208" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb29-209"><a href="#cb29-209" aria-hidden="true" tabindex="-1"></a>  nsam <span class="ot">=</span> <span class="fu">length</span>(idlist)</span>
<span id="cb29-210"><a href="#cb29-210" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb29-211"><a href="#cb29-211" aria-hidden="true" tabindex="-1"></a>  <span class="do">## CALCULATE SIGMA</span></span>
<span id="cb29-212"><a href="#cb29-212" aria-hidden="true" tabindex="-1"></a>  ID_mat <span class="ot">=</span> <span class="fu">diag</span>(<span class="fu">rep</span>(<span class="dv">1</span>,nsam))</span>
<span id="cb29-213"><a href="#cb29-213" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb29-214"><a href="#cb29-214" aria-hidden="true" tabindex="-1"></a>  <span class="co">#testing_scale_grm = TRUE</span></span>
<span id="cb29-215"><a href="#cb29-215" aria-hidden="true" tabindex="-1"></a>  <span class="co">#if(testing_scale_grm) grm_mat = sweep( sweep(grm_mat,2, 1/sqrt(diag(grm_mat)),"*"), 1, 1/sqrt(diag(grm_mat)),"*")    </span></span>
<span id="cb29-216"><a href="#cb29-216" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb29-217"><a href="#cb29-217" aria-hidden="true" tabindex="-1"></a>  Sigma <span class="ot">=</span> grm_mat[idlist,idlist] <span class="sc">*</span> h2 <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">-</span> h2) <span class="sc">*</span> ID_mat</span>
<span id="cb29-218"><a href="#cb29-218" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb29-219"><a href="#cb29-219" aria-hidden="true" tabindex="-1"></a>  Sig_eigen <span class="ot">=</span> <span class="fu">eigen</span>(Sigma)</span>
<span id="cb29-220"><a href="#cb29-220" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(Sig_eigen<span class="sc">$</span>vectors) <span class="ot">=</span> <span class="fu">rownames</span>(Sigma)</span>
<span id="cb29-221"><a href="#cb29-221" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb29-222"><a href="#cb29-222" aria-hidden="true" tabindex="-1"></a>  isighalf <span class="ot">=</span> Sig_eigen<span class="sc">$</span>vectors <span class="sc">%*%</span> <span class="fu">diag</span>( <span class="dv">1</span> <span class="sc">/</span> <span class="fu">sqrt</span>(  Sig_eigen<span class="sc">$</span>values  ) ) <span class="sc">%*%</span> <span class="fu">t</span>(Sig_eigen<span class="sc">$</span>vectors)</span>
<span id="cb29-223"><a href="#cb29-223" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb29-224"><a href="#cb29-224" aria-hidden="true" tabindex="-1"></a>  <span class="do">## perform raw association</span></span>
<span id="cb29-225"><a href="#cb29-225" aria-hidden="true" tabindex="-1"></a>  cormat_raw <span class="ot">=</span> <span class="fu">matrix_lm</span>(phenomat[idlist,, <span class="at">drop =</span> <span class="cn">FALSE</span>], exp_mat[idlist,])</span>
<span id="cb29-226"><a href="#cb29-226" aria-hidden="true" tabindex="-1"></a>  pmat_raw <span class="ot">=</span> <span class="fu">cor2pval</span>(cormat_raw,nsam)</span>
<span id="cb29-227"><a href="#cb29-227" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(pmat_raw) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"cor_"</span>, <span class="st">"pval_"</span>, <span class="fu">colnames</span>(pmat_raw))</span>
<span id="cb29-228"><a href="#cb29-228" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb29-229"><a href="#cb29-229" aria-hidden="true" tabindex="-1"></a>  <span class="do">## perform corrected association</span></span>
<span id="cb29-230"><a href="#cb29-230" aria-hidden="true" tabindex="-1"></a>  cormat_correct <span class="ot">=</span> <span class="fu">matrix_lm</span>(isighalf<span class="sc">%*%</span> phenomat[idlist,, <span class="at">drop =</span> <span class="cn">FALSE</span>], isighalf <span class="sc">%*%</span> exp_mat[idlist,])</span>
<span id="cb29-231"><a href="#cb29-231" aria-hidden="true" tabindex="-1"></a>  pmat_correct <span class="ot">=</span> <span class="fu">cor2pval</span>(cormat_correct,nsam)</span>
<span id="cb29-232"><a href="#cb29-232" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(pmat_correct) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"cor_"</span>, <span class="st">"pval_"</span>, <span class="fu">colnames</span>(pmat_correct))</span>
<span id="cb29-233"><a href="#cb29-233" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb29-234"><a href="#cb29-234" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.null</span>(out))</span>
<span id="cb29-235"><a href="#cb29-235" aria-hidden="true" tabindex="-1"></a>  {</span>
<span id="cb29-236"><a href="#cb29-236" aria-hidden="true" tabindex="-1"></a>    <span class="fu">saveRDS</span>(cormat_correct,<span class="at">file =</span> <span class="fu">glue</span>(<span class="st">"{out}_cormat_correct.RDS"</span>))</span>
<span id="cb29-237"><a href="#cb29-237" aria-hidden="true" tabindex="-1"></a>    <span class="fu">saveRDS</span>(pmat_correct,  <span class="at">file =</span> <span class="fu">glue</span>(<span class="st">"{out}_pmat_correct.RDS"</span>))</span>
<span id="cb29-238"><a href="#cb29-238" aria-hidden="true" tabindex="-1"></a>    <span class="fu">saveRDS</span>(cormat_raw,    <span class="at">file =</span> <span class="fu">glue</span>(<span class="st">"{out}_cormat_raw.RDS"</span>))</span>
<span id="cb29-239"><a href="#cb29-239" aria-hidden="true" tabindex="-1"></a>    <span class="fu">saveRDS</span>(pmat_raw,      <span class="at">file =</span> <span class="fu">glue</span>(<span class="st">"{out}_pmat_raw.RDS"</span>))</span>
<span id="cb29-240"><a href="#cb29-240" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb29-241"><a href="#cb29-241" aria-hidden="true" tabindex="-1"></a>  res <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb29-242"><a href="#cb29-242" aria-hidden="true" tabindex="-1"></a>    <span class="at">cormat_correct=</span>cormat_correct, </span>
<span id="cb29-243"><a href="#cb29-243" aria-hidden="true" tabindex="-1"></a>    <span class="at">pmat_correct=</span>pmat_correct, </span>
<span id="cb29-244"><a href="#cb29-244" aria-hidden="true" tabindex="-1"></a>    <span class="at">cormat_raw=</span>cormat_raw, </span>
<span id="cb29-245"><a href="#cb29-245" aria-hidden="true" tabindex="-1"></a>    <span class="at">pmat_raw=</span>pmat_raw)</span>
<span id="cb29-246"><a href="#cb29-246" aria-hidden="true" tabindex="-1"></a>  res</span>
<span id="cb29-247"><a href="#cb29-247" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb29-248"><a href="#cb29-248" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb29-249"><a href="#cb29-249" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-250"><a href="#cb29-250" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-251"><a href="#cb29-251" aria-hidden="true" tabindex="-1"></a><span class="fu">## run ratXcan</span></span>
<span id="cb29-254"><a href="#cb29-254" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb29-255"><a href="#cb29-255" aria-hidden="true" tabindex="-1"></a><span class="do">## read simulated Y</span></span>
<span id="cb29-256"><a href="#cb29-256" aria-hidden="true" tabindex="-1"></a>pheno <span class="ot">=</span> <span class="fu">read_table</span>(<span class="fu">glue</span>(<span class="st">"{WEBDATA}/2023-11-27-ratxcan-tutorial/scratch/sim/tempo/PRS_output_100-32240.profile"</span>))</span>
<span id="cb29-257"><a href="#cb29-257" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-258"><a href="#cb29-258" aria-hidden="true" tabindex="-1"></a><span class="do">## calculate h2 of pheno</span></span>
<span id="cb29-259"><a href="#cb29-259" aria-hidden="true" tabindex="-1"></a><span class="do">## mpheno 4 means fourth column after FID and IID</span></span>
<span id="cb29-260"><a href="#cb29-260" aria-hidden="true" tabindex="-1"></a><span class="do">## $GCTA --grm $OUTPUT/rat6k_autosome --reml --pheno $OUTPUT/sim/tempo/PRS_output_100-32240.profile --mpheno 4 --out $OUTPUT/test-2024-08-29</span></span>
<span id="cb29-261"><a href="#cb29-261" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-262"><a href="#cb29-262" aria-hidden="true" tabindex="-1"></a><span class="do">## create phenotype with h2</span></span>
<span id="cb29-263"><a href="#cb29-263" aria-hidden="true" tabindex="-1"></a>h2<span class="ot">=</span><span class="fl">0.5</span></span>
<span id="cb29-264"><a href="#cb29-264" aria-hidden="true" tabindex="-1"></a>pheno<span class="sc">$</span>trait <span class="ot">=</span> (pheno<span class="sc">$</span>SCORE <span class="sc">-</span> <span class="fu">mean</span>(pheno<span class="sc">$</span>SCORE))<span class="sc">/</span><span class="fu">sd</span>(pheno<span class="sc">$</span>SCORE) <span class="sc">*</span> <span class="fu">sqrt</span>(h2) <span class="sc">+</span> <span class="fu">sqrt</span>(<span class="dv">1</span> <span class="sc">-</span> h2)<span class="sc">*</span><span class="fu">rnorm</span>(<span class="fu">nrow</span>(pheno))</span>
<span id="cb29-265"><a href="#cb29-265" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-266"><a href="#cb29-266" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(recalc)</span>
<span id="cb29-267"><a href="#cb29-267" aria-hidden="true" tabindex="-1"></a>{res <span class="ot">=</span> <span class="fu">lmmGRM</span>(pheno, grm_mat, h2, pred_expr,<span class="at">pheno_id_col=</span><span class="dv">1</span>,<span class="at">pheno_value_cols=</span><span class="dv">7</span><span class="sc">:</span><span class="dv">7</span>,<span class="at">out=</span><span class="cn">NULL</span>)</span>
<span id="cb29-268"><a href="#cb29-268" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(res,<span class="fu">glue</span>(<span class="st">"{OUT_TEMPO}/ratxcan-ac-res.RDS"</span>))</span>
<span id="cb29-269"><a href="#cb29-269" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span></span>
<span id="cb29-270"><a href="#cb29-270" aria-hidden="true" tabindex="-1"></a>  res <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="fu">glue</span>(<span class="st">"{OUT_TEMPO}/ratxcan-ac-res.RDS"</span>))</span>
<span id="cb29-271"><a href="#cb29-271" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-272"><a href="#cb29-272" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-273"><a href="#cb29-273" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-274"><a href="#cb29-274" aria-hidden="true" tabindex="-1"></a><span class="fu">## calc GRM without chr 1</span></span>
<span id="cb29-275"><a href="#cb29-275" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-276"><a href="#cb29-276" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash calculate GRM without chr01}</span></span>
<span id="cb29-277"><a href="#cb29-277" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb29-278"><a href="#cb29-278" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-279"><a href="#cb29-279" aria-hidden="true" tabindex="-1"></a><span class="va">$PLINK</span> <span class="at">--bfile</span> <span class="va">$GENO</span>/rat6k_autosome <span class="at">--chr</span> 1 <span class="at">--write-snplist</span> <span class="at">--out</span> <span class="va">$OUT_TEMPO</span>/chr1_snps</span>
<span id="cb29-280"><a href="#cb29-280" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-281"><a href="#cb29-281" aria-hidden="true" tabindex="-1"></a><span class="va">$GCTA</span> <span class="at">--bfile</span> <span class="va">$GENO</span>/rat6k_autosome <span class="at">--exclude</span> <span class="va">$OUT_TEMPO</span>/chr1_snps.snplist <span class="at">--make-grm-bin</span> <span class="at">--out</span> <span class="va">$OUT_TEMPO</span>/rat6k_autosome_loco_chr01 <span class="at">--thread-num</span> 8</span>
<span id="cb29-282"><a href="#cb29-282" aria-hidden="true" tabindex="-1"></a><span class="co">## that took </span></span>
<span id="cb29-283"><a href="#cb29-283" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-284"><a href="#cb29-284" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-285"><a href="#cb29-285" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-286"><a href="#cb29-286" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-287"><a href="#cb29-287" aria-hidden="true" tabindex="-1"></a><span class="in">```{text }</span></span>
<span id="cb29-288"><a href="#cb29-288" aria-hidden="true" tabindex="-1"></a><span class="in">*******************************************************************</span></span>
<span id="cb29-289"><a href="#cb29-289" aria-hidden="true" tabindex="-1"></a><span class="in">* Genome-wide Complex Trait Analysis (GCTA)</span></span>
<span id="cb29-290"><a href="#cb29-290" aria-hidden="true" tabindex="-1"></a><span class="in">* version v1.94.1 Mac</span></span>
<span id="cb29-291"><a href="#cb29-291" aria-hidden="true" tabindex="-1"></a><span class="in">* (C) 2010-present, Yang Lab, Westlake University</span></span>
<span id="cb29-292"><a href="#cb29-292" aria-hidden="true" tabindex="-1"></a><span class="in">* Please report bugs to Jian Yang &lt;jian.yang@westlake.edu.cn&gt;</span></span>
<span id="cb29-293"><a href="#cb29-293" aria-hidden="true" tabindex="-1"></a><span class="in">*******************************************************************</span></span>
<span id="cb29-294"><a href="#cb29-294" aria-hidden="true" tabindex="-1"></a><span class="in">Analysis started at 16:52:27 CDT on Thu Aug 29 2024.</span></span>
<span id="cb29-295"><a href="#cb29-295" aria-hidden="true" tabindex="-1"></a><span class="in">Hostname: MED-ML-464.local</span></span>
<span id="cb29-296"><a href="#cb29-296" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-297"><a href="#cb29-297" aria-hidden="true" tabindex="-1"></a><span class="in">Accepted options:</span></span>
<span id="cb29-298"><a href="#cb29-298" aria-hidden="true" tabindex="-1"></a><span class="in">--bfile /Users/haekyungim/Library/CloudStorage/Box-Box/LargeFiles/imlab-data/data-Github/web-data/2023-11-27-ratxcan-tutorial/data/genotype//rat6k_autosome</span></span>
<span id="cb29-299"><a href="#cb29-299" aria-hidden="true" tabindex="-1"></a><span class="in">--exclude /Users/haekyungim/Library/CloudStorage/Box-Box/LargeFiles/imlab-data/data-Github/web-data/2023-11-27-ratxcan-tutorial/scratch/testing-2024-08-29/chr1_snps.snplist</span></span>
<span id="cb29-300"><a href="#cb29-300" aria-hidden="true" tabindex="-1"></a><span class="in">--make-grm-bin</span></span>
<span id="cb29-301"><a href="#cb29-301" aria-hidden="true" tabindex="-1"></a><span class="in">--out /Users/haekyungim/Library/CloudStorage/Box-Box/LargeFiles/imlab-data/data-Github/web-data/2023-11-27-ratxcan-tutorial/scratch/rat6k_autosome_loco_chr1</span></span>
<span id="cb29-302"><a href="#cb29-302" aria-hidden="true" tabindex="-1"></a><span class="in">--thread-num 8</span></span>
<span id="cb29-303"><a href="#cb29-303" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-304"><a href="#cb29-304" aria-hidden="true" tabindex="-1"></a><span class="in">Note: the program will be running on 8 threads.</span></span>
<span id="cb29-305"><a href="#cb29-305" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-306"><a href="#cb29-306" aria-hidden="true" tabindex="-1"></a><span class="in">Reading PLINK FAM file from [/Users/haekyungim/Library/CloudStorage/Box-Box/LargeFiles/imlab-data/data-Github/web-data/2023-11-27-ratxcan-tutorial/data/genotype//rat6k_autosome.fam].</span></span>
<span id="cb29-307"><a href="#cb29-307" aria-hidden="true" tabindex="-1"></a><span class="in">5628 individuals to be included from [/Users/haekyungim/Library/CloudStorage/Box-Box/LargeFiles/imlab-data/data-Github/web-data/2023-11-27-ratxcan-tutorial/data/genotype//rat6k_autosome.fam].</span></span>
<span id="cb29-308"><a href="#cb29-308" aria-hidden="true" tabindex="-1"></a><span class="in">Reading PLINK BIM file from [/Users/haekyungim/Library/CloudStorage/Box-Box/LargeFiles/imlab-data/data-Github/web-data/2023-11-27-ratxcan-tutorial/data/genotype//rat6k_autosome.bim].</span></span>
<span id="cb29-309"><a href="#cb29-309" aria-hidden="true" tabindex="-1"></a><span class="in">179895 SNPs to be included from [/Users/haekyungim/Library/CloudStorage/Box-Box/LargeFiles/imlab-data/data-Github/web-data/2023-11-27-ratxcan-tutorial/data/genotype//rat6k_autosome.bim].</span></span>
<span id="cb29-310"><a href="#cb29-310" aria-hidden="true" tabindex="-1"></a><span class="in">Reading a list of SNPs from [/Users/haekyungim/Library/CloudStorage/Box-Box/LargeFiles/imlab-data/data-Github/web-data/2023-11-27-ratxcan-tutorial/scratch/testing-2024-08-29/chr1_snps.snplist].</span></span>
<span id="cb29-311"><a href="#cb29-311" aria-hidden="true" tabindex="-1"></a><span class="in">17602 SNPs are excluded from [/Users/haekyungim/Library/CloudStorage/Box-Box/LargeFiles/imlab-data/data-Github/web-data/2023-11-27-ratxcan-tutorial/scratch/testing-2024-08-29/chr1_snps.snplist] and there are 162293 SNPs remaining.</span></span>
<span id="cb29-312"><a href="#cb29-312" aria-hidden="true" tabindex="-1"></a><span class="in">Reading PLINK BED file from [/Users/haekyungim/Library/CloudStorage/Box-Box/LargeFiles/imlab-data/data-Github/web-data/2023-11-27-ratxcan-tutorial/data/genotype//rat6k_autosome.bed] in SNP-major format ...</span></span>
<span id="cb29-313"><a href="#cb29-313" aria-hidden="true" tabindex="-1"></a><span class="in">Genotype data for 5628 individuals and 162293 SNPs to be included from [/Users/haekyungim/Library/CloudStorage/Box-Box/LargeFiles/imlab-data/data-Github/web-data/2023-11-27-ratxcan-tutorial/data/genotype//rat6k_autosome.bed].</span></span>
<span id="cb29-314"><a href="#cb29-314" aria-hidden="true" tabindex="-1"></a><span class="in">Calculating allele frequencies ...</span></span>
<span id="cb29-315"><a href="#cb29-315" aria-hidden="true" tabindex="-1"></a><span class="in">Recoding genotypes (individual major mode) ...</span></span>
<span id="cb29-316"><a href="#cb29-316" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-317"><a href="#cb29-317" aria-hidden="true" tabindex="-1"></a><span class="in">Calculating the genetic relationship matrix (GRM) ... (Note: default speed-optimized mode, may use huge RAM)</span></span>
<span id="cb29-318"><a href="#cb29-318" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-319"><a href="#cb29-319" aria-hidden="true" tabindex="-1"></a><span class="in">Summary of the GRM:</span></span>
<span id="cb29-320"><a href="#cb29-320" aria-hidden="true" tabindex="-1"></a><span class="in">Mean of diagonals = 0.990807</span></span>
<span id="cb29-321"><a href="#cb29-321" aria-hidden="true" tabindex="-1"></a><span class="in">Variance of diagonals = 0.00404564</span></span>
<span id="cb29-322"><a href="#cb29-322" aria-hidden="true" tabindex="-1"></a><span class="in">Mean of off-diagonals = -0.000176097</span></span>
<span id="cb29-323"><a href="#cb29-323" aria-hidden="true" tabindex="-1"></a><span class="in">Variance of off-diagonals = 0.00216189</span></span>
<span id="cb29-324"><a href="#cb29-324" aria-hidden="true" tabindex="-1"></a><span class="in">GRM of 5628 individuals has been saved in the file [/Users/haekyungim/Library/CloudStorage/Box-Box/LargeFiles/imlab-data/data-Github/web-data/2023-11-27-ratxcan-tutorial/scratch/rat6k_autosome_loco_chr1.grm.bin] (in binary format).</span></span>
<span id="cb29-325"><a href="#cb29-325" aria-hidden="true" tabindex="-1"></a><span class="in">Number of SNPs to calculate the genetic relationship between each pair of individuals has been saved in the file [/Users/haekyungim/Library/CloudStorage/Box-Box/LargeFiles/imlab-data/data-Github/web-data/2023-11-27-ratxcan-tutorial/scratch/rat6k_autosome_loco_chr1.grm.N.bin] (in binary format).</span></span>
<span id="cb29-326"><a href="#cb29-326" aria-hidden="true" tabindex="-1"></a><span class="in">IDs for the GRM file [/Users/haekyungim/Library/CloudStorage/Box-Box/LargeFiles/imlab-data/data-Github/web-data/2023-11-27-ratxcan-tutorial/scratch/rat6k_autosome_loco_chr1.grm.bin] have been saved in the file [/Users/haekyungim/Library/CloudStorage/Box-Box/LargeFiles/imlab-data/data-Github/web-data/2023-11-27-ratxcan-tutorial/scratch/rat6k_autosome_loco_chr1.grm.id].</span></span>
<span id="cb29-327"><a href="#cb29-327" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-328"><a href="#cb29-328" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-329"><a href="#cb29-329" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-330"><a href="#cb29-330" aria-hidden="true" tabindex="-1"></a><span class="fu">## compute GRM loco chr01 and lmmGRM with updated h2</span></span>
<span id="cb29-331"><a href="#cb29-331" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-332"><a href="#cb29-332" aria-hidden="true" tabindex="-1"></a><span class="in">```{r compute GRM no chr01 and lmmGRM}</span></span>
<span id="cb29-333"><a href="#cb29-333" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-334"><a href="#cb29-334" aria-hidden="true" tabindex="-1"></a>grm_mat_no_chr01 <span class="ot">=</span> <span class="fu">read_GRMBin</span>(<span class="fu">glue</span>(<span class="st">"{OUT_TEMPO}/rat6k_autosome_loco_chr01.grm"</span>))</span>
<span id="cb29-335"><a href="#cb29-335" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-336"><a href="#cb29-336" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-337"><a href="#cb29-337" aria-hidden="true" tabindex="-1"></a><span class="do">## need to recalculate h2</span></span>
<span id="cb29-338"><a href="#cb29-338" aria-hidden="true" tabindex="-1"></a><span class="co"># $GCTA --reml --grm $OUT_TEMPO/rat6k_autosome  --out $OUT_TEMPO/rat6k_autosome_loco_chr01 --thread-num 8</span></span>
<span id="cb29-339"><a href="#cb29-339" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-340"><a href="#cb29-340" aria-hidden="true" tabindex="-1"></a><span class="do">## save the phenotype with error added</span></span>
<span id="cb29-341"><a href="#cb29-341" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(pheno <span class="sc">%&gt;%</span> <span class="fu">select</span>(IID, FID, trait),<span class="at">file=</span><span class="fu">glue</span>(<span class="st">"{OUT_TEMPO}/pheno_PRS_output_100-32240.txt"</span>))</span>
<span id="cb29-342"><a href="#cb29-342" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-343"><a href="#cb29-343" aria-hidden="true" tabindex="-1"></a><span class="co"># $GCTA --grm $OUT_TEMPO/rat6k_autosome_loco_chr01 --reml --pheno $OUT_TEMPO/pheno_PRS_output_100-32240.txt --mpheno 1 --out $OUT_TEMPO/loco_chr01 --thread-num 8</span></span>
<span id="cb29-344"><a href="#cb29-344" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-345"><a href="#cb29-345" aria-hidden="true" tabindex="-1"></a>h2_loco_chr01 <span class="ot">=</span> <span class="fu">read_tsv</span>(<span class="fu">glue</span>(<span class="st">"{OUT_TEMPO}/loco_chr01.hsq"</span>)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Source<span class="sc">==</span><span class="st">"V(G)/Vp"</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(Variance)</span>
<span id="cb29-346"><a href="#cb29-346" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-347"><a href="#cb29-347" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(recalc)</span>
<span id="cb29-348"><a href="#cb29-348" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb29-349"><a href="#cb29-349" aria-hidden="true" tabindex="-1"></a>  res_no_chr01 <span class="ot">=</span> <span class="fu">lmmGRM</span>(pheno, grm_mat_no_chr01, <span class="at">h2=</span> h2_loco_chr01, pred_expr,<span class="at">pheno_id_col=</span><span class="dv">1</span>,<span class="at">pheno_value_cols=</span><span class="dv">7</span><span class="sc">:</span><span class="dv">7</span>,<span class="at">out=</span><span class="cn">NULL</span>)</span>
<span id="cb29-350"><a href="#cb29-350" aria-hidden="true" tabindex="-1"></a>  <span class="fu">saveRDS</span>(res_no_chr01,<span class="at">file=</span><span class="fu">glue</span>(<span class="st">"{OUT_TEMPO}/ratxcan-ac-res-loco-chr01.RDS"</span>))</span>
<span id="cb29-351"><a href="#cb29-351" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> res_no_chr01<span class="ot">=</span><span class="fu">readRDS</span>(<span class="fu">glue</span>(<span class="st">"{OUT_TEMPO}/ratxcan-ac-res-loco-chr01.RDS"</span>))</span>
<span id="cb29-352"><a href="#cb29-352" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-353"><a href="#cb29-353" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-354"><a href="#cb29-354" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-355"><a href="#cb29-355" aria-hidden="true" tabindex="-1"></a><span class="fu">## compare pvalues in chr 1 vs other chromosomes</span></span>
<span id="cb29-356"><a href="#cb29-356" aria-hidden="true" tabindex="-1"></a><span class="in">```{r compare pvalues in chr 1 vs other chromosomes}</span></span>
<span id="cb29-357"><a href="#cb29-357" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-358"><a href="#cb29-358" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-359"><a href="#cb29-359" aria-hidden="true" tabindex="-1"></a>pmat2df <span class="ot">=</span> <span class="cf">function</span>(pmat)</span>
<span id="cb29-360"><a href="#cb29-360" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb29-361"><a href="#cb29-361" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">=</span> <span class="fu">data.frame</span>(pmat)</span>
<span id="cb29-362"><a href="#cb29-362" aria-hidden="true" tabindex="-1"></a>  df<span class="sc">$</span>gene <span class="ot">=</span> <span class="fu">rownames</span>(pmat)</span>
<span id="cb29-363"><a href="#cb29-363" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(df) <span class="ot">=</span> <span class="cn">NULL</span></span>
<span id="cb29-364"><a href="#cb29-364" aria-hidden="true" tabindex="-1"></a>  df <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(gene_annotation <span class="sc">%&gt;%</span> <span class="fu">select</span>(gene_id,chr,gene_name),<span class="at">by=</span><span class="fu">c</span>(<span class="st">"gene"</span><span class="ot">=</span><span class="st">"gene_id"</span>) )</span>
<span id="cb29-365"><a href="#cb29-365" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb29-366"><a href="#cb29-366" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-367"><a href="#cb29-367" aria-hidden="true" tabindex="-1"></a>df_loco_chr01 <span class="ot">=</span> <span class="fu">pmat2df</span>(res_no_chr01<span class="sc">$</span>pmat_correct) </span>
<span id="cb29-368"><a href="#cb29-368" aria-hidden="true" tabindex="-1"></a>df <span class="ot">=</span> <span class="fu">pmat2df</span>(res<span class="sc">$</span>pmat_correct) </span>
<span id="cb29-369"><a href="#cb29-369" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-370"><a href="#cb29-370" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow=</span><span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">2</span>))</span>
<span id="cb29-371"><a href="#cb29-371" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(df_loco_chr01 <span class="sc">%&gt;%</span> <span class="fu">filter</span>(chr<span class="sc">==</span><span class="st">"1"</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(trait),<span class="at">main=</span><span class="st">"ratxcan GRM no chr1 - genes in chr1"</span>)</span>
<span id="cb29-372"><a href="#cb29-372" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(df_loco_chr01 <span class="sc">%&gt;%</span> <span class="fu">filter</span>(chr<span class="sc">!=</span><span class="st">"1"</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(trait),<span class="at">main=</span><span class="st">"ratxcan GRM no chr1 - genes chr!=1"</span>)</span>
<span id="cb29-373"><a href="#cb29-373" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(df <span class="sc">%&gt;%</span> <span class="fu">filter</span>(chr<span class="sc">==</span><span class="st">"1"</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(trait),<span class="at">main=</span><span class="st">"ratxcan GRM all - genes in chr1"</span>)</span>
<span id="cb29-374"><a href="#cb29-374" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(df <span class="sc">%&gt;%</span> <span class="fu">filter</span>(chr<span class="sc">!=</span><span class="st">"1"</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(trait),<span class="at">main=</span><span class="st">"ratxcan GRM all - genes chr!=1"</span>)</span>
<span id="cb29-375"><a href="#cb29-375" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">1</span>))</span>
<span id="cb29-376"><a href="#cb29-376" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-377"><a href="#cb29-377" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow=</span><span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">2</span>))</span>
<span id="cb29-378"><a href="#cb29-378" aria-hidden="true" tabindex="-1"></a>rango <span class="ot">=</span> <span class="fu">range</span>(<span class="sc">-</span><span class="fu">log10</span>(<span class="fu">c</span>(df_loco_chr01<span class="sc">$</span>trait, df<span class="sc">$</span>trait)))</span>
<span id="cb29-379"><a href="#cb29-379" aria-hidden="true" tabindex="-1"></a><span class="fu">qqunif</span>(df_loco_chr01 <span class="sc">%&gt;%</span> <span class="fu">filter</span>(chr<span class="sc">==</span><span class="st">"1"</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(trait),<span class="at">main=</span><span class="st">"ratxcan GRM no chr1 - genes in chr1"</span>,<span class="at">BH=</span><span class="cn">FALSE</span>,<span class="at">CI=</span><span class="cn">FALSE</span>,<span class="at">xlim=</span>rango,<span class="at">ylim=</span>rango)</span>
<span id="cb29-380"><a href="#cb29-380" aria-hidden="true" tabindex="-1"></a><span class="fu">qqunif</span>(df_loco_chr01 <span class="sc">%&gt;%</span> <span class="fu">filter</span>(chr<span class="sc">!=</span><span class="st">"1"</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(trait),<span class="at">main=</span><span class="st">"ratxcan GRM no chr1 - genes chr!=1"</span>,<span class="at">BH=</span><span class="cn">FALSE</span>,<span class="at">CI=</span><span class="cn">FALSE</span>,<span class="at">xlim=</span>rango,<span class="at">ylim=</span>rango)</span>
<span id="cb29-381"><a href="#cb29-381" aria-hidden="true" tabindex="-1"></a><span class="fu">qqunif</span>(df <span class="sc">%&gt;%</span> <span class="fu">filter</span>(chr<span class="sc">==</span><span class="st">"1"</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(trait),<span class="at">main=</span><span class="st">"ratxcan genes in chr1"</span>,<span class="at">BH=</span><span class="cn">FALSE</span>,<span class="at">CI=</span><span class="cn">FALSE</span>,<span class="at">xlim=</span>rango,<span class="at">ylim=</span>rango)</span>
<span id="cb29-382"><a href="#cb29-382" aria-hidden="true" tabindex="-1"></a><span class="fu">qqunif</span>(df <span class="sc">%&gt;%</span> <span class="fu">filter</span>(chr<span class="sc">!=</span><span class="st">"1"</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(trait),<span class="at">main=</span><span class="st">"ratxcan genes chr!=1"</span>,<span class="at">BH=</span><span class="cn">FALSE</span>,<span class="at">CI=</span><span class="cn">FALSE</span>,<span class="at">xlim=</span>rango,<span class="at">ylim=</span>rango)</span>
<span id="cb29-383"><a href="#cb29-383" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">1</span>))</span>
<span id="cb29-384"><a href="#cb29-384" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-385"><a href="#cb29-385" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-386"><a href="#cb29-386" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-387"><a href="#cb29-387" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-388"><a href="#cb29-388" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-389"><a href="#cb29-389" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-390"><a href="#cb29-390" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-391"><a href="#cb29-391" aria-hidden="true" tabindex="-1"></a>::: {.callout-warning}</span>
<span id="cb29-392"><a href="#cb29-392" aria-hidden="true" tabindex="-1"></a><span class="fu">## Take home</span></span>
<span id="cb29-393"><a href="#cb29-393" aria-hidden="true" tabindex="-1"></a>removing chr1 from GRM negates the correction of ratxcan in chr1</span>
<span id="cb29-394"><a href="#cb29-394" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb29-395"><a href="#cb29-395" aria-hidden="true" tabindex="-1"></a></span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->



</body></html>