<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.335">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Haky Im">
<meta name="dcterms.date" content="2022-06-10">

<title>Hakyimlab Notes – Transcriptome QGT Training 2022</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<script src="../../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../../">
<script src="../../../site_libs/quarto-html/quarto.js"></script>
<script src="../../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>

  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<link rel="stylesheet" href="../../../styles.css">
</head>

<body class="nav-fixed fullcontent">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../../../index.html">
    <span class="navbar-title">Hakyimlab Notes</span>
    </a>
  </div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../../about.html">
 <span class="menu-text">Miscel</span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/hakyimlab/"><i class="bi bi-github" role="img">
</i> 
 <span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://twitter.com/hakyim"><i class="bi bi-twitter" role="img">
</i> 
 <span class="menu-text"></span></a>
  </li>  
</ul>
              <div id="quarto-search" class="" title="Search"></div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">Transcriptome QGT Training 2022</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>
  <div class="quarto-categories">
    <div class="quarto-category">Lab</div>
  </div>
  </div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Haky Im </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">June 10, 2022</p>
    </div>
  </div>
  
    
  </div>
  

</header>

<section id="goals-of-lab" class="level1">
<h1>Goals of Lab</h1>
<ul>
<li>Predict whole blood expression</li>
<li>Check how well the prediction works with GEUVADIS expression data</li>
<li>Run association between predicted expression and a simulated phenotype</li>
<li>Calculate association between expression levels and coronary artery disease risk using s-predixcan</li>
<li>Fine-map the coronary artery disease gwas results using torus</li>
<li>Calculate colocalization probability using fastenloc</li>
<li>(skip) Run transcriptome-wide mendelian randomization in one locus of interest</li>
<li>Run cTWAS (fine-map SNPs and genes jointly)</li>
</ul>
</section>
<section id="inital-remarks" class="level1">
<h1>Inital Remarks</h1>
<ul>
<li><p>We ask you to actively participate in today’s hands on activities. <strong>Notice that we may ask you to share your screen for pedagogic purposes.</strong></p></li>
<li><p>If you have any concerns about this, please ask me or one of the TAs for assistance. We are here to help you learn.</p></li>
<li><p>After each section, we ask you to fill out the questionnaires</p></li>
<li><p>01 Preliminary https://forms.gle/FDwPcjLFynqQP7Fd8</p></li>
<li><p>02 Prediction https://forms.gle/CZYGyNdfNVH2W9QF7</p></li>
<li><p>03 PrediXcan https://forms.gle/Z8YdYjgYs2zQY5cP6</p></li>
<li><p>04 S-PrediXcan https://forms.gle/VVMYsP7u5Fh18fSY7</p></li>
<li><p>05 Colocalization https://forms.gle/8fiYqfiqv31XHsUz9</p></li>
<li><p>06 cTWAS https://forms.gle/ZpuEuv8Xmxw1eDPf6</p>
<ul>
<li>Name is a mandatory field. You can use your name or a pseudonymn. Just keep using the same name for each submission</li>
<li>We will ask you to submit partial set of answers so that we can get a sense of where people are</li>
</ul></li>
<li><p>Through out the lab you will see code chucks that are written with bash, eval=FALSE in place of R. These chunks are meant to be run in your Rstudio terminal and not in the Rmarkdown file itself.</p></li>
<li><p>R code chunks also have “eval=FALSE”. When you click on the green triangle, the code will run in the R console but not when we knit the document. This just happened to be more convenient for us to generate a knitted html to be shared <a href="https://lab-notes.hakyimlab.org/post/2022/06/10/qgt-training/">here</a>.</p></li>
</ul>
</section>
<section id="set-up" class="level1">
<h1>Set Up</h1>
<ul class="task-list">
<li><input type="checkbox" disabled="">Open the RStudio cloud project <a href="https://rstudio.cloud/spaces/232505/content/4013383">here</a></li>
<li><input type="checkbox" disabled="">Load Rstudio Libraries</li>
</ul>
<div class="cell">
<details>
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="do">## packages needed for susie+coloc</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(BEDMatrix)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Rfast)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(susieR)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(coloc)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="do">##library(tidyverse)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>-[ ] Navigate to starting directory</p>
<div class="cell">
<details>
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> <span class="st">"/cloud/project/"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<ul class="task-list">
<li><input type="checkbox" disabled="">activate the the imlabtools environment, which will make sure that the right version of python modules are available</li>
</ul>
<div class="cell">
<details>
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> activate imlabtools</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<ul class="task-list">
<li><input type="checkbox" disabled="">To define some variables to access the data more easily within the R session, run the following r chunk</li>
</ul>
<div class="cell">
<details>
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">getwd</span>())</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>lab<span class="ot">=</span><span class="st">"/cloud/project/QGT-Columbia-HKI-repo/"</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>CODE<span class="ot">=</span>glue<span class="sc">::</span><span class="fu">glue</span>(<span class="st">"{lab}/code"</span>)</span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(glue<span class="sc">::</span><span class="fu">glue</span>(<span class="st">"{CODE}/load_data_functions.R"</span>))</span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(glue<span class="sc">::</span><span class="fu">glue</span>(<span class="st">"{CODE}/plotting_utils_functions.R"</span>))</span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>PRE<span class="ot">=</span><span class="st">"/cloud/project/QGT-Columbia-HKI-repo/box_files"</span></span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a>MODEL<span class="ot">=</span>glue<span class="sc">::</span><span class="fu">glue</span>(<span class="st">"{PRE}/models"</span>)</span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a>DATA<span class="ot">=</span>glue<span class="sc">::</span><span class="fu">glue</span>(<span class="st">"{PRE}/data"</span>)</span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a>RESULTS<span class="ot">=</span>glue<span class="sc">::</span><span class="fu">glue</span>(<span class="st">"{PRE}/results"</span>)</span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a>METAXCAN<span class="ot">=</span>glue<span class="sc">::</span><span class="fu">glue</span>(<span class="st">"{PRE}/repos/MetaXcan-master/software"</span>)</span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a>FASTENLOC<span class="ot">=</span>glue<span class="sc">::</span><span class="fu">glue</span>(<span class="st">"{PRE}/repos/fastenloc-master"</span>)</span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a><span class="co"># This is a reference table we'll use a lot throughout the lab. It contains information about the genes.</span></span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a>gencode_df <span class="ot">=</span> <span class="fu">load_gencode_df</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<ul class="task-list">
<li><input type="checkbox" disabled="">check the values of the variables you just defined in R</li>
</ul>
<div class="cell">
<details>
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>MODEL</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>DATA</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<ul class="task-list">
<li><input type="checkbox" disabled="">define some variables to access the data more easily in the terminal. Remember we are running R code in the R console and command line code in the terminal.</li>
</ul>
<div class="cell">
<details>
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="bu">export</span> <span class="va">PRE</span><span class="op">=</span><span class="st">"/cloud/project/QGT-Columbia-HKI-repo/box_files"</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="bu">export</span> <span class="va">LAB</span><span class="op">=</span><span class="st">"/cloud/project/QGT-Columbia-HKI-repo/"</span></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a><span class="bu">export</span> <span class="va">CODE</span><span class="op">=</span><span class="va">$LAB</span>/code</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a><span class="bu">export</span> <span class="va">DATA</span><span class="op">=</span><span class="va">$PRE</span>/data</span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a><span class="bu">export</span> <span class="va">MODEL</span><span class="op">=</span><span class="va">$PRE</span>/models</span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a><span class="bu">export</span> <span class="va">RESULTS</span><span class="op">=</span><span class="va">$PRE</span>/results</span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a><span class="bu">export</span> <span class="va">METAXCAN</span><span class="op">=</span><span class="va">$PRE</span>/repos/MetaXcan-master/software</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<ul class="task-list">
<li><input type="checkbox" disabled="">check the values of the variables you just defined</li>
</ul>
<div class="cell">
<details>
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="va">$CODE</span></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="va">$RESULTS</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="questionnaire-01" class="level1">
<h1>Questionnaire 01</h1>
<p><a href="https://forms.gle/FDwPcjLFynqQP7Fd8">Fill out this form</a></p>
</section>
<section id="transcriptome-wide-association-review" class="level1">
<h1>Transcriptome-wide association (Review)</h1>
<p>Now we will perform a transcriptome-wide association analysis using the PrediXcan suite of tools. <img src="https://raw.githubusercontent.com/hakyimlab/QGT-Columbia-HKI/master/extras/figures/Association-Methods.png" class="img-fluid" alt="Transcriptome-wide association methods"></p>
<p>We start by predicting the expression levels of genes using the genotype data and the prediction weights and then perform an association between the predicted expression and the phenotype (denoted trait in the figure below).</p>
<p><img src="https://raw.githubusercontent.com/hakyimlab/QGT-Columbia-HKI/master/extras/figures/PrediXcan-run.png" class="img-fluid"></p>
</section>
<section id="predict-expression-from-genotype" class="level1">
<h1>Predict Expression from genotype</h1>
<p>In this section we will predict expression of genes in whole blood using the Predict.py code in the METAXCAN folder.</p>
<ul>
<li><p>Prediction models (weights) are located in the MODEL folder. Additional models for different tissues and transcriptome studies can be downloaded from <a href="http://predictdb.org">predictdb.org</a>.</p></li>
<li><p>This run should take about one minute.</p></li>
<li><p><input type="checkbox" disabled="">run the following code in the terminal.</p></li>
</ul>
<div class="cell">
<details>
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="bu">printf</span> <span class="st">"Predict expression\n\n"</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a><span class="ex">python3</span> <span class="va">$METAXCAN</span>/Predict.py <span class="dt">\</span></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>--model_db_path <span class="va">$PRE</span>/models/gtex_v8_en/en_Whole_Blood.db <span class="dt">\</span></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>--vcf_genotypes <span class="va">$DATA</span>/predixcan/genotype/filtered.vcf.gz <span class="dt">\</span></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>--vcf_mode genotyped <span class="dt">\</span></span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>--variant_mapping <span class="va">$DATA</span>/predixcan/gtex_v8_eur_filtered_maf0.01_monoallelic_variants.txt.gz id rsid <span class="dt">\</span></span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>--on_the_fly_mapping METADATA <span class="st">"chr{}_{}_{}_{}_b38"</span> <span class="dt">\</span></span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a>--prediction_output <span class="va">$RESULTS</span>/predixcan/Whole_Blood__predict.txt <span class="dt">\</span></span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a>--prediction_summary_output <span class="va">$RESULTS</span>/predixcan/Whole_Blood__summary.txt <span class="dt">\</span></span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a>--verbosity 9 <span class="dt">\</span></span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a>--throw</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<blockquote class="blockquote">
<p>Note we are only predicting chromosome 22 here (check by running “predicted_expression %&gt;% count(chromosome)” in the console)</p>
</blockquote>
<ul class="task-list">
<li><input type="checkbox" disabled="">run following code in the console to get information on reported prediction performance.</li>
</ul>
<blockquote class="blockquote">
<p>Find additional information <a href="https://github.com/hakyimlab/MetaXcan/wiki/Individual-level-PrediXcan:-introduction,-tutorials-and-manual">in this wiki</a></p>
</blockquote>
<div class="cell">
<details>
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>prediction_fp <span class="ot">=</span> glue<span class="sc">::</span><span class="fu">glue</span>(<span class="st">"{RESULTS}/predixcan/Whole_Blood__predict.txt"</span>)</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a><span class="do">## Read the Predict.py output into a dataframe. This function reorganizes the data and adds gene names.</span></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>predicted_expression <span class="ot">=</span> <span class="fu">load_predicted_expression</span>(prediction_fp, gencode_df)</span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(predicted_expression)</span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a><span class="do">## read summary of prediction, number of SNPs per gene, cross validated prediction performance</span></span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a>prediction_summary <span class="ot">=</span> <span class="fu">load_prediction_summary</span>(glue<span class="sc">::</span><span class="fu">glue</span>(<span class="st">"{RESULTS}/predixcan/Whole_Blood__summary.txt"</span>), gencode_df)</span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a><span class="do">## number of genes with a prediction model</span></span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(prediction_summary)</span>
<span id="cb9-12"><a href="#cb9-12" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(prediction_summary)</span>
<span id="cb9-13"><a href="#cb9-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-14"><a href="#cb9-14" aria-hidden="true" tabindex="-1"></a><span class="do">## how many unique genes were predicted?</span></span>
<span id="cb9-15"><a href="#cb9-15" aria-hidden="true" tabindex="-1"></a>predicted_expression <span class="sc">%&gt;%</span> .[[<span class="st">"gene_id"</span>]] <span class="sc">%&gt;%</span> <span class="fu">unique</span>() <span class="sc">%&gt;%</span> <span class="fu">length</span>()</span>
<span id="cb9-16"><a href="#cb9-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-17"><a href="#cb9-17" aria-hidden="true" tabindex="-1"></a><span class="do">## gene expression were predicted for how many people?</span></span>
<span id="cb9-18"><a href="#cb9-18" aria-hidden="true" tabindex="-1"></a>predicted_expression <span class="sc">%&gt;%</span> .[[<span class="st">"IID"</span>]] <span class="sc">%&gt;%</span> <span class="fu">unique</span>() <span class="sc">%&gt;%</span> <span class="fu">length</span>()</span>
<span id="cb9-19"><a href="#cb9-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-20"><a href="#cb9-20" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="st">"distribution of prediction performance r2"</span>)</span>
<span id="cb9-21"><a href="#cb9-21" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(prediction_summary<span class="sc">$</span>pred_perf_r2)</span>
<span id="cb9-22"><a href="#cb9-22" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(prediction_summary<span class="sc">$</span>pred_perf_r2)</span>
<span id="cb9-23"><a href="#cb9-23" aria-hidden="true" tabindex="-1"></a><span class="do">## Note: this is what the prediction trainer reported as prediction performance</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="optional-assess-actual-prediction-performance" class="level2">
<h2 class="anchored" data-anchor-id="optional-assess-actual-prediction-performance">(Optional) Assess Actual Prediction Performance</h2>
<div class="cell">
<details>
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="do">## download and read observed expression data from GEUVADIS </span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="do">## from https://uchicago.box.com/s/4y7xle5l0pnq9d1fwmthe2ewhogrnlrv</span></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>obs_exp<span class="ot">&lt;-</span> <span class="fu">read_csv</span>(glue<span class="sc">::</span><span class="fu">glue</span>(<span class="st">"{DATA}/predixcan/GEUVADIS.observed_df.csv.gz"</span>))</span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a><span class="do">## Note that the version of the ensemble id of the gene was removed</span></span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(predicted_expression)</span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a><span class="do">## Q: how many genes were predicted?</span></span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">unique</span>(predicted_expression<span class="sc">$</span>gene_id))</span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-12"><a href="#cb10-12" aria-hidden="true" tabindex="-1"></a><span class="do">## inner join predicted expression with observed expression data (by IID and gene)</span></span>
<span id="cb10-13"><a href="#cb10-13" aria-hidden="true" tabindex="-1"></a><span class="do">## common errors occur when ensemble id's have versions in one set and not the other set</span></span>
<span id="cb10-14"><a href="#cb10-14" aria-hidden="true" tabindex="-1"></a>fullset<span class="ot">=</span><span class="fu">inner_join</span>(predicted_expression, obs_exp, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"gene_id"</span>,<span class="st">"IID"</span>))</span>
<span id="cb10-15"><a href="#cb10-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-16"><a href="#cb10-16" aria-hidden="true" tabindex="-1"></a><span class="do">## calculate spearman correlation for all genes</span></span>
<span id="cb10-17"><a href="#cb10-17" aria-hidden="true" tabindex="-1"></a>genelist <span class="ot">=</span> <span class="fu">unique</span>(predicted_expression<span class="sc">$</span>gene_id)</span>
<span id="cb10-18"><a href="#cb10-18" aria-hidden="true" tabindex="-1"></a>corvec <span class="ot">=</span> <span class="fu">rep</span>(<span class="cn">NA</span>,<span class="fu">length</span>(genelist))</span>
<span id="cb10-19"><a href="#cb10-19" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(corvec) <span class="ot">=</span> genelist</span>
<span id="cb10-20"><a href="#cb10-20" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(gg <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(genelist))</span>
<span id="cb10-21"><a href="#cb10-21" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb10-22"><a href="#cb10-22" aria-hidden="true" tabindex="-1"></a>  ind <span class="ot">=</span> fullset<span class="sc">$</span>gene_id<span class="sc">==</span>genelist[gg]</span>
<span id="cb10-23"><a href="#cb10-23" aria-hidden="true" tabindex="-1"></a>  corvec[gg] <span class="ot">=</span> <span class="fu">cor</span>(fullset<span class="sc">$</span>predicted_expression[ind], fullset<span class="sc">$</span>observed_expression[ind])</span>
<span id="cb10-24"><a href="#cb10-24" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb10-25"><a href="#cb10-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-26"><a href="#cb10-26" aria-hidden="true" tabindex="-1"></a><span class="do">## what's the best performing gene?</span></span>
<span id="cb10-27"><a href="#cb10-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-28"><a href="#cb10-28" aria-hidden="true" tabindex="-1"></a><span class="do">## plot the histogram of the prediction performance</span></span>
<span id="cb10-29"><a href="#cb10-29" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(corvec)</span>
<span id="cb10-30"><a href="#cb10-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-31"><a href="#cb10-31" aria-hidden="true" tabindex="-1"></a><span class="do">## list the top 10 best performing genes</span></span>
<span id="cb10-32"><a href="#cb10-32" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">sort</span>(corvec,<span class="at">decreasing =</span> <span class="cn">TRUE</span>),<span class="dv">2</span>)</span>
<span id="cb10-33"><a href="#cb10-33" aria-hidden="true" tabindex="-1"></a><span class="fu">tail</span>(<span class="fu">sort</span>(corvec,<span class="at">decreasing =</span> <span class="cn">TRUE</span>),<span class="dv">2</span>)</span>
<span id="cb10-34"><a href="#cb10-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-35"><a href="#cb10-35" aria-hidden="true" tabindex="-1"></a><span class="do">## plot the correlation of the top 2 best performing genes bottom 2</span></span>
<span id="cb10-36"><a href="#cb10-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-37"><a href="#cb10-37" aria-hidden="true" tabindex="-1"></a>geneid <span class="ot">=</span> <span class="st">"ENSG00000100376"</span></span>
<span id="cb10-38"><a href="#cb10-38" aria-hidden="true" tabindex="-1"></a>genename <span class="ot">=</span> gencode_df <span class="sc">%&gt;%</span> <span class="fu">filter</span>(gene_id<span class="sc">==</span>geneid) <span class="sc">%&gt;%</span> .[[<span class="st">"gene_name"</span>]]</span>
<span id="cb10-39"><a href="#cb10-39" aria-hidden="true" tabindex="-1"></a>fullset <span class="sc">%&gt;%</span> <span class="fu">filter</span>(gene_id<span class="sc">==</span>geneid) <span class="sc">%&gt;%</span> <span class="fu">ggplot</span>(<span class="fu">aes</span>(observed_expression, predicted_expression))<span class="sc">+</span><span class="fu">geom_point</span>()<span class="sc">+</span><span class="fu">ggtitle</span>(<span class="fu">paste</span>(genename, <span class="st">"-"</span>,geneid))</span>
<span id="cb10-40"><a href="#cb10-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-41"><a href="#cb10-41" aria-hidden="true" tabindex="-1"></a>geneid <span class="ot">=</span> <span class="st">"ENSG00000075234"</span></span>
<span id="cb10-42"><a href="#cb10-42" aria-hidden="true" tabindex="-1"></a>genename <span class="ot">=</span> gencode_df <span class="sc">%&gt;%</span> <span class="fu">filter</span>(gene_id<span class="sc">==</span>geneid) <span class="sc">%&gt;%</span> .[[<span class="st">"gene_name"</span>]]</span>
<span id="cb10-43"><a href="#cb10-43" aria-hidden="true" tabindex="-1"></a>fullset <span class="sc">%&gt;%</span> <span class="fu">filter</span>(gene_id<span class="sc">==</span>geneid) <span class="sc">%&gt;%</span> <span class="fu">ggplot</span>(<span class="fu">aes</span>(observed_expression, predicted_expression))<span class="sc">+</span><span class="fu">geom_point</span>()<span class="sc">+</span><span class="fu">ggtitle</span>(<span class="fu">paste</span>(genename, <span class="st">"-"</span>, geneid))</span>
<span id="cb10-44"><a href="#cb10-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-45"><a href="#cb10-45" aria-hidden="true" tabindex="-1"></a>geneid <span class="ot">=</span> <span class="st">"ENSG00000070371"</span></span>
<span id="cb10-46"><a href="#cb10-46" aria-hidden="true" tabindex="-1"></a>genename <span class="ot">=</span> gencode_df <span class="sc">%&gt;%</span> <span class="fu">filter</span>(gene_id<span class="sc">==</span>geneid) <span class="sc">%&gt;%</span> .[[<span class="st">"gene_name"</span>]]</span>
<span id="cb10-47"><a href="#cb10-47" aria-hidden="true" tabindex="-1"></a>fullset <span class="sc">%&gt;%</span> <span class="fu">filter</span>(gene_id<span class="sc">==</span>geneid) <span class="sc">%&gt;%</span> <span class="fu">ggplot</span>(<span class="fu">aes</span>(observed_expression, predicted_expression))<span class="sc">+</span><span class="fu">geom_point</span>()<span class="sc">+</span><span class="fu">ggtitle</span>(<span class="fu">paste</span>(genename, <span class="st">"-"</span>, geneid))</span>
<span id="cb10-48"><a href="#cb10-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-49"><a href="#cb10-49" aria-hidden="true" tabindex="-1"></a>geneid <span class="ot">=</span> <span class="st">"ENSG00000184164"</span></span>
<span id="cb10-50"><a href="#cb10-50" aria-hidden="true" tabindex="-1"></a>genename <span class="ot">=</span> gencode_df <span class="sc">%&gt;%</span> <span class="fu">filter</span>(gene_id<span class="sc">==</span>geneid) <span class="sc">%&gt;%</span> .[[<span class="st">"gene_name"</span>]]</span>
<span id="cb10-51"><a href="#cb10-51" aria-hidden="true" tabindex="-1"></a>fullset <span class="sc">%&gt;%</span> <span class="fu">filter</span>(gene_id<span class="sc">==</span>geneid) <span class="sc">%&gt;%</span> <span class="fu">ggplot</span>(<span class="fu">aes</span>(observed_expression, predicted_expression))<span class="sc">+</span><span class="fu">geom_point</span>()<span class="sc">+</span><span class="fu">ggtitle</span>(<span class="fu">paste</span>(genename, <span class="st">"-"</span>, geneid))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</section>
<section id="questionnaire-02" class="level1">
<h1>Questionnaire 02</h1>
<p>Prediction <a href="https://forms.gle/CZYGyNdfNVH2W9QF7">Fill out this form</a></p>
</section>
<section id="run-predixcan-association" class="level1">
<h1>Run PrediXcan association</h1>
<p>We are going to use a simulated phenotype for which only UPK3A has an effect on the phenotype (<span class="math inline">\(\beta=-0.9887378\)</span>)</p>
<p><span class="math inline">\(Y = \sum_k T_k \beta_k + \epsilon\)</span></p>
<p>with random effects <span class="math inline">\(\beta_k \sim (1-\pi)\cdot \delta_0 + \pi\cdot N(0,1)\)</span></p>
<div class="cell">
<details>
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a><span class="bu">export</span> <span class="va">PHENO</span><span class="op">=</span><span class="st">"sim.spike_n_slab_0.01_pve0.1"</span></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a><span class="bu">printf</span> <span class="st">"association\n\n"</span></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a><span class="ex">python3</span> <span class="va">$METAXCAN</span>/PrediXcanAssociation.py <span class="dt">\</span></span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>--expression_file <span class="va">$RESULTS</span>/predixcan/Whole_Blood__predict.txt <span class="dt">\</span></span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a>--input_phenos_file <span class="va">$DATA</span>/predixcan/phenotype/<span class="va">$PHENO</span>.txt <span class="dt">\</span></span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a>--input_phenos_column pheno <span class="dt">\</span></span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a>--output <span class="va">$RESULTS</span>/predixcan/<span class="va">$PHENO</span>/Whole_Blood__association.txt <span class="dt">\</span></span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a>--verbosity 9 <span class="dt">\</span></span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a>--throw</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>More predicted phenotypes can be found in $DATA/predixcan/phenotype/. The naming of the phenotypes provides information about the genetic architecture: the number after pve is the proportion of variance of Y explained by the genetic component of expression. The number after spike_n_slab represents the probability that a gene is causal π (i.e.&nbsp;prob β≠0)</p>
<section id="looking-at-association-results" class="level2">
<h2 class="anchored" data-anchor-id="looking-at-association-results">Looking at Association Results</h2>
<div class="cell">
<details>
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="do">## read association results</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>PHENO<span class="ot">=</span><span class="st">"sim.spike_n_slab_0.01_pve0.1"</span></span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>predixcan_association <span class="ot">=</span> <span class="fu">load_predixcan_association</span>(glue<span class="sc">::</span><span class="fu">glue</span>(<span class="st">"{RESULTS}/predixcan/{PHENO}/Whole_Blood__association.txt"</span>), gencode_df)</span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a><span class="do">## take a look at the results</span></span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(predixcan_association)</span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a>predixcan_association <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(pvalue) <span class="sc">%&gt;%</span> <span class="fu">select</span>(gene_name,effect,se,pvalue,gene) <span class="sc">%&gt;%</span> head</span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a>predixcan_association <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(pvalue) <span class="sc">%&gt;%</span> <span class="fu">ggplot</span>(<span class="fu">aes</span>(pvalue)) <span class="sc">+</span> <span class="fu">geom_histogram</span>(<span class="at">bins=</span><span class="dv">10</span>)</span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a><span class="do">## compare distribution against the null (uniform)</span></span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a><span class="fu">gg_qqplot</span>(predixcan_association<span class="sc">$</span>pvalue, <span class="at">max_yval =</span> <span class="dv">40</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="comparing-the-estimated-effect-size-with-true-effect-size" class="level2">
<h2 class="anchored" data-anchor-id="comparing-the-estimated-effect-size-with-true-effect-size">Comparing the Estimated Effect Size with True Effect Size</h2>
<div class="cell">
<details>
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>truebetas <span class="ot">=</span> <span class="fu">load_truebetas</span>(glue<span class="sc">::</span><span class="fu">glue</span>(<span class="st">"{DATA}/predixcan/phenotype/gene-effects/{PHENO}.txt"</span>), gencode_df)</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>betas <span class="ot">=</span> (predixcan_association <span class="sc">%&gt;%</span> </span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>               <span class="fu">inner_join</span>(truebetas,<span class="at">by=</span><span class="fu">c</span>(<span class="st">"gene"</span><span class="ot">=</span><span class="st">"gene_id"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>               <span class="fu">select</span>(<span class="fu">c</span>(<span class="st">'estimated_beta'</span><span class="ot">=</span><span class="st">'effect'</span>, </span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a>                        <span class="st">'true_beta'</span><span class="ot">=</span><span class="st">'effect_size'</span>,</span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a>                        <span class="st">'pvalue'</span>, </span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a>                        <span class="st">'gene_id'</span><span class="ot">=</span><span class="st">'gene'</span>, </span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a>                        <span class="st">'gene_name'</span><span class="ot">=</span><span class="st">'gene_name.x'</span>, </span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a>                        <span class="st">'region_id'</span><span class="ot">=</span><span class="st">'region_id.x'</span>)))</span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a>betas <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(pvalue) <span class="sc">%&gt;%</span> <span class="fu">select</span>(gene_name,estimated_beta,true_beta,pvalue) <span class="sc">%&gt;%</span> head</span>
<span id="cb13-11"><a href="#cb13-11" aria-hidden="true" tabindex="-1"></a><span class="do">## do you see examples of potential LD contamination?</span></span>
<span id="cb13-12"><a href="#cb13-12" aria-hidden="true" tabindex="-1"></a>betas <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">causal=</span> true_beta<span class="sc">!=</span><span class="dv">0</span>) <span class="sc">%&gt;%</span> <span class="fu">ggplot</span>(<span class="fu">aes</span>(estimated_beta, true_beta,<span class="at">col=</span>causal))<span class="sc">+</span><span class="fu">geom_point</span>(<span class="at">alpha=</span><span class="fl">0.6</span>,<span class="at">size=</span><span class="dv">5</span>)<span class="sc">+</span><span class="fu">geom_abline</span>()<span class="sc">+</span><span class="fu">theme_bw</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<blockquote class="blockquote">
<p>UPK3A is the causal gene and has the most significant pvalue. RIBC2 is also significantly associated but has no causal role (we know because we simulated the phenotype that way). Why?</p>
</blockquote>
<p>Hint: correlation between the genes</p>
</section>
</section>
<section id="questionnaire-03" class="level1">
<h1>Questionnaire 03</h1>
<p>PrediXcan Association [Fill out this form] (https://forms.gle/Z8YdYjgYs2zQY5cP6)</p>
</section>
<section id="run-summary-predixcan" class="level1">
<h1>Run Summary PrediXcan</h1>
<p>Now we will use the summary results from a GWAS of coronary artery disease to calculate the association between the genetic component of the expression of genes and coronary artery disease risk. We will use the SPrediXcan.py.</p>
<p><img src="https://uchicago.box.com/shared/static/m45nnkeskzh88ifnv8td5e3fntdfzhjs.png" class="img-fluid"></p>
<p>The GWAS results (harmonized and imputed) for coronary artery disease are available in $PRE/spredixcan/data/ ## Running S-PrediXcan</p>
<div class="cell">
<details>
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a><span class="ex">python</span> <span class="va">$METAXCAN</span>/SPrediXcan.py <span class="dt">\</span></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>--gwas_file  <span class="va">$DATA</span>/spredixcan/imputed_CARDIoGRAM_C4D_CAD_ADDITIVE.txt.gz <span class="dt">\</span></span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>--snp_column panel_variant_id <span class="dt">\</span></span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>--effect_allele_column effect_allele <span class="dt">\</span></span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>--non_effect_allele_column non_effect_allele <span class="dt">\</span></span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a>--zscore_column zscore <span class="dt">\</span></span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a>--model_db_path <span class="va">$MODEL</span>/gtex_v8_mashr/mashr_Whole_Blood.db <span class="dt">\</span></span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a>--covariance <span class="va">$MODEL</span>/gtex_v8_mashr/mashr_Whole_Blood.txt.gz <span class="dt">\</span></span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a>--keep_non_rsid <span class="dt">\</span></span>
<span id="cb14-11"><a href="#cb14-11" aria-hidden="true" tabindex="-1"></a>--additional_output <span class="dt">\</span></span>
<span id="cb14-12"><a href="#cb14-12" aria-hidden="true" tabindex="-1"></a>--model_db_snp_key varID <span class="dt">\</span></span>
<span id="cb14-13"><a href="#cb14-13" aria-hidden="true" tabindex="-1"></a>--throw <span class="dt">\</span></span>
<span id="cb14-14"><a href="#cb14-14" aria-hidden="true" tabindex="-1"></a>--output_file <span class="va">$RESULTS</span>/spredixcan/eqtl/CARDIoGRAM_C4D_CAD_ADDITIVE__PM__Whole_Blood.csv</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<blockquote class="blockquote">
<p>We can run the full genome because the summary statistics based PrediXcan is much faster than individual level one.</p>
</blockquote>
<section id="plot-and-interpret-results" class="level2">
<h2 class="anchored" data-anchor-id="plot-and-interpret-results">Plot and Interpret Results</h2>
<div class="cell">
<details>
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>spredixcan_association <span class="ot">=</span> <span class="fu">load_spredixcan_association</span>(glue<span class="sc">::</span><span class="fu">glue</span>(<span class="st">"{RESULTS}/spredixcan/eqtl/CARDIoGRAM_C4D_CAD_ADDITIVE__PM__Whole_Blood.csv"</span>), gencode_df)</span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(spredixcan_association)</span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>spredixcan_association <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(pvalue) <span class="sc">%&gt;%</span> head</span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>spredixcan_association <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(pvalue) <span class="sc">%&gt;%</span> <span class="fu">ggplot</span>(<span class="fu">aes</span>(pvalue)) <span class="sc">+</span> <span class="fu">geom_histogram</span>(<span class="at">bins=</span><span class="dv">20</span>)</span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a><span class="fu">gg_qqplot</span>(spredixcan_association<span class="sc">$</span>pvalue)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<ul>
<li><p><em>Question:</em> SORT1, considered to be a causal gene for LDL cholesterol and as a consequence of coronary artery disease, is not found here. Why?</p></li>
<li><p><input type="checkbox" disabled="">check whether SORT1 is expressed in whole blood <a href="https://gtexportal.org/home/gene/SORT1">GTEx portal</a></p></li>
<li><p><input type="checkbox" disabled="">check whether SORT1 has eQTL in whole blood <a href="https://gtexportal.org/home/gene/SORT1">GTEx portal</a></p></li>
</ul>
</section>
<section id="run-s-predixcan-using-gene-expression-predicted-in-liver" class="level2">
<h2 class="anchored" data-anchor-id="run-s-predixcan-using-gene-expression-predicted-in-liver">Run S-PrediXcan using gene expression predicted in liver</h2>
<p>-[] Run s-predixcan with liver model, do you find SORT1? Is it significant?</p>
<div class="cell">
<details>
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="co">#loction Liver models </span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a><span class="co">#/cloud/project/QGT-Columbia-HKI-repo/box_files/models/gtex_v8_mashr/</span></span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a><span class="ex">python</span> <span class="va">$METAXCAN</span>/SPrediXcan.py <span class="dt">\</span></span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>--gwas_file  <span class="va">$DATA</span>/spredixcan/imputed_CARDIoGRAM_C4D_CAD_ADDITIVE.txt.gz <span class="dt">\</span></span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>--snp_column panel_variant_id <span class="dt">\</span></span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>--effect_allele_column effect_allele <span class="dt">\</span></span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a>--non_effect_allele_column non_effect_allele <span class="dt">\</span></span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a>--zscore_column zscore <span class="dt">\</span></span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a>--model_db_path <span class="va">$MODEL</span>/gtex_v8_mashr/mashr_Liver.db <span class="dt">\</span></span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a>--covariance <span class="va">$MODEL</span>/gtex_v8_mashr/mashr_Liver.txt.gz <span class="dt">\</span></span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true" tabindex="-1"></a>--keep_non_rsid <span class="dt">\</span></span>
<span id="cb16-12"><a href="#cb16-12" aria-hidden="true" tabindex="-1"></a>--additional_output <span class="dt">\</span></span>
<span id="cb16-13"><a href="#cb16-13" aria-hidden="true" tabindex="-1"></a>--model_db_snp_key varID <span class="dt">\</span></span>
<span id="cb16-14"><a href="#cb16-14" aria-hidden="true" tabindex="-1"></a>--throw <span class="dt">\</span></span>
<span id="cb16-15"><a href="#cb16-15" aria-hidden="true" tabindex="-1"></a>--output_file <span class="va">$RESULTS</span>/spredixcan/eqtl/CARDIoGRAM_C4D_CAD_ADDITIVE__PM__Liver.csv</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>spredixcan_association_L<span class="ot">=</span> <span class="fu">load_spredixcan_association</span>(glue<span class="sc">::</span><span class="fu">glue</span>(<span class="st">"{RESULTS}/spredixcan/eqtl/CARDIoGRAM_C4D_CAD_ADDITIVE__PM__Liver.csv"</span>), gencode_df)</span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(spredixcan_association_L)</span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>spredixcan_association_L <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(pvalue) <span class="sc">%&gt;%</span> head</span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>spredixcan_association_L <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(pvalue) <span class="sc">%&gt;%</span> <span class="fu">ggplot</span>(<span class="fu">aes</span>(pvalue)) <span class="sc">+</span> <span class="fu">geom_histogram</span>(<span class="at">bins=</span><span class="dv">20</span>)</span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a><span class="fu">gg_qqplot</span>(spredixcan_association_L<span class="sc">$</span>pvalue)</span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a>col_order<span class="ot">=</span> <span class="fu">c</span>(<span class="st">"gene_name"</span>,<span class="st">"gene"</span>,<span class="st">"zscore"</span>,<span class="st">"effect_size"</span>,<span class="st">"pvalue"</span>,<span class="st">"var_g"</span>,<span class="st">"pred_perf_r2"</span>, <span class="st">"pred_perf_pval"</span>,<span class="st">"pred_perf_qval"</span>, <span class="st">"n_snps_used"</span>, <span class="st">"n_snps_in_cov"</span>, <span class="st">"n_snps_in_model"</span>,<span class="st">"best_gwas_p"</span>,<span class="st">"largest_weight"</span>)</span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a>spredixcan_association_L <span class="ot">&lt;-</span> spredixcan_association_L[, col_order]</span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a><span class="fu">filter</span>(spredixcan_association_L, gene_name<span class="sc">==</span><span class="st">"SORT1"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="optional-compare-zscores-in-liver-and-whole-blood." class="level2">
<h2 class="anchored" data-anchor-id="optional-compare-zscores-in-liver-and-whole-blood.">(Optional) Compare zscores in liver and whole blood.</h2>
<blockquote class="blockquote">
<p>Recall that zscore is the effect size divided by the standard error</p>
</blockquote>
<div class="cell">
<details>
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>spredixcan_association_L<span class="ot">=</span><span class="fu">rename</span>(spredixcan_association_L, <span class="at">zscore_liver =</span> <span class="st">"zscore"</span>)</span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(spredixcan_association_L)</span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>test<span class="ot">=</span><span class="fu">left_join</span>(spredixcan_association, spredixcan_association_L, <span class="at">by=</span><span class="st">"gene_name"</span>)</span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>test<span class="ot">=</span><span class="fu">select</span>(test,<span class="st">"gene_name"</span>,<span class="st">"zscore"</span>,<span class="st">"zscore_liver"</span>)</span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>test <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(zscore_liver) <span class="sc">%&gt;%</span> head</span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a>test <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">zscore_WB=</span>zscore) <span class="sc">%&gt;%</span> <span class="fu">ggplot</span>(<span class="fu">aes</span>(zscore_WB,zscore_liver)) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">size=</span><span class="dv">3</span>,<span class="at">alpha=</span>.<span class="dv">6</span>) <span class="sc">+</span> <span class="fu">geom_abline</span>()</span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a><span class="do">## S-PrediXcan association in liver and whole blood are significantly correlated</span></span>
<span id="cb18-10"><a href="#cb18-10" aria-hidden="true" tabindex="-1"></a><span class="fu">cor.test</span>(test<span class="sc">$</span>zscore,test<span class="sc">$</span>zscore_liver)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="optional-run-multixcan" class="level2">
<h2 class="anchored" data-anchor-id="optional-run-multixcan">(Optional) Run MultiXcan</h2>
<ul>
<li><p>multixcan aggregates information across multiple tissues to boost the power to detect association. It was developed movivated by the fact that eQTLs are shared across multiple tissues, i.e.&nbsp;many genetic variants that regulate expression are common across tissues.</p></li>
<li><p>before you run multixcan ensure you have run s-predixcan for all the tissues you want to multixcan. In this tutorial we have two tissues (liver and whole blood), ensure you have run s-predixcan with the two tissues before running multixcan.</p></li>
<li><p>One thing to note is to ensure similar naming pattern for the output files. This is to ensure the files are captured correctly when running multixcan’s filter.</p></li>
</ul>
<div class="cell">
<details>
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a><span class="ex">python</span> <span class="va">$METAXCAN</span>/SMulTiXcan.py <span class="dt">\</span></span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>--models_folder <span class="va">$MODEL</span>/gtex_v8_mashr <span class="dt">\</span></span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>--models_name_pattern <span class="st">"mashr_(.*).db"</span> <span class="dt">\</span></span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a>--snp_covariance <span class="va">$MODEL</span>/gtex_v8_expression_mashr_snp_smultixcan_covariance.txt.gz <span class="dt">\</span></span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a>--metaxcan_folder <span class="va">$RESULTS</span>/spredixcan/eqtl/ <span class="dt">\</span></span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a>--metaxcan_filter <span class="st">"CARDIoGRAM_C4D_CAD_ADDITIVE__PM__(.*).csv"</span> <span class="dt">\</span></span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a>--metaxcan_file_name_parse_pattern <span class="st">"(.*)__PM__(.*).csv"</span> <span class="dt">\</span></span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a>--gwas_file <span class="va">$DATA</span>/spredixcan/imputed_CARDIoGRAM_C4D_CAD_ADDITIVE.txt.gz <span class="dt">\</span></span>
<span id="cb19-10"><a href="#cb19-10" aria-hidden="true" tabindex="-1"></a>--snp_column panel_variant_id <span class="at">--effect_allele_column</span> effect_allele <span class="at">--non_effect_allele_column</span> non_effect_allele <span class="at">--zscore_column</span> zscore <span class="at">--keep_non_rsid</span> <span class="at">--model_db_snp_key</span> varID <span class="dt">\</span></span>
<span id="cb19-11"><a href="#cb19-11" aria-hidden="true" tabindex="-1"></a>--cutoff_condition_number 30 <span class="dt">\</span></span>
<span id="cb19-12"><a href="#cb19-12" aria-hidden="true" tabindex="-1"></a>--verbosity 9 <span class="dt">\</span></span>
<span id="cb19-13"><a href="#cb19-13" aria-hidden="true" tabindex="-1"></a>--throw <span class="dt">\</span></span>
<span id="cb19-14"><a href="#cb19-14" aria-hidden="true" tabindex="-1"></a>--output <span class="va">$RESULTS</span>/smultixcan/eqtl/CARDIoGRAM_C4D_CAD_ADDITIVE_smultixcan.txt</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="questionnaire-04" class="level2">
<h2 class="anchored" data-anchor-id="questionnaire-04">Questionnaire 04</h2>
<p>04 S-PrediXcan <a href="https://forms.gle/VVMYsP7u5Fh18fSY7">Fill out this form</a></p>
</section>
</section>
<section id="colocalization-methods-review" class="level1">
<h1>Colocalization Methods (Review)</h1>
<p>Colocalization methods seek to estimate the probability that the complex trait and expression causal variants are the same. We favor methods that calculate the probability of causality for each trait (posterior inclusion probability), called fine-mapping methods. Here we use susie for fine-mapping and coloc for colocalization.</p>
<p><img src="https://raw.githubusercontent.com/hakyimlab/QGT-Columbia-HKI/master/extras/figures/colocalization-run.png" class="img-fluid"></p>
</section>
<section id="run-colocalization" class="level1">
<h1>Run colocalization</h1>
<p>When you use coloc on your own data, you may want to check out coloc’s documentation, with good advice and tips on avoiding common mistakes https://cran.r-project.org/web/packages/coloc/vignettes</p>
<p>Due to time constraints, we will run one region and one gene only</p>
<ol type="1">
<li>Finemap GWAS of CAD</li>
<li>Finemap eQTL of SORT1</li>
</ol>
<p>For finemapping, we need the summary statistics (effect size, standard errors, etc) and the correlation between SNPs (LD matrix)</p>
<section id="load-the-genotype-to-calculate-the-ld-matrix" class="level2">
<h2 class="anchored" data-anchor-id="load-the-genotype-to-calculate-the-ld-matrix">load the genotype to calculate the ld matrix</h2>
<div class="cell">
<details>
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="co"># load the genotype to calculate the ld matrix</span></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>X_mat <span class="ot">&lt;-</span> <span class="fu">BEDMatrix</span>(glue<span class="sc">::</span><span class="fu">glue</span>(<span class="st">"{DATA}/colocalization/geuvadis_chr1"</span>))</span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(X_mat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">_.*"</span>, <span class="st">""</span>,<span class="fu">colnames</span>(X_mat))</span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(X_mat) <span class="ot">&lt;-</span> <span class="fu">str_replace_all</span>(<span class="fu">colnames</span>(X_mat),<span class="st">":"</span>,<span class="st">"_"</span>)</span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a>snp_info <span class="ot">&lt;-</span> <span class="fu">fread</span>(glue<span class="sc">::</span><span class="fu">glue</span>(<span class="st">"{DATA}/colocalization/geuvadis_chr1.bim"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">setnames</span>(., <span class="fu">colnames</span>(.), <span class="fu">c</span>(<span class="st">"chr"</span>, <span class="st">"snp"</span>, <span class="st">"cm"</span>, <span class="st">"pos"</span>, <span class="st">"alt"</span>, <span class="st">"ref"</span>)) </span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-8"><a href="#cb20-8" aria-hidden="true" tabindex="-1"></a>snp_info<span class="sc">$</span>snp <span class="ot">&lt;-</span> <span class="fu">str_replace_all</span>(snp_info<span class="sc">$</span>snp,<span class="st">":"</span>,<span class="st">"_"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="load-the-eqtl-and-gwas-effect-size-files" class="level2">
<h2 class="anchored" data-anchor-id="load-the-eqtl-and-gwas-effect-size-files">Load the eqtl and gwas effect size files</h2>
<div class="cell">
<details>
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="co"># load the eqtl effect sizes</span></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>gene_ss <span class="ot">&lt;-</span> <span class="fu">fread</span>(glue<span class="sc">::</span><span class="fu">glue</span>(<span class="st">"{DATA}/colocalization/Liver_chr1.txt"</span>))</span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a><span class="co"># load gwas effect sizes</span></span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>gwas <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(glue<span class="sc">::</span><span class="fu">glue</span>(<span class="st">"{DATA}/spredixcan/imputed_CARDIoGRAM_C4D_CAD_ADDITIVE.txt.gz"</span>))</span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a><span class="co"># filter to select genome wide significant snps at 5 × 10−8</span></span>
<span id="cb21-8"><a href="#cb21-8" aria-hidden="true" tabindex="-1"></a>filtered_regions <span class="ot">&lt;-</span> gwas <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(pvalue <span class="sc">&lt;</span> <span class="fl">5e-8</span>)</span>
<span id="cb21-9"><a href="#cb21-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-10"><a href="#cb21-10" aria-hidden="true" tabindex="-1"></a><span class="co"># load the ld block</span></span>
<span id="cb21-11"><a href="#cb21-11" aria-hidden="true" tabindex="-1"></a>ldblocks <span class="ot">&lt;-</span><span class="fu">read_tsv</span>(glue<span class="sc">::</span><span class="fu">glue</span>(<span class="st">"{DATA}/spredixcan/eur_ld.hg38.txt.gz"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="find-regions-with-the-strongest-signal-in-the-gwas" class="level2">
<h2 class="anchored" data-anchor-id="find-regions-with-the-strongest-signal-in-the-gwas">Find regions with the strongest signal in the gwas</h2>
<div class="cell">
<details>
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="co"># get the loci where the significant snps are located</span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (n <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(filtered_regions)){</span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>  <span class="co"># extract genename, start and end</span></span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a>  variant_id <span class="ot">&lt;-</span> <span class="fu">as.character</span>(filtered_regions[n,<span class="st">"variant_id"</span>])</span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a>  variant_chr <span class="ot">&lt;-</span> <span class="fu">as.character</span>(filtered_regions[n,<span class="st">"chromosome"</span>])</span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a>  variant_pos <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(filtered_regions[n,<span class="st">"position"</span>])</span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a>  <span class="co">#gene_end &lt;- as.numeric(genes[n,"end"])</span></span>
<span id="cb22-8"><a href="#cb22-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-9"><a href="#cb22-9" aria-hidden="true" tabindex="-1"></a>  locus <span class="ot">&lt;-</span> ldblocks <span class="sc">%&gt;%</span></span>
<span id="cb22-10"><a href="#cb22-10" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(chr <span class="sc">==</span> variant_chr) <span class="sc">%&gt;%</span></span>
<span id="cb22-11"><a href="#cb22-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(variant_pos <span class="sc">&gt;=</span> start <span class="sc">&amp;</span> variant_pos <span class="sc">&lt;</span> stop) <span class="sc">%&gt;%</span></span>
<span id="cb22-12"><a href="#cb22-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">locus_name =</span> <span class="fu">paste0</span>(chr,<span class="st">"_"</span>,start,<span class="st">"_"</span>,stop)) <span class="sc">%&gt;%</span></span>
<span id="cb22-13"><a href="#cb22-13" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">locus_start=</span>start,<span class="at">locus_end=</span>stop) <span class="sc">%&gt;%</span></span>
<span id="cb22-14"><a href="#cb22-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">variant_id =</span> variant_id, <span class="at">position=</span>variant_pos)</span>
<span id="cb22-15"><a href="#cb22-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-16"><a href="#cb22-16" aria-hidden="true" tabindex="-1"></a>  <span class="co"># create a data frame with info</span></span>
<span id="cb22-17"><a href="#cb22-17" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">exists</span>(<span class="st">'all_loci'</span>) <span class="sc">&amp;&amp;</span> <span class="fu">is.data.frame</span>(<span class="fu">get</span>(<span class="st">'all_loci'</span>))) {</span>
<span id="cb22-18"><a href="#cb22-18" aria-hidden="true" tabindex="-1"></a>    all_loci <span class="ot">&lt;-</span> <span class="fu">rbind</span>(all_loci,locus)</span>
<span id="cb22-19"><a href="#cb22-19" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb22-20"><a href="#cb22-20" aria-hidden="true" tabindex="-1"></a>    all_loci <span class="ot">&lt;-</span> locus</span>
<span id="cb22-21"><a href="#cb22-21" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb22-22"><a href="#cb22-22" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb22-23"><a href="#cb22-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-24"><a href="#cb22-24" aria-hidden="true" tabindex="-1"></a><span class="co"># select uniq loci</span></span>
<span id="cb22-25"><a href="#cb22-25" aria-hidden="true" tabindex="-1"></a>d_loci <span class="ot">&lt;-</span> all_loci <span class="sc">%&gt;%</span></span>
<span id="cb22-26"><a href="#cb22-26" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(locus_name,chr,locus_start,locus_end) <span class="sc">%&gt;%</span></span>
<span id="cb22-27"><a href="#cb22-27" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">distinct</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="select-a-region-to-run-coloc" class="level2">
<h2 class="anchored" data-anchor-id="select-a-region-to-run-coloc">select a region to run coloc</h2>
<div class="cell">
<details>
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="co"># select regions to fine map. we are going to use regions in chromosome 1</span></span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>uniq_loci <span class="ot">&lt;-</span> d_loci <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="st">"chr1"</span> <span class="sc">==</span> chr)</span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a>n <span class="ot">=</span> <span class="dv">3</span> <span class="co"># chr1_107867043_109761309 region</span></span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a><span class="co"># extract information</span></span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a>l_chr <span class="ot">=</span> <span class="fu">as.numeric</span>(<span class="fu">str_remove</span>(uniq_loci[n,<span class="st">"chr"</span>],<span class="st">"chr"</span>))</span>
<span id="cb23-7"><a href="#cb23-7" aria-hidden="true" tabindex="-1"></a>s_chr <span class="ot">=</span> uniq_loci[n,]<span class="sc">$</span>chr</span>
<span id="cb23-8"><a href="#cb23-8" aria-hidden="true" tabindex="-1"></a>l_start <span class="ot">=</span> uniq_loci[n,]<span class="sc">$</span>locus_start</span>
<span id="cb23-9"><a href="#cb23-9" aria-hidden="true" tabindex="-1"></a>l_stop <span class="ot">=</span> uniq_loci[n,]<span class="sc">$</span>locus_end</span>
<span id="cb23-10"><a href="#cb23-10" aria-hidden="true" tabindex="-1"></a>l_name <span class="ot">=</span> uniq_loci[n,]<span class="sc">$</span>locus_name</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="prepare-gwas-data-for-coloc" class="level2">
<h2 class="anchored" data-anchor-id="prepare-gwas-data-for-coloc">Prepare gwas data for coloc</h2>
<div class="cell">
<details>
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="co"># select snps for the region from the summary stats</span></span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>ss <span class="ot">&lt;-</span> gwas <span class="sc">%&gt;%</span> </span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(chromosome <span class="sc">==</span> s_chr) <span class="sc">%&gt;%</span></span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(position <span class="sc">&gt;=</span> l_start <span class="sc">&amp;</span> position <span class="sc">&lt;=</span> l_stop) <span class="sc">%&gt;%</span> </span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span> <span class="fu">is.na</span>(effect_size))</span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a><span class="co"># find the snps in the genotype to calculate the correlation</span></span>
<span id="cb24-8"><a href="#cb24-8" aria-hidden="true" tabindex="-1"></a>g.snps <span class="ot">&lt;-</span> ss <span class="sc">%&gt;%</span> <span class="fu">inner_join</span>(snp_info <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">chr =</span> glue<span class="sc">::</span><span class="fu">glue</span>(<span class="st">"chr{chr}"</span>)), </span>
<span id="cb24-9"><a href="#cb24-9" aria-hidden="true" tabindex="-1"></a>                            <span class="at">by=</span><span class="fu">c</span>(<span class="st">"chromosome"</span> <span class="ot">=</span> <span class="st">"chr"</span>,<span class="st">"panel_variant_id"</span> <span class="ot">=</span> <span class="st">"snp"</span>))</span>
<span id="cb24-10"><a href="#cb24-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-11"><a href="#cb24-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-12"><a href="#cb24-12" aria-hidden="true" tabindex="-1"></a><span class="co"># select genotype to calculate correlation</span></span>
<span id="cb24-13"><a href="#cb24-13" aria-hidden="true" tabindex="-1"></a><span class="co">#f_mat &lt;- X_mat[,g.snps$snp]</span></span>
<span id="cb24-14"><a href="#cb24-14" aria-hidden="true" tabindex="-1"></a>f_mat <span class="ot">&lt;-</span> X_mat[,g.snps<span class="sc">$</span>panel_variant_id]</span>
<span id="cb24-15"><a href="#cb24-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-16"><a href="#cb24-16" aria-hidden="true" tabindex="-1"></a><span class="co"># calculate corr</span></span>
<span id="cb24-17"><a href="#cb24-17" aria-hidden="true" tabindex="-1"></a>R <span class="ot">=</span> <span class="fu">cora</span>(f_mat) <span class="co"># the package is for speed</span></span>
<span id="cb24-18"><a href="#cb24-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-19"><a href="#cb24-19" aria-hidden="true" tabindex="-1"></a><span class="do">## clean up</span></span>
<span id="cb24-20"><a href="#cb24-20" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(f_mat)</span>
<span id="cb24-21"><a href="#cb24-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-22"><a href="#cb24-22" aria-hidden="true" tabindex="-1"></a>ff <span class="ot">&lt;-</span> g.snps <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span> <span class="fu">is.na</span>(effect_size)) <span class="sc">%&gt;%</span> <span class="co">#select(-snp) %&gt;% </span></span>
<span id="cb24-23"><a href="#cb24-23" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">snp=</span>panel_variant_id,<span class="at">beta=</span>effect_size) <span class="sc">%&gt;%</span> </span>
<span id="cb24-24"><a href="#cb24-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">varbeta =</span> standard_error<span class="sc">^</span><span class="dv">2</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb24-25"><a href="#cb24-25" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(beta,varbeta,snp,position) <span class="sc">%&gt;%</span> <span class="fu">as.list</span>()</span>
<span id="cb24-26"><a href="#cb24-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-27"><a href="#cb24-27" aria-hidden="true" tabindex="-1"></a>ff<span class="sc">$</span>type <span class="ot">&lt;-</span> <span class="st">"cc"</span></span>
<span id="cb24-28"><a href="#cb24-28" aria-hidden="true" tabindex="-1"></a>ff<span class="sc">$</span>sdY <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb24-29"><a href="#cb24-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-30"><a href="#cb24-30" aria-hidden="true" tabindex="-1"></a>ff<span class="sc">$</span>LD <span class="ot">=</span> R</span>
<span id="cb24-31"><a href="#cb24-31" aria-hidden="true" tabindex="-1"></a>ff<span class="sc">$</span>N <span class="ot">=</span> <span class="dv">184305</span></span>
<span id="cb24-32"><a href="#cb24-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-33"><a href="#cb24-33" aria-hidden="true" tabindex="-1"></a><span class="do">## check the data (NULL means it's fine)</span></span>
<span id="cb24-34"><a href="#cb24-34" aria-hidden="true" tabindex="-1"></a><span class="fu">check_dataset</span>(ff)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="prepare-eqtl-data-for-coloc" class="level2">
<h2 class="anchored" data-anchor-id="prepare-eqtl-data-for-coloc">Prepare eqtl data for coloc</h2>
<div class="cell">
<details>
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Using SORT1 gene and liver tissue</span></span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>gene <span class="ot">&lt;-</span> gene_ss <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(gene_id <span class="sc">==</span> <span class="st">"ENSG00000134243.11"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">snp =</span> variant_id,<span class="at">beta =</span> slope, <span class="at">MAF =</span> maf,</span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a>                <span class="at">pvalue =</span> pval_nominal) <span class="sc">%&gt;%</span> </span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">varbeta =</span> slope_se<span class="sc">^</span><span class="dv">2</span>, <span class="at">name =</span> snp) <span class="sc">%&gt;%</span> </span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span> <span class="fu">is.na</span>(varbeta)) <span class="sc">%&gt;%</span> </span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(name, <span class="at">into =</span> <span class="fu">c</span>(<span class="st">"chr"</span>, <span class="st">"position"</span>,<span class="st">"ref"</span>,<span class="st">"alt"</span>,<span class="st">"build"</span>),<span class="at">sep =</span> <span class="st">"_"</span>)</span>
<span id="cb25-8"><a href="#cb25-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-9"><a href="#cb25-9" aria-hidden="true" tabindex="-1"></a><span class="do">## calculate the ld matrix</span></span>
<span id="cb25-10"><a href="#cb25-10" aria-hidden="true" tabindex="-1"></a><span class="do">### get the snps</span></span>
<span id="cb25-11"><a href="#cb25-11" aria-hidden="true" tabindex="-1"></a>gene.snps <span class="ot">&lt;-</span> gene <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">position =</span> <span class="fu">as.integer</span>(position)) <span class="sc">%&gt;%</span> </span>
<span id="cb25-12"><a href="#cb25-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(snp_info <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">chr =</span> glue<span class="sc">::</span><span class="fu">glue</span>(<span class="st">"chr{chr}"</span>)), </span>
<span id="cb25-13"><a href="#cb25-13" aria-hidden="true" tabindex="-1"></a>                              <span class="at">by=</span><span class="fu">c</span>(<span class="st">"chr"</span> <span class="ot">=</span> <span class="st">"chr"</span>,<span class="st">"snp"</span> <span class="ot">=</span> <span class="st">"snp"</span>))</span>
<span id="cb25-14"><a href="#cb25-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-15"><a href="#cb25-15" aria-hidden="true" tabindex="-1"></a><span class="co"># select genotype to calculate correlation</span></span>
<span id="cb25-16"><a href="#cb25-16" aria-hidden="true" tabindex="-1"></a>g_mat <span class="ot">&lt;-</span> X_mat[,gene.snps<span class="sc">$</span>snp]</span>
<span id="cb25-17"><a href="#cb25-17" aria-hidden="true" tabindex="-1"></a><span class="co"># calculate corr</span></span>
<span id="cb25-18"><a href="#cb25-18" aria-hidden="true" tabindex="-1"></a>g.R <span class="ot">=</span> <span class="fu">cora</span>(g_mat)</span>
<span id="cb25-19"><a href="#cb25-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-20"><a href="#cb25-20" aria-hidden="true" tabindex="-1"></a><span class="co"># clean up</span></span>
<span id="cb25-21"><a href="#cb25-21" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(g_mat)</span>
<span id="cb25-22"><a href="#cb25-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-23"><a href="#cb25-23" aria-hidden="true" tabindex="-1"></a><span class="co"># format data for coloc</span></span>
<span id="cb25-24"><a href="#cb25-24" aria-hidden="true" tabindex="-1"></a>gg <span class="ot">&lt;-</span> gene <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(snp <span class="sc">%in%</span> gene.snps<span class="sc">$</span>snp) <span class="sc">%&gt;%</span></span>
<span id="cb25-25"><a href="#cb25-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">position =</span> <span class="fu">as.integer</span>(position)) <span class="sc">%&gt;%</span> </span>
<span id="cb25-26"><a href="#cb25-26" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(beta,varbeta,snp,position,MAF, pvalue) <span class="sc">%&gt;%</span> <span class="fu">as.list</span>()</span>
<span id="cb25-27"><a href="#cb25-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-28"><a href="#cb25-28" aria-hidden="true" tabindex="-1"></a>gg<span class="sc">$</span>type <span class="ot">&lt;-</span> <span class="st">"quant"</span></span>
<span id="cb25-29"><a href="#cb25-29" aria-hidden="true" tabindex="-1"></a>gg<span class="sc">$</span>LD <span class="ot">=</span> g.R</span>
<span id="cb25-30"><a href="#cb25-30" aria-hidden="true" tabindex="-1"></a>gg<span class="sc">$</span>N <span class="ot">=</span> <span class="dv">208</span> <span class="co"># 670 for blood</span></span>
<span id="cb25-31"><a href="#cb25-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-32"><a href="#cb25-32" aria-hidden="true" tabindex="-1"></a><span class="do">## check the data</span></span>
<span id="cb25-33"><a href="#cb25-33" aria-hidden="true" tabindex="-1"></a><span class="fu">check_dataset</span>(gg)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="run-older-version-of-coloc-which-assumes-single-causal-variant" class="level2">
<h2 class="anchored" data-anchor-id="run-older-version-of-coloc-which-assumes-single-causal-variant">run older version of coloc which assumes single causal variant</h2>
<p><em>coloc.abf</em> makes the simplifying assumption that each trait has at most one causal variant in the region under consideration</p>
<div class="cell">
<details>
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a>my.res <span class="ot">&lt;-</span> <span class="fu">coloc.abf</span>(<span class="at">dataset1=</span>ff, <span class="at">dataset2=</span>gg)</span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a><span class="fu">sensitivity</span>(my.res,<span class="st">"H4 &gt; 0.9"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="run-coloc-allowing-multiple-causal-variants" class="level2">
<h2 class="anchored" data-anchor-id="run-coloc-allowing-multiple-causal-variants">run coloc allowing multiple causal variants</h2>
<p>Multiple causal variants, using SuSiE to separate the signals</p>
<div class="cell">
<details>
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Run susie fine maooing</span></span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a>S3 <span class="ot">=</span> <span class="fu">runsusie</span>(ff)</span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a>S4 <span class="ot">=</span> <span class="fu">runsusie</span>(gg)</span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a><span class="co">#summary(S3)</span></span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-6"><a href="#cb27-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Run coloc</span></span>
<span id="cb27-7"><a href="#cb27-7" aria-hidden="true" tabindex="-1"></a>susie.res<span class="ot">=</span><span class="fu">coloc.susie</span>(S3,S4)</span>
<span id="cb27-8"><a href="#cb27-8" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(susie.res<span class="sc">$</span>summary)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>SuSiE can take a while to run on larger datasets, so it is best to run once per dataset with the =runsusie= function, store the results and feed those into subsequent analyses.</p>
<p>plot the coloc result with the sensitivity function because weird effects are much easier to understand visually</p>
<div class="cell">
<details>
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sensitivity</span>(susie.res,<span class="st">"H4 &gt; 0.9"</span>,<span class="at">row=</span><span class="dv">1</span>,<span class="at">dataset1=</span>ff,<span class="at">dataset2=</span>gg,)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</section>
<section id="questionnaire-05" class="level1">
<h1>Questionnaire 05</h1>
<p>Colocalization <a href="https://forms.gle/8fiYqfiqv31XHsUz9">Fill out this form</a></p>
<table class="table">
<colgroup>
<col style="width: 5%">
</colgroup>
<thead>
<tr class="header">
<th># Mendelian Randomization Methods</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>title: Transcriptome QGT Training 2022 author: Haky Im date: ‘2022-06-10’ slug: qgt-training categories: - Lab tags: [] editor_options: chunk_output_type: console</td>
</tr>
</tbody>
</table>
</section>
<section id="goals-of-lab-1" class="level1">
<h1>Goals of Lab</h1>
<ul>
<li>Predict whole blood expression</li>
<li>Check how well the prediction works with GEUVADIS expression data</li>
<li>Run association between predicted expression and a simulated phenotype</li>
<li>Calculate association between expression levels and coronary artery disease risk using s-predixcan</li>
<li>Fine-map the coronary artery disease gwas results using torus</li>
<li>Calculate colocalization probability using fastenloc</li>
<li>(skip) Run transcriptome-wide mendelian randomization in one locus of interest</li>
<li>Run cTWAS (fine-map SNPs and genes jointly)</li>
</ul>
</section>
<section id="inital-remarks-1" class="level1">
<h1>Inital Remarks</h1>
<ul>
<li><p>We ask you to actively participate in today’s hands on activities. <strong>Notice that we may ask you to share your screen for pedagogic purposes.</strong></p></li>
<li><p>If you have any concerns about this, please ask me or one of the TAs for assistance. We are here to help you learn.</p></li>
<li><p>After each section, we ask you to fill out the questionnaires</p></li>
<li><p>01 Preliminary https://forms.gle/FDwPcjLFynqQP7Fd8</p></li>
<li><p>02 Prediction https://forms.gle/CZYGyNdfNVH2W9QF7</p></li>
<li><p>03 PrediXcan https://forms.gle/Z8YdYjgYs2zQY5cP6</p></li>
<li><p>04 S-PrediXcan https://forms.gle/VVMYsP7u5Fh18fSY7</p></li>
<li><p>05 Colocalization https://forms.gle/8fiYqfiqv31XHsUz9</p></li>
<li><p>06 cTWAS https://forms.gle/ZpuEuv8Xmxw1eDPf6</p>
<ul>
<li>Name is a mandatory field. You can use your name or a pseudonymn. Just keep using the same name for each submission</li>
<li>We will ask you to submit partial set of answers so that we can get a sense of where people are</li>
</ul></li>
<li><p>Through out the lab you will see code chucks that are written with bash, eval=FALSE in place of R. These chunks are meant to be run in your Rstudio terminal and not in the Rmarkdown file itself.</p></li>
<li><p>R code chunks also have “eval=FALSE”. When you click on the green triangle, the code will run in the R console but not when we knit the document. This just happened to be more convenient for us to generate a knitted html to be shared <a href="https://lab-notes.hakyimlab.org/post/2022/06/10/qgt-training/">here</a>.</p></li>
</ul>
</section>
<section id="set-up-1" class="level1">
<h1>Set Up</h1>
<ul class="task-list">
<li><input type="checkbox" disabled="">Open the RStudio cloud project <a href="https://rstudio.cloud/spaces/232505/content/4013383">here</a></li>
<li><input type="checkbox" disabled="">Load Rstudio Libraries</li>
</ul>
<pre class="{r,eval=false}"><code>library(tidyverse)

## packages needed for susie+coloc
library(data.table)
library(BEDMatrix)
library(Rfast)
library(susieR)
library(coloc)
##library(tidyverse)

</code></pre>
<p>-[ ] Navigate to starting directory <code>{bash, eval=FALSE} cd "/cloud/project/"</code></p>
<ul class="task-list">
<li><input type="checkbox" disabled="">activate the the imlabtools environment, which will make sure that the right version of python modules are available</li>
</ul>
<pre class="{bash,eval=false}"><code>conda activate imlabtools</code></pre>
<ul class="task-list">
<li><input type="checkbox" disabled="">To define some variables to access the data more easily within the R session, run the following r chunk</li>
</ul>
<pre class="{r,eval=false}"><code>print(getwd())

lab="/cloud/project/QGT-Columbia-HKI-repo/"
CODE=glue::glue("{lab}/code")
source(glue::glue("{CODE}/load_data_functions.R"))
source(glue::glue("{CODE}/plotting_utils_functions.R"))

PRE="/cloud/project/QGT-Columbia-HKI-repo/box_files"
MODEL=glue::glue("{PRE}/models")
DATA=glue::glue("{PRE}/data")
RESULTS=glue::glue("{PRE}/results")
METAXCAN=glue::glue("{PRE}/repos/MetaXcan-master/software")
FASTENLOC=glue::glue("{PRE}/repos/fastenloc-master")

# This is a reference table we'll use a lot throughout the lab. It contains information about the genes.
gencode_df = load_gencode_df()
</code></pre>
<ul class="task-list">
<li><input type="checkbox" disabled="">check the values of the variables you just defined in R ```{r, eval=FALSE}</li>
</ul>
<p>MODEL</p>
<p>DATA</p>
<pre><code>

- [ ] define some variables to access the data more easily in the terminal. Remember we are running R code in the R console and command line code in the terminal.
```{bash, eval=FALSE}
export PRE="/cloud/project/QGT-Columbia-HKI-repo/box_files"
export LAB="/cloud/project/QGT-Columbia-HKI-repo/"
export CODE=$LAB/code
export DATA=$PRE/data
export MODEL=$PRE/models
export RESULTS=$PRE/results
export METAXCAN=$PRE/repos/MetaXcan-master/software</code></pre>
<ul class="task-list">
<li><input type="checkbox" disabled="">check the values of the variables you just defined</li>
</ul>
<p>```{bash, eval=FALSE}</p>
<p>echo $CODE echo $RESULTS</p>
<pre><code>

# Questionnaire 01
[Fill out this form](https://forms.gle/FDwPcjLFynqQP7Fd8)



# Transcriptome-wide association (Review) 
Now we will perform a transcriptome-wide association analysis using the PrediXcan suite of tools.
![Transcriptome-wide association methods](https://raw.githubusercontent.com/hakyimlab/QGT-Columbia-HKI/master/extras/figures/Association-Methods.png)

We start by predicting the expression levels of genes using the genotype data and the prediction weights and then perform an association between the predicted expression and the phenotype (denoted trait in the figure below).

![](https://raw.githubusercontent.com/hakyimlab/QGT-Columbia-HKI/master/extras/figures/PrediXcan-run.png)



# Predict Expression from genotype

In this section we will predict expression of genes in whole blood using the Predict.py code in the METAXCAN folder.

- Prediction models (weights) are located in the MODEL folder. Additional models for different tissues and transcriptome studies can be downloaded from [predictdb.org](http://predictdb.org).

- This run should take about one minute.

- [ ] run the following code in the terminal.
 
```{bash, eval=FALSE}
printf "Predict expression\n\n"
python3 $METAXCAN/Predict.py \
--model_db_path $PRE/models/gtex_v8_en/en_Whole_Blood.db \
--vcf_genotypes $DATA/predixcan/genotype/filtered.vcf.gz \
--vcf_mode genotyped \
--variant_mapping $DATA/predixcan/gtex_v8_eur_filtered_maf0.01_monoallelic_variants.txt.gz id rsid \
--on_the_fly_mapping METADATA "chr{}_{}_{}_{}_b38" \
--prediction_output $RESULTS/predixcan/Whole_Blood__predict.txt \
--prediction_summary_output $RESULTS/predixcan/Whole_Blood__summary.txt \
--verbosity 9 \
--throw</code></pre>
<blockquote class="blockquote">
<p>Note we are only predicting chromosome 22 here (check by running “predicted_expression %&gt;% count(chromosome)” in the console)</p>
</blockquote>
<ul class="task-list">
<li><input type="checkbox" disabled="">run following code in the console to get information on reported prediction performance.</li>
</ul>
<blockquote class="blockquote">
<p>Find additional information <a href="https://github.com/hakyimlab/MetaXcan/wiki/Individual-level-PrediXcan:-introduction,-tutorials-and-manual">in this wiki</a></p>
</blockquote>
<pre class="{r,eval=false}"><code>
prediction_fp = glue::glue("{RESULTS}/predixcan/Whole_Blood__predict.txt")

## Read the Predict.py output into a dataframe. This function reorganizes the data and adds gene names.
predicted_expression = load_predicted_expression(prediction_fp, gencode_df)

head(predicted_expression)

## read summary of prediction, number of SNPs per gene, cross validated prediction performance
prediction_summary = load_prediction_summary(glue::glue("{RESULTS}/predixcan/Whole_Blood__summary.txt"), gencode_df)
## number of genes with a prediction model
dim(prediction_summary)
head(prediction_summary)

## how many unique genes were predicted?
predicted_expression %&gt;% .[["gene_id"]] %&gt;% unique() %&gt;% length()

## gene expression were predicted for how many people?
predicted_expression %&gt;% .[["IID"]] %&gt;% unique() %&gt;% length()

print("distribution of prediction performance r2")
summary(prediction_summary$pred_perf_r2)
hist(prediction_summary$pred_perf_r2)
## Note: this is what the prediction trainer reported as prediction performance
</code></pre>
<section id="optional-assess-actual-prediction-performance-1" class="level2">
<h2 class="anchored" data-anchor-id="optional-assess-actual-prediction-performance-1">(Optional) Assess Actual Prediction Performance</h2>
<pre class="{r,eval=false}"><code>## download and read observed expression data from GEUVADIS 
## from https://uchicago.box.com/s/4y7xle5l0pnq9d1fwmthe2ewhogrnlrv

obs_exp&lt;- read_csv(glue::glue("{DATA}/predixcan/GEUVADIS.observed_df.csv.gz"))

## Note that the version of the ensemble id of the gene was removed
head(predicted_expression)

## Q: how many genes were predicted?
length(unique(predicted_expression$gene_id))

## inner join predicted expression with observed expression data (by IID and gene)
## common errors occur when ensemble id's have versions in one set and not the other set
fullset=inner_join(predicted_expression, obs_exp, by = c("gene_id","IID"))

## calculate spearman correlation for all genes
genelist = unique(predicted_expression$gene_id)
corvec = rep(NA,length(genelist))
names(corvec) = genelist
for(gg in 1:length(genelist))
{
  ind = fullset$gene_id==genelist[gg]
  corvec[gg] = cor(fullset$predicted_expression[ind], fullset$observed_expression[ind])
}

## what's the best performing gene?

## plot the histogram of the prediction performance
hist(corvec)

## list the top 10 best performing genes
head(sort(corvec,decreasing = TRUE),2)
tail(sort(corvec,decreasing = TRUE),2)

## plot the correlation of the top 2 best performing genes bottom 2

geneid = "ENSG00000100376"
genename = gencode_df %&gt;% filter(gene_id==geneid) %&gt;% .[["gene_name"]]
fullset %&gt;% filter(gene_id==geneid) %&gt;% ggplot(aes(observed_expression, predicted_expression))+geom_point()+ggtitle(paste(genename, "-",geneid))

geneid = "ENSG00000075234"
genename = gencode_df %&gt;% filter(gene_id==geneid) %&gt;% .[["gene_name"]]
fullset %&gt;% filter(gene_id==geneid) %&gt;% ggplot(aes(observed_expression, predicted_expression))+geom_point()+ggtitle(paste(genename, "-", geneid))

geneid = "ENSG00000070371"
genename = gencode_df %&gt;% filter(gene_id==geneid) %&gt;% .[["gene_name"]]
fullset %&gt;% filter(gene_id==geneid) %&gt;% ggplot(aes(observed_expression, predicted_expression))+geom_point()+ggtitle(paste(genename, "-", geneid))

geneid = "ENSG00000184164"
genename = gencode_df %&gt;% filter(gene_id==geneid) %&gt;% .[["gene_name"]]
fullset %&gt;% filter(gene_id==geneid) %&gt;% ggplot(aes(observed_expression, predicted_expression))+geom_point()+ggtitle(paste(genename, "-", geneid))
</code></pre>
</section>
</section>
<section id="questionnaire-02-1" class="level1">
<h1>Questionnaire 02</h1>
<p>Prediction <a href="https://forms.gle/CZYGyNdfNVH2W9QF7">Fill out this form</a></p>
</section>
<section id="run-predixcan-association-1" class="level1">
<h1>Run PrediXcan association</h1>
<p>We are going to use a simulated phenotype for which only UPK3A has an effect on the phenotype (<span class="math inline">\(\beta=-0.9887378\)</span>)</p>
<p><span class="math inline">\(Y = \sum_k T_k \beta_k + \epsilon\)</span></p>
<p>with random effects <span class="math inline">\(\beta_k \sim (1-\pi)\cdot \delta_0 + \pi\cdot N(0,1)\)</span></p>
<p>```{bash, eval=FALSE}</p>
<p>export PHENO=“sim.spike_n_slab_0.01_pve0.1”</p>
<p>printf “association” python3 $METAXCAN/PrediXcanAssociation.py<br>
–expression_file $RESULTS/predixcan/Whole_Blood__predict.txt<br>
–input_phenos_file <span class="math inline">\(DATA/predixcan/phenotype/\)</span>PHENO.txt<br>
–input_phenos_column pheno<br>
–output <span class="math inline">\(RESULTS/predixcan/\)</span>PHENO/Whole_Blood__association.txt<br>
–verbosity 9<br>
–throw</p>
<pre><code>
More predicted phenotypes can be found in $DATA/predixcan/phenotype/. The naming of the phenotypes provides information about the genetic architecture: the number after pve is the proportion of variance of Y explained by the genetic component of expression. The number after spike_n_slab represents the probability that a gene is causal π (i.e. prob β≠0)

## Looking at Association Results 

```{r,eval=FALSE}
## read association results
PHENO="sim.spike_n_slab_0.01_pve0.1"

predixcan_association = load_predixcan_association(glue::glue("{RESULTS}/predixcan/{PHENO}/Whole_Blood__association.txt"), gencode_df)

## take a look at the results
dim(predixcan_association)
predixcan_association %&gt;% arrange(pvalue) %&gt;% select(gene_name,effect,se,pvalue,gene) %&gt;% head
predixcan_association %&gt;% arrange(pvalue) %&gt;% ggplot(aes(pvalue)) + geom_histogram(bins=10)
## compare distribution against the null (uniform)
gg_qqplot(predixcan_association$pvalue, max_yval = 40)
</code></pre>
<section id="comparing-the-estimated-effect-size-with-true-effect-size-1" class="level2">
<h2 class="anchored" data-anchor-id="comparing-the-estimated-effect-size-with-true-effect-size-1">Comparing the Estimated Effect Size with True Effect Size</h2>
<p>```{r, eval=FALSE}</p>
<p>truebetas = load_truebetas(glue::glue(“{DATA}/predixcan/phenotype/gene-effects/{PHENO}.txt”), gencode_df) betas = (predixcan_association %&gt;% inner_join(truebetas,by=c(“gene”=“gene_id”)) %&gt;% select(c(‘estimated_beta’=‘effect’, ‘true_beta’=‘effect_size’, ‘pvalue’, ‘gene_id’=‘gene’, ‘gene_name’=‘gene_name.x’, ‘region_id’=‘region_id.x’))) betas %&gt;% arrange(pvalue) %&gt;% select(gene_name,estimated_beta,true_beta,pvalue) %&gt;% head ## do you see examples of potential LD contamination? betas %&gt;% mutate(causal= true_beta!=0) %&gt;% ggplot(aes(estimated_beta, true_beta,col=causal))+geom_point(alpha=0.6,size=5)+geom_abline()+theme_bw()</p>
<pre><code>
&gt; UPK3A is the causal gene and has the most significant pvalue. RIBC2 is also significantly associated but has no causal role (we know because we simulated the phenotype that way). Why?

Hint: correlation between the genes

# Questionnaire 03
PrediXcan Association [Fill out this form] (https://forms.gle/Z8YdYjgYs2zQY5cP6)

# Run Summary PrediXcan

Now we will use the summary results from a GWAS of coronary artery disease to calculate the association between the genetic component of the expression of genes and coronary artery disease risk. We will use the SPrediXcan.py.

![](https://uchicago.box.com/shared/static/m45nnkeskzh88ifnv8td5e3fntdfzhjs.png)

The GWAS results (harmonized and imputed) for coronary artery disease are available in $PRE/spredixcan/data/
## Running S-PrediXcan 

```{bash, eval=FALSE}

python $METAXCAN/SPrediXcan.py \
--gwas_file  $DATA/spredixcan/imputed_CARDIoGRAM_C4D_CAD_ADDITIVE.txt.gz \
--snp_column panel_variant_id \
--effect_allele_column effect_allele \
--non_effect_allele_column non_effect_allele \
--zscore_column zscore \
--model_db_path $MODEL/gtex_v8_mashr/mashr_Whole_Blood.db \
--covariance $MODEL/gtex_v8_mashr/mashr_Whole_Blood.txt.gz \
--keep_non_rsid \
--additional_output \
--model_db_snp_key varID \
--throw \
--output_file $RESULTS/spredixcan/eqtl/CARDIoGRAM_C4D_CAD_ADDITIVE__PM__Whole_Blood.csv
</code></pre>
<blockquote class="blockquote">
<p>We can run the full genome because the summary statistics based PrediXcan is much faster than individual level one.</p>
</blockquote>
</section>
<section id="plot-and-interpret-results-1" class="level2">
<h2 class="anchored" data-anchor-id="plot-and-interpret-results-1">Plot and Interpret Results</h2>
<pre class="{r,eval=false}"><code>spredixcan_association = load_spredixcan_association(glue::glue("{RESULTS}/spredixcan/eqtl/CARDIoGRAM_C4D_CAD_ADDITIVE__PM__Whole_Blood.csv"), gencode_df)
dim(spredixcan_association)
spredixcan_association %&gt;% arrange(pvalue) %&gt;% head
spredixcan_association %&gt;% arrange(pvalue) %&gt;% ggplot(aes(pvalue)) + geom_histogram(bins=20)

gg_qqplot(spredixcan_association$pvalue)</code></pre>
<ul>
<li><p><em>Question:</em> SORT1, considered to be a causal gene for LDL cholesterol and as a consequence of coronary artery disease, is not found here. Why?</p></li>
<li><p><input type="checkbox" disabled="">check whether SORT1 is expressed in whole blood <a href="https://gtexportal.org/home/gene/SORT1">GTEx portal</a></p></li>
<li><p><input type="checkbox" disabled="">check whether SORT1 has eQTL in whole blood <a href="https://gtexportal.org/home/gene/SORT1">GTEx portal</a></p></li>
</ul>
</section>
<section id="run-s-predixcan-using-gene-expression-predicted-in-liver-1" class="level2">
<h2 class="anchored" data-anchor-id="run-s-predixcan-using-gene-expression-predicted-in-liver-1">Run S-PrediXcan using gene expression predicted in liver</h2>
<p>-[] Run s-predixcan with liver model, do you find SORT1? Is it significant? <code>{bash, eval=FALSE} #loction Liver models  #/cloud/project/QGT-Columbia-HKI-repo/box_files/models/gtex_v8_mashr/ python $METAXCAN/SPrediXcan.py \ --gwas_file  $DATA/spredixcan/imputed_CARDIoGRAM_C4D_CAD_ADDITIVE.txt.gz \ --snp_column panel_variant_id \ --effect_allele_column effect_allele \ --non_effect_allele_column non_effect_allele \ --zscore_column zscore \ --model_db_path $MODEL/gtex_v8_mashr/mashr_Liver.db \ --covariance $MODEL/gtex_v8_mashr/mashr_Liver.txt.gz \ --keep_non_rsid \ --additional_output \ --model_db_snp_key varID \ --throw \ --output_file $RESULTS/spredixcan/eqtl/CARDIoGRAM_C4D_CAD_ADDITIVE__PM__Liver.csv</code></p>
<pre class="{r,eval=false}"><code>spredixcan_association_L= load_spredixcan_association(glue::glue("{RESULTS}/spredixcan/eqtl/CARDIoGRAM_C4D_CAD_ADDITIVE__PM__Liver.csv"), gencode_df)
dim(spredixcan_association_L)
spredixcan_association_L %&gt;% arrange(pvalue) %&gt;% head
spredixcan_association_L %&gt;% arrange(pvalue) %&gt;% ggplot(aes(pvalue)) + geom_histogram(bins=20)

gg_qqplot(spredixcan_association_L$pvalue)
col_order= c("gene_name","gene","zscore","effect_size","pvalue","var_g","pred_perf_r2", "pred_perf_pval","pred_perf_qval", "n_snps_used", "n_snps_in_cov", "n_snps_in_model","best_gwas_p","largest_weight")
spredixcan_association_L &lt;- spredixcan_association_L[, col_order]
filter(spredixcan_association_L, gene_name=="SORT1")</code></pre>
</section>
<section id="optional-compare-zscores-in-liver-and-whole-blood.-1" class="level2">
<h2 class="anchored" data-anchor-id="optional-compare-zscores-in-liver-and-whole-blood.-1">(Optional) Compare zscores in liver and whole blood.</h2>
<blockquote class="blockquote">
<p>Recall that zscore is the effect size divided by the standard error</p>
</blockquote>
<p>```{r, eval=FALSE} spredixcan_association_L=rename(spredixcan_association_L, zscore_liver = “zscore”) head(spredixcan_association_L) test=left_join(spredixcan_association, spredixcan_association_L, by=“gene_name”) test=select(test,“gene_name”,“zscore”,“zscore_liver”) test %&gt;% arrange(zscore_liver) %&gt;% head</p>
<p>test %&gt;% mutate(zscore_WB=zscore) %&gt;% ggplot(aes(zscore_WB,zscore_liver)) + geom_point(size=3,alpha=.6) + geom_abline()</p>
</section>
<section id="s-predixcan-association-in-liver-and-whole-blood-are-significantly-correlated" class="level2">
<h2 class="anchored" data-anchor-id="s-predixcan-association-in-liver-and-whole-blood-are-significantly-correlated">S-PrediXcan association in liver and whole blood are significantly correlated</h2>
<p>cor.test(test<span class="math inline">\(zscore,test\)</span>zscore_liver)</p>
<pre><code>
## (Optional) Run MultiXcan
- multixcan aggregates information across multiple tissues to boost the power to detect association. It was developed movivated by the fact that eQTLs are shared across multiple tissues, i.e. many genetic variants that regulate expression are common across tissues.

- before you run multixcan ensure you have run s-predixcan for all the tissues you want to multixcan. In this tutorial we have two tissues (liver and whole blood), ensure you have run s-predixcan with the two tissues before running multixcan. 

- One thing to note is to ensure similar naming pattern for the output files. This is to ensure the files are captured correctly when running multixcan's filter.

```{bash, eval=FALSE}

python $METAXCAN/SMulTiXcan.py \
--models_folder $MODEL/gtex_v8_mashr \
--models_name_pattern "mashr_(.*).db" \
--snp_covariance $MODEL/gtex_v8_expression_mashr_snp_smultixcan_covariance.txt.gz \
--metaxcan_folder $RESULTS/spredixcan/eqtl/ \
--metaxcan_filter "CARDIoGRAM_C4D_CAD_ADDITIVE__PM__(.*).csv" \
--metaxcan_file_name_parse_pattern "(.*)__PM__(.*).csv" \
--gwas_file $DATA/spredixcan/imputed_CARDIoGRAM_C4D_CAD_ADDITIVE.txt.gz \
--snp_column panel_variant_id --effect_allele_column effect_allele --non_effect_allele_column non_effect_allele --zscore_column zscore --keep_non_rsid --model_db_snp_key varID \
--cutoff_condition_number 30 \
--verbosity 9 \
--throw \
--output $RESULTS/smultixcan/eqtl/CARDIoGRAM_C4D_CAD_ADDITIVE_smultixcan.txt
</code></pre>
</section>
<section id="questionnaire-04-1" class="level2">
<h2 class="anchored" data-anchor-id="questionnaire-04-1">Questionnaire 04</h2>
<p>04 S-PrediXcan <a href="https://forms.gle/VVMYsP7u5Fh18fSY7">Fill out this form</a></p>
</section>
</section>
<section id="colocalization-methods-review-1" class="level1">
<h1>Colocalization Methods (Review)</h1>
<p>Colocalization methods seek to estimate the probability that the complex trait and expression causal variants are the same. We favor methods that calculate the probability of causality for each trait (posterior inclusion probability), called fine-mapping methods. Here we use susie for fine-mapping and coloc for colocalization.</p>
<p><img src="https://raw.githubusercontent.com/hakyimlab/QGT-Columbia-HKI/master/extras/figures/colocalization-run.png" class="img-fluid"></p>
</section>
<section id="run-colocalization-1" class="level1">
<h1>Run colocalization</h1>
<p>When you use coloc on your own data, you may want to check out coloc’s documentation, with good advice and tips on avoiding common mistakes https://cran.r-project.org/web/packages/coloc/vignettes</p>
<p>Due to time constraints, we will run one region and one gene only</p>
<ol type="1">
<li>Finemap GWAS of CAD</li>
<li>Finemap eQTL of SORT1</li>
</ol>
<p>For finemapping, we need the summary statistics (effect size, standard errors, etc) and the correlation between SNPs (LD matrix)</p>
<section id="load-the-genotype-to-calculate-the-ld-matrix-1" class="level2">
<h2 class="anchored" data-anchor-id="load-the-genotype-to-calculate-the-ld-matrix-1">load the genotype to calculate the ld matrix</h2>
<p>```{r, eval=FALSE} # load the genotype to calculate the ld matrix X_mat &lt;- BEDMatrix(glue::glue(“{DATA}/colocalization/geuvadis_chr1”)) colnames(X_mat) &lt;- gsub(“\_.*“,”“,colnames(X_mat)) colnames(X_mat) &lt;- str_replace_all(colnames(X_mat),”:“,”_“) snp_info &lt;- fread(glue::glue(”{DATA}/colocalization/geuvadis_chr1.bim”)) %&gt;% setnames(., colnames(.), c(“chr”, “snp”, “cm”, “pos”, “alt”, “ref”))</p>
<p>snp_info<span class="math inline">\(snp &lt;- str_replace_all(snp_info\)</span>snp,“:”,“_“)</p>
<pre><code>
## Load the eqtl and gwas effect size files
```{r, eval=FALSE}
# load the eqtl effect sizes
gene_ss &lt;- fread(glue::glue("{DATA}/colocalization/Liver_chr1.txt"))

# load gwas effect sizes
gwas &lt;- data.table::fread(glue::glue("{DATA}/spredixcan/imputed_CARDIoGRAM_C4D_CAD_ADDITIVE.txt.gz"))

# filter to select genome wide significant snps at 5 × 10−8
filtered_regions &lt;- gwas %&gt;% dplyr::filter(pvalue &lt; 5e-8)

# load the ld block
ldblocks &lt;-read_tsv(glue::glue("{DATA}/spredixcan/eur_ld.hg38.txt.gz"))</code></pre>
</section>
<section id="find-regions-with-the-strongest-signal-in-the-gwas-1" class="level2">
<h2 class="anchored" data-anchor-id="find-regions-with-the-strongest-signal-in-the-gwas-1">Find regions with the strongest signal in the gwas</h2>
<p>```{r, eval=FALSE} # get the loci where the significant snps are located for (n in 1:nrow(filtered_regions)){ # extract genename, start and end variant_id &lt;- as.character(filtered_regions[n,“variant_id”]) variant_chr &lt;- as.character(filtered_regions[n,“chromosome”]) variant_pos &lt;- as.numeric(filtered_regions[n,“position”]) #gene_end &lt;- as.numeric(genes[n,“end”])</p>
<p>locus &lt;- ldblocks %&gt;% dplyr::filter(chr == variant_chr) %&gt;% filter(variant_pos &gt;= start &amp; variant_pos &lt; stop) %&gt;% mutate(locus_name = paste0(chr,“<em>”,start,”</em>”,stop)) %&gt;% dplyr::rename(locus_start=start,locus_end=stop) %&gt;% mutate(variant_id = variant_id, position=variant_pos)</p>
<p># create a data frame with info if (exists(‘all_loci’) &amp;&amp; is.data.frame(get(‘all_loci’))) { all_loci &lt;- rbind(all_loci,locus) } else { all_loci &lt;- locus } }</p>
</section>
</section>
<section id="select-uniq-loci" class="level1">
<h1>select uniq loci</h1>
<p>d_loci &lt;- all_loci %&gt;% dplyr::select(locus_name,chr,locus_start,locus_end) %&gt;% dplyr::distinct()</p>
<pre><code>
## select a region to run coloc
```{r, eval=FALSE}
# select regions to fine map. we are going to use regions in chromosome 1
uniq_loci &lt;- d_loci %&gt;% dplyr::filter("chr1" == chr)
n = 3 # chr1_107867043_109761309 region

# extract information
l_chr = as.numeric(str_remove(uniq_loci[n,"chr"],"chr"))
s_chr = uniq_loci[n,]$chr
l_start = uniq_loci[n,]$locus_start
l_stop = uniq_loci[n,]$locus_end
l_name = uniq_loci[n,]$locus_name
</code></pre>
<section id="prepare-gwas-data-for-coloc-1" class="level2">
<h2 class="anchored" data-anchor-id="prepare-gwas-data-for-coloc-1">Prepare gwas data for coloc</h2>
<p>```{r, eval=FALSE} # select snps for the region from the summary stats ss &lt;- gwas %&gt;% dplyr::filter(chromosome == s_chr) %&gt;% dplyr::filter(position &gt;= l_start &amp; position &lt;= l_stop) %&gt;% dplyr::filter(! is.na(effect_size))</p>
</section>
</section>
<section id="find-the-snps-in-the-genotype-to-calculate-the-correlation" class="level1">
<h1>find the snps in the genotype to calculate the correlation</h1>
<p>g.snps &lt;- ss %&gt;% inner_join(snp_info %&gt;% mutate(chr = glue::glue(“chr{chr}”)), by=c(“chromosome” = “chr”,“panel_variant_id” = “snp”))</p>
</section>
<section id="select-genotype-to-calculate-correlation" class="level1">
<h1>select genotype to calculate correlation</h1>
<p>#f_mat &lt;- X_mat[,g.snps<span class="math inline">\(snp] f_mat &lt;- X_mat[,g.snps\)</span>panel_variant_id]</p>
</section>
<section id="calculate-corr" class="level1">
<h1>calculate corr</h1>
<p>R = cora(f_mat) # the package is for speed</p>
<section id="clean-up" class="level2">
<h2 class="anchored" data-anchor-id="clean-up">clean up</h2>
<p>rm(f_mat)</p>
<p>ff &lt;- g.snps %&gt;% dplyr::filter(! is.na(effect_size)) %&gt;% #select(-snp) %&gt;% dplyr::rename(snp=panel_variant_id,beta=effect_size) %&gt;% mutate(varbeta = standard_error^2) %&gt;% dplyr::select(beta,varbeta,snp,position) %&gt;% as.list()</p>
<p>ff<span class="math inline">\(type &lt;- "cc" ff\)</span>sdY &lt;- 1</p>
<p>ff<span class="math inline">\(LD = R ff\)</span>N = 184305</p>
</section>
<section id="check-the-data-null-means-its-fine" class="level2">
<h2 class="anchored" data-anchor-id="check-the-data-null-means-its-fine">check the data (NULL means it’s fine)</h2>
<p>check_dataset(ff)</p>
<pre><code>
## Prepare eqtl data for coloc
```{r, eval=FALSE}
#Using SORT1 gene and liver tissue
gene &lt;- gene_ss %&gt;% dplyr::filter(gene_id == "ENSG00000134243.11") %&gt;% 
  dplyr::rename(snp = variant_id,beta = slope, MAF = maf,
                pvalue = pval_nominal) %&gt;% 
  mutate(varbeta = slope_se^2, name = snp) %&gt;% 
  filter(! is.na(varbeta)) %&gt;% 
  separate(name, into = c("chr", "position","ref","alt","build"),sep = "_")

## calculate the ld matrix
### get the snps
gene.snps &lt;- gene %&gt;% mutate(position = as.integer(position)) %&gt;% 
  inner_join(snp_info %&gt;% mutate(chr = glue::glue("chr{chr}")), 
                              by=c("chr" = "chr","snp" = "snp"))

# select genotype to calculate correlation
g_mat &lt;- X_mat[,gene.snps$snp]
# calculate corr
g.R = cora(g_mat)

# clean up
rm(g_mat)

# format data for coloc
gg &lt;- gene %&gt;% dplyr::filter(snp %in% gene.snps$snp) %&gt;%
  mutate(position = as.integer(position)) %&gt;% 
  dplyr::select(beta,varbeta,snp,position,MAF, pvalue) %&gt;% as.list()

gg$type &lt;- "quant"
gg$LD = g.R
gg$N = 208 # 670 for blood

## check the data
check_dataset(gg)</code></pre>
</section>
<section id="run-older-version-of-coloc-which-assumes-single-causal-variant-1" class="level2">
<h2 class="anchored" data-anchor-id="run-older-version-of-coloc-which-assumes-single-causal-variant-1">run older version of coloc which assumes single causal variant</h2>
<p><em>coloc.abf</em> makes the simplifying assumption that each trait has at most one causal variant in the region under consideration <code>{r, eval=FALSE} my.res &lt;- coloc.abf(dataset1=ff, dataset2=gg) sensitivity(my.res,"H4 &gt; 0.9")</code></p>
</section>
<section id="run-coloc-allowing-multiple-causal-variants-1" class="level2">
<h2 class="anchored" data-anchor-id="run-coloc-allowing-multiple-causal-variants-1">run coloc allowing multiple causal variants</h2>
<p>Multiple causal variants, using SuSiE to separate the signals ```{r, eval=FALSE} # Run susie fine maooing S3 = runsusie(ff) S4 = runsusie(gg) #summary(S3)</p>
</section>
</section>
<section id="run-coloc" class="level1">
<h1>Run coloc</h1>
<p>susie.res=coloc.susie(S3,S4) print(susie.res$summary)</p>
<pre><code>SuSiE can take a while to run on larger datasets, so it is best to run once per dataset with the =runsusie= function, store the results and feed those into subsequent analyses.

plot the coloc result with the sensitivity function because weird effects are much easier to understand visually
```{r, eval=FALSE}
sensitivity(susie.res,"H4 &gt; 0.9",row=1,dataset1=ff,dataset2=gg,)</code></pre>
</section>
<section id="questionnaire-05-1" class="level1">
<h1>Questionnaire 05</h1>
<p>Colocalization <a href="https://forms.gle/8fiYqfiqv31XHsUz9">Fill out this form</a></p>
<hr>
</section>
<section id="mendelian-randomization-methods" class="level1">
<h1>Mendelian Randomization Methods</h1>
<section id="twmr-for-a-locus" class="level2">
<h2 class="anchored" data-anchor-id="twmr-for-a-locus">TWMR (for a locus)</h2>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="https://raw.githubusercontent.com/hakyimlab/QGT-Columbia-HKI/master/extras/figures/TWMR.png" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">TWMR</figcaption><p></p>
</figure>
</div>
</section>
</section>
<section id="ctwas" class="level1">
<h1>cTWAS</h1>
<p>cTWAS stands for causal TWAS and seeks to calculate the posterior inclusion probability of genes and SNPs in a given region to be causal. It’s a modification of SUSIER that adds predicted gene expression levels in addition to SNPs in the analysis. It is not yet published, but the authors (Siming Zhao, Wesley Crouse, Matthew Stephens et al) have shared a vignette with us to try it out.</p>
<blockquote class="blockquote">
<p>Thank you, Wes, for creating this portion of the lab!</p>
</blockquote>
<pre class="{r,eval=false}"><code>#install.packages("R.utils")

#install.packages("remotes")
#remotes::install_github("simingz/ctwas", ref = "develop")

library(ctwas)

#get positions for region of interest (SORT1/PSRC1 locus)
region &lt;- unlist(strsplit(spredixcan_association$region_id[spredixcan_association$gene_name=="PSRC1"], "_"))
chr &lt;- region[2]
start &lt;- as.numeric(region[3])
end &lt;- as.numeric(region[4])

#format summary statistics (and subset to variants in region to save memory)
z_snp &lt;- data.table::fread(glue::glue("{DATA}/spredixcan/imputed_CARDIoGRAM_C4D_CAD_ADDITIVE.txt.gz"), select=c("chromosome", "position", "variant_id", "effect_allele", "non_effect_allele", "zscore", "sample_size"))
z_snp &lt;- z_snp[z_snp$chromosome==chr &amp; z_snp$position &gt;= start &amp; z_snp$position &lt;= end,]
z_snp &lt;- z_snp[!is.na(z_snp$variant_id),-(1:2)]
colnames(z_snp) &lt;- c("id", "A1", "A2", "z", "ss")

#specify directories for LD matrices and weights
ld_R_dir &lt;- glue::glue("{DATA}/cTWAS/LD_matrices")
weight &lt;-  glue::glue("{MODEL}/gtex_v8_mashr/mashr_Liver.db")

#specify output locations and names for cTWAS
outputdir &lt;- glue::glue("{DATA}/cTWAS/results/")
outname.e &lt;- "CARDIoGRAM_Liver_expr"
outname &lt;- "CARDIoGRAM_Liver_ctwas"

#impute gene z scores using cTWAS and save the results
##################################
# NOTE: we are skipping this step and using the precomputed values 
## takes ~10 minutes
##################################
# ctwas_imputation &lt;- impute_expr_z(z_snp=z_snp, weight=weight, ld_R_dir=ld_R_dir, outputdir=outputdir, outname=outname.e, harmonize_z=F, harmonize_wgt=F)
# save(ctwas_imputation, file = paste0(outputdir, outname.e, "_output.Rd"))
load(paste0(outputdir, outname.e, "_output.Rd"))

z_gene &lt;- ctwas_imputation$z_gene
ld_exprfs &lt;- ctwas_imputation$ld_exprfs
z_snp &lt;- ctwas_imputation$z_snp

#make custom region file for single region
ld_regions_custom &lt;- data.frame("chr" = chr, "start" = start, "stop" = end)

write.table(ld_regions_custom, 
            file= paste0(outputdir, "ld_regions_custom.txt"),
            row.names=F, col.names=T, sep="\t", quote = F)
    
ld_regions_custom &lt;- paste0(outputdir, "ld_regions_custom.txt")

#run cTWAS with pre-specified prior parameters at a single locus
#estimating prior requires genome-wide data, too slow for demonstration
#prior is 1% inclusion for genes and is 100x more likely than SNPs
#prior assumes genes have larger effect size than SNPs, in reasonable range for data we've looked at
ctwas_rss(z_gene=z_gene, z_snp=z_snp, ld_exprfs=ld_exprfs, ld_R_dir = ld_R_dir, ld_regions_custom=ld_regions_custom, outputdir = outputdir, outname = outname, thin = 0.01,
          estimate_group_prior = F,
          estimate_group_prior_var = F,
          group_prior=c(0.01, 0.0001),
          group_prior_var=c(50, 25))

#load results
ctwas_results &lt;- data.table::fread(paste0(outputdir,outname,".susieIrss.txt"))

#merge gene names into the results
sqlite &lt;- RSQLite::dbDriver("SQLite")
db = RSQLite::dbConnect(sqlite, weight)
query &lt;- function(...) RSQLite::dbGetQuery(db, ...)
extra_table &lt;- query("select * from extra")
RSQLite::dbDisconnect(db)

ctwas_results$genename &lt;- extra_table$genename[match(ctwas_results$id, extra_table$gene)]

#show results with highest PIPs
col_order_2= c("genename", "chrom", "id", "pos", "type", "region_tag1", "region_tag2", "cs_index", "susie_pip" , "mu2")
ctwas_results &lt;- ctwas_results[, ..col_order_2]
head(ctwas_results[order(-ctwas_results$susie_pip),])
</code></pre>
</section>
<section id="questionnaire-06" class="level1">
<h1>Questionnaire 06</h1>
<p>cTWAS <a href="https://forms.gle/ZpuEuv8Xmxw1eDPf6">Fill out this form</a></p>
</section>
<section id="faq-and-frequent-errors" class="level1">
<h1>FAQ and frequent errors</h1>
<ul>
<li><p>conda environment may not load correctly after resuming the project in Rstudio cloud</p></li>
<li><p>when inner joining by ensemble id of genes, make sure that both datasets have either the same versions or remove versions</p></li>
</ul>
</section>
<section id="contributors" class="level1">
<h1>Contributors</h1>
<ul>
<li>Festus Nyasimi</li>
<li>Margaret Perry</li>
<li>Wes Crouse</li>
<li>Owen Melia</li>
</ul>
<p>``````````````````` :::</p>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>