---
title: "BrainXcan supplementary tables content"
author: "Haky Im"
date: "2023-08-09"
---

```{r}
suppressMessages(library(tidyverse))
suppressMessages(library(glue))
PRE = "/Users/haekyungim/Library/CloudStorage/Box-Box/LargeFiles/imlab-data/data-Github/web-data"
SLUG="brainxcan-supplementary-tables-content" ## copy the slug from the header
bDATE='2023-08-09' ## copy the date from the blog's header here
DATA = glue("{PRE}/{bDATE}-{SLUG}")
if(!file.exists(DATA)) system(glue::glue("mkdir {DATA}"))
WORK=DATA

##system(glue("open {DATA}")) ## this will open the folder 


```

```{r}
system(glue("ls -r {DATA}/*"))
```





```{latex}
\begin{table}[H]
\centering
\begin{tabular}{lll}
  \toprule
{\textbf{variable}} & {\textbf{information}} & {\textbf{detail}} \\
  \midrule
age & 54.9 (7.34) & range: 40--70 \\
  sex & 52.5\%/47.5\% & raw count: 12814/11595 \\
   \bottomrule
\end{tabular}
\caption[Demographic information of the IDP cohort]{\textbf{Demographic information of the IDP cohort.} A summary of age and sex of the selected 24,409 individuals (IDP cohort) is shown. For age, \textbf{information} shows the average age with standard deviation of age in parentheses and \textbf{detail} shows the range of the age (min and max). For sex, \textbf{information} shows the fraction of each age among the cohort and \textbf{detail} shows the raw counts (female's is listed first and followed by male's)}
\label{S-tab:ukb_cohort}
\end{table}

\begin{table}[H]
  % \begin{footnotesize}
  \caption[UK Biobank brain IDPs being analyzed]{\textbf{UK Biobank brain IDPs being analyzed.} 
  % The table is available at \url{https://docs.google.com/spreadsheets/d/1xzLvwtIshyX10orGnGVyDNuyDHU-zgoYhGYemf-6I10}. 
    This table contains the detailed information about the brain IDPs analyzed in this paper. 
    The columns are listed below.
    \textbf{IDP} shows the IDP identifier used in this paper which uses the format of ``IDP-XXXXX'' with XXXXX being the Field ID in UK Biobank;
    \textbf{modality} indicates whether the IDP is from T1 MRI or diffusion MRI;
    \textbf{subtype} lists the name of the IDP subtypes which is used in the paper to refer to different types of brain IDPs;
    \textbf{pc1\_name} shows the name of the corresponding principal component of the subtype (this will be used as the IDP identifier for the IDP PC1); 
    \textbf{region} shows the anatomical region of the MRI measurement;
    \textbf{side} indicates whether the measurement is from the left or right part of the brain; 
    \textbf{measurement\_type} shows the details of the measurement and the image processing procedure;
    \textbf{dmri\_measure} indicates the type of statistic extracted from the dMRI measurement;
    \textbf{t1\_anatomy\_group} indicates the anatomical grouping for T1 MRI;
    \textbf{notes} shows the description of the IDP extracted from the UK Biobank database;
    \textbf{ukb\_link} lists the link to the Data Field in UK Biobank website.
  }
  \label{S-tab:idp_table}
  % \end{footnotesize}
\end{table}

\begin{table}[H]
  % \begin{footnotesize}
  \caption[The heritability and polygenicity estimates of brain IDPs]{\textbf{The heritability and poligenicity estimates of brain IDPs.} 
    \textbf{IDP}: IDP identifier;
    \textbf{h2} and \textbf{h2\_SE}: The estimated heritability and the corresponding standard error;
    \textbf{Me} and \textbf{Me\_SE}: The estimated polygenicity (defined as $M_e$ in \cite{oconnor:2019}) and the corresponding standard error.
  }
  \label{S-tab:gen_arch}
  % \end{footnotesize}
\end{table}

\begin{table}[H]
  % \begin{footnotesize}
  \caption[The prediction performance of the ridge and elastic net predictors]{\textbf{The prediction performance of the ridge and elastic net predictors.} 
    It contains prediction performance (via cross-validation scheme) of the ridge regression and elastic net based genetic predictors of brain IDPs.
    \textbf{IDP}: IDP identifier;
    \textbf{model\_name}: The type of prediction model being used (ridge regression or elastic net);
    \textbf{Spearman}: The Spearman correlation between the predicted and observed IDP values;
    \textbf{Pearson}: The Pearson correlation between the predicted and observed IDP values;
    \textbf{R2}: The $R^2$ measured from the predicted and observed IDP values;
    \textbf{is\_kept}: An indicator showing whether the genetic predictor passes the quality control filter (e.g. Spearman correlation $>0.1$) so that will be included in BrainXcan software.
  }
  \label{S-tab:cv_perf}
  % \end{footnotesize}
\end{table}


\begin{table}[H]
  % \begin{footnotesize}
  \caption[The list of 9 UK Biobank based phenotypes being analyzed by individual-level BrainXcan]{\textbf{The list of 9 UK Biobank based phenotypes analyzed by individual-level BrainXcan.} 
  % The table is available at \url{https://docs.google.com/spreadsheets/d/150mFxAV1p_IK1l36GKjazBt0pIb5eSC0LO7JeFaTS1I}. 
    It contains the definition of the 9 UK Biobank phenotypes used for individual-level BrainXcan.
    For some phenotypes, they were constructed from multiple UK Biobank fields and the aggregation method across these fields was either taking the sum (`sum') or taking logical OR (`or').
    For binary phenotype, we defined its value as 1 if the field takes the desired values indicated in \textbf{target\_value}. 
    Missing values were either treated as 0 or removed.   
    \textbf{phenotype} shows the name of the defined phenotype;
    \textbf{ukb\_field} shows UK Biobank fields from which the phenotype was constructed;
    \textbf{target\_value}: For binary phenotypes, we defined its value as 1 if the field takes the desired value. For quantitative traits, we used the value as it is (labeled as `asis');
    \textbf{aggregate\_method}: For some phenotypes, they were constructed from multiple UK Biobank fields and the aggregation method across these fields was either taking the sum (`sum') or taking the logical OR (`or');
    \textbf{missing\_values} shows how missing values in a field were handled (either treated as 0 or removed).   
  }
  \label{S-tab:indiv_bxcan_phenotype}
  % \end{footnotesize}
\end{table}

% \begin{table}[H]
%   % \begin{footnotesize}
%   \caption[Individual-level BrainXcan results on 9 UK Biobank phenotypes]{\textbf{Individual-level BrainXcan results on 9 UK Biobank phenotypes.} 
%     \textbf{model}: The prediction model type of the IDP predictor (either ridge or elastic net);
%     \textbf{IDP}: IDP identifier; 
%     \textbf{phenotype}: The name of the phenotype (defined in table~\ref{S-tab:indiv_bxcan_phenotype});
%     \textbf{bhat}: BrainXcan effect size estimate, $\hat\beta$;
%     \textbf{pval}: BrainXcan p-value (without adjustment by permutation);
%     \textbf{zscore}: BrainXcan z-score (without adjustment by permutation).   
%   }
%   \label{S-tab:indiv_bxcan_results}
%   % \end{footnotesize}
% \end{table}

\begin{table}[H]
  % \begin{footnotesize}
  \caption[The list of 35 GWASs being analyzed by S-BrainXcan]{\textbf{The list of 35 GWAS  analyzed by S-BrainXcan.} 
  % The table is available at \url{https://docs.google.com/spreadsheets/d/1jjDZZBsRNTvgidXXfJqXNOvFQ1U9Ree0G-WFcXtWGAg}. 
    It contains the information about the 35 GWASs used in S-BrainXcan analysis. 
    \textbf{phenotype}: The name of the phenotype;
    \textbf{phenotype\_id}: The phenotype identifiers;
    \textbf{short\_name}: The short names of the phenotypes;
    \textbf{sample\_size}: The sample size of the GWAS. If there are different sample sizes for different SNPs, the mean sample size is shown;
    \textbf{portal}: The website from which the GWAS was downloaded;
    \textbf{filename}: The name of the raw GWAS file being downloaded.   
  }
  \label{S-tab:s_bxcan_phenotype}
  % \end{footnotesize}
\end{table}

\begin{table}[H]
  % \begin{footnotesize}
  \caption[S-BrainXcan results for 35 GWASs]{\textbf{S-BrainXcan results for 35 GWASs.} 
    \textbf{model}: The prediction model type of the IDP predictor (either ridge or elastic net);
    \textbf{IDP}: IDP identifier; 
    \textbf{phenotype}: The name of the phenotype (defined in table~\ref{S-tab:s_bxcan_phenotype});
    \textbf{bhat}: BrainXcan effect size estimate, $\hat\beta$;
    \textbf{pval}: BrainXcan p-value (without adjustment by permutation);
    \textbf{zscore}: BrainXcan z-score (without adjustment by permutation);
    \textbf{nsnp\_used}: The number of variants used in S-BrainXcan computation;
    \textbf{nsnp\_total}: The number of variants in the prediction model (\textbf{nsnp\_used} should be no larger than \textbf{nsnp\_total} but the closer the better since the difference is the number of variants being missed by the GWAS);
    \textbf{z\_adj\_perm\_null}: BrainXcan z-score with adjustment by permutation;
    \textbf{pval\_adj\_perm\_null}: The corresponding p-value computed from the adjusted BrainXcan z-score (\textbf{z\_adj\_perm\_null}).
  }
  \label{S-tab:s_bxcan_results}
  % \end{footnotesize}
\end{table}


\begin{table}[H]
  % \begin{footnotesize}
  \caption[LDSC based genetic correlation results for 35 GWASs]{\textbf{LDSC based genetic correlation results for 35 GWASs.} 
    \textbf{IDP}: IDP identifier; 
    \textbf{phenotype}: The name of the phenotype (defined in table~\ref{S-tab:s_bxcan_phenotype});
    \textbf{rg}: Genetic correlation;
    \textbf{pval}: Genetic correlation p-value (null hypothesis: \textbf{rg} is zero);
    \textbf{zscore}: The corresponding z-score calculated from \textbf{pval}.
  }
  \label{S-tab:gencor_results}
  % \end{footnotesize}
\end{table}

```

